{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Win-funcs-begin\" data-toc-modified-id=\"Win-funcs-begin-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Win funcs begin</a></span></li><li><span><a href=\"#Process-window\" data-toc-modified-id=\"Process-window-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Process window</a></span></li><li><span><a href=\"#Call-func\" data-toc-modified-id=\"Call-func-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Call func</a></span></li><li><span><a href=\"#Merge-outputs\" data-toc-modified-id=\"Merge-outputs-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Merge outputs</a></span></li><li><span><a href=\"#Compare-vcf-file-sizes-(cluster-vs-source-ftp-server)\" data-toc-modified-id=\"Compare-vcf-file-sizes-(cluster-vs-source-ftp-server)-5\"><span class=\"toc-item-num\">5&nbsp;&nbsp;</span>Compare vcf file sizes (cluster vs source ftp server)</a></span></li></ul></div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# validate gz format\n",
    "import gzip\n",
    "pathgz = r\"C:\\Data\\HUJI\\hgdp\\classes\\windows\\maf_0.49\\count_dist_window_0.tsv.gz\"\n",
    "path = r\"C:\\Data\\HUJI\\hgdp\\classes\\windows\\maf_0.49\\count_dist_window_0.tsv\"\n",
    "with gzip.open(pathgz, 'rb') as f:\n",
    "    file_contentgz = f.read()\n",
    "\n",
    "with open(path, 'r') as f:\n",
    "    file_content = f.read()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "file_contentgzs = file_contentgz.decode()\n",
    "for i in range(len(file_content)):\n",
    "    if file_content[i]!=file_contentgzs[i]:\n",
    "        print(i, file_content[i],file_contentgzs[i])\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T10:20:03.619723Z",
     "start_time": "2021-01-27T10:20:03.591249Z"
    }
   },
   "outputs": [],
   "source": [
    "# todo - think about the infra to get to each file, and naming conventions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Process 012 files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T10:20:07.531630Z",
     "start_time": "2021-01-27T10:20:07.516136Z"
    }
   },
   "outputs": [],
   "source": [
    "#imports and consts\n",
    "import os.path\n",
    "import logging\n",
    "import pandas as pd\n",
    "import math\n",
    "\n",
    "SUFFIX_POS = '.012.pos'\n",
    "SUFFIX_012 = '.012'\n",
    "OUTPUT_PATTERN_DIST_FILE = 'count_dist_slice_{slice_index}.tsv'\n",
    "OUTPUT_PATTERN_SITES_FILE = 'sites_ids_slice_{slice_index}.tsv'\n",
    "OUTPUT_PATTERN_LOG_FILE = 'Process012_{chr_name}_MAF_{maf}.log'\n",
    "\n",
    "logger = logging.getLogger(__name__)\n",
    "\n",
    "def _set_logger(chr_name, maf):\n",
    "    log_file = OUTPUT_PATTERN_LOG_FILE.format(chr_name=chr_name, maf=maf)\n",
    "    fhandler = logging.FileHandler(filename=log_file, mode='a')\n",
    "    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')\n",
    "    fhandler.setFormatter(formatter)\n",
    "    # TODO - file is empty!\n",
    "    logger.addHandler(fhandler)\n",
    "\n",
    "def _log(msg, level=logging.INFO):\n",
    "    print(msg)\n",
    "    logger.log(level, msg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:13:55.814866Z",
     "start_time": "2021-02-03T08:13:55.761467Z"
    }
   },
   "outputs": [],
   "source": [
    "def _validate_file_exist(p):\n",
    "    if not os.path.isfile(p):\n",
    "        print(f'Cant file file {p}')\n",
    "        \n",
    "def _get_chr_name(path_pos):\n",
    "    with open(path_pos, 'r') as f:\n",
    "        l = f.readline()\n",
    "        char_name = l.split('\\t')[0]\n",
    "        return char_name\n",
    "    \n",
    "def _prepare_output_folder(base_output_path, chr_name):\n",
    "    output_dir = f'{base_output_path}/{chr_name}/'\n",
    "    print(f'output_dir is {output_dir}')\n",
    "    os.makedirs(output_dir, exist_ok=True)\n",
    "    return output_dir\n",
    "\n",
    "\n",
    "def _build_pairwise_db(number, value):\n",
    "    return [[value]*(number-i) for i in range (1,number)]\n",
    "\n",
    "def _slice_calc_pairwise_distances_with_guardrails(slice_index, num_of_slices, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected):\n",
    "    # for each column, we calc the pairwise distances and add it to the grand total\n",
    "    # for performance, we use 2 lists of lists, one for distances and one for counts\n",
    "    window_pairwise_counts = _build_pairwise_db(len(slice_df), 0)\n",
    "    window_pairwise_dist = _build_pairwise_db(len(slice_df), 0.0)\n",
    "    for site_index in range(len(slice_df.columns)):\n",
    "        _site_calc_pairwise_distances_with_guardrails(slice_index, num_of_slices, site_index, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, window_pairwise_counts, window_pairwise_dist)\n",
    "    return window_pairwise_counts, window_pairwise_dist\n",
    "\n",
    "\n",
    "def _site_calc_pairwise_distances(slice_index, num_of_slices, site_index, genotypes, num_individuals, ref_freq, non_ref_freq, window_pairwise_counts, window_pairwise_dist):\n",
    "    for i1 in range(num_individuals-1):\n",
    "        i1_val = genotypes[i1]\n",
    "        # if this entry is not valid for i1, no need to go over all the others, nothing to add to freq nor counts\n",
    "        if i1%100==0:\n",
    "            print(f'Slice: {slice_index}/{num_of_slices}. Site: {site_index}. Done with individual {i1}/{len(genotypes)}')\n",
    "        if i1_val == -1:\n",
    "            continue\n",
    "        for i2 in range(i1+1, num_individuals):\n",
    "            i2_val = genotypes[i2]\n",
    "            if i2_val == -1:\n",
    "                continue\n",
    "            else:            \n",
    "                # this is a valid entry, we add 1 to the count\n",
    "                window_pairwise_counts[i1][i2-i1-1] += 1\n",
    "                window_pairwise_dist[i1][i2-i1-1] += _calc_dist(i1_val, i2_val, ref_freq, non_ref_freq)\n",
    "\n",
    "def _site_calc_pairwise_distances_with_guardrails(slice_index, num_of_slices, site_index, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, window_pairwise_counts, window_pairwise_dist):\n",
    "    genotypes = slice_df.iloc[:,site_index].values\n",
    "    # get counts\n",
    "    num_individuals = len(genotypes)\n",
    "    # count the amount of not -1 in alleles\n",
    "    num_valid_genotypes = len(genotypes[genotypes!=-1])\n",
    "    non_ref_count = sum(genotypes[genotypes>0])\n",
    "    ref_count = 2*num_valid_genotypes-non_ref_count\n",
    "    non_ref_freq = float(non_ref_count)/(2*num_valid_genotypes)\n",
    "    ref_freq = float(ref_count)/(2*num_valid_genotypes)\n",
    "    print(f'Site index: {site_index}, non ref allele frequency: {non_ref_freq}')\n",
    "    print(f'Site index: {site_index}, ref allele frequency: {ref_freq}')\n",
    "    # guardrails \n",
    "    assert abs(ref_freq+non_ref_freq-1)<1e-04\n",
    "    _check_guardrails(num_individuals, num_valid_genotypes, ref_count, non_ref_count, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected)\n",
    "\n",
    "    # now when we have all freq and counts, we can start the pairwise comparison\n",
    "    # takes 0.5 seconds for one site. This means ~4 days if we are not seperating the task (which we will)\n",
    "    # 178 sites took arounc 2 minutes.\n",
    "    # Note that if we split this by chromozome, the big ones may have around 1000 in a given freq range, \n",
    "    # which can take ~8 minutes. 10k will be around 83 minutes - also cool.\n",
    "    _site_calc_pairwise_distances(slice_index, num_of_slices, site_index, genotypes, num_individuals, ref_freq, non_ref_freq, window_pairwise_counts, window_pairwise_dist)\n",
    "\n",
    "    \n",
    "def _check_guardrails(num_individuals, num_valid_genotypes, ref_count, non_ref_count, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected):\n",
    "    print(f'Check guardrails')\n",
    "    # guardrail #1 - min_valid_sites_precentage\n",
    "    percentage_valid_sites = float(num_valid_genotypes)/num_individuals\n",
    "    print(f'Precentage of valid sites: {percentage_valid_sites}')\n",
    "    if percentage_valid_sites < min_valid_sites_precentage:\n",
    "        print(f'ERROR: % of valid sites is {percentage_valid_sites}, lower than allowd: {min_valid_sites_precentage}.')\n",
    "        assert percentage_valid_sites < min_valid_sites_precentage\n",
    "\n",
    "    # guardrail #2 mac/maf validation\n",
    "    if (min_minor_freq_expected==-1 or max_minor_freq_expected==-1) and (min_minor_count_expected==-1 or max_minor_count_expected==-1):\n",
    "        print(f'ERROR: min_minor_freq_expected, max_minor_freq_expected or min_minor_count_expected, max_minor_count_expected must be >-1')\n",
    "        assert (min_minor_freq_expected==-1 or max_minor_freq_expected==-1) and (min_minor_count_expected==-1 or max_minor_count_expected==-1)\n",
    "    \n",
    "    #if maf: validate min_minor_freq_expected and max_minor_freq_expected\n",
    "    if min_minor_freq_expected>-1 and max_minor_freq_expected>-1:\n",
    "        minor_count = min(ref_count, non_ref_count)\n",
    "        minor_freq = float(minor_count)/(2*num_valid_genotypes)\n",
    "        print(f'Minor allele frequency: {minor_freq}')\n",
    "        if minor_freq < min_minor_freq_expected:\n",
    "            print(f'ERROR: minor frequency is too low - {minor_freq}, allowd: {min_minor_freq_expected}.')\n",
    "            assert minor_freq < min_minor_freq_expected\n",
    "        if minor_freq > max_minor_freq_expected:\n",
    "            print(f'ERROR: minor frequency is too high - {minor_freq}, allowd: {max_minor_freq_expected}.')\n",
    "            assert minor_freq > max_minor_freq_expected\n",
    "    \n",
    "    #if mac: validate min_minor_count_expected and max_minor_count_expected\n",
    "    if min_minor_count_expected>-1 and max_minor_count_expected>-1:\n",
    "        minor_count = min(ref_count, non_ref_count)\n",
    "        print(f'Minor allele count: {minor_count}')\n",
    "        if minor_count < min_minor_count_expected:\n",
    "            print(f'ERROR: minor frequency is too low - {minor_freq}, allowd: {min_minor_freq_expected}.')\n",
    "            assert minor_count < min_minor_count_expected\n",
    "        if minor_count > max_minor_count_expected:\n",
    "            print(f'ERROR: minor frequency is too high - {minor_freq}, allowd: {max_minor_freq_expected}.')\n",
    "            assert minor_count > max_minor_count_expected\n",
    "    print(f'Passed guardrails')\n",
    "    \n",
    "    \n",
    "def _write_output_sites_file(output_sites_file, slice_df):\n",
    "    # TODO - output per site the calculated number of missing values and MAF\n",
    "    sites_ids = slice_df.columns.values\n",
    "    with open(output_sites_file, 'w') as f:\n",
    "        for site_id in sites_ids:\n",
    "            f.write(site_id.replace('_','\\t') + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:40:02.744247Z",
     "start_time": "2021-02-03T08:40:02.728490Z"
    }
   },
   "outputs": [],
   "source": [
    "def _calc_dist(i1_val, i2_val, ref_freq, non_ref_freq):\n",
    "    #print(i1_val, i2_val, ref_freq, non_ref_freq)\n",
    "    # from VCFtools manual:\n",
    "    # \"Genotypes are represented as 0, 1 and 2, where the number represent that number of non-reference alleles\"\n",
    "    # So - v1_val and v2_val are the amount of ***non-ref alleles***.\n",
    "    # The distance function is:\n",
    "    # 1/4[(1-f_a)(Iac+Iad) + (1-f_b)(Ibc+Ibd)]\n",
    "    # Now, for each combination of v1_val and v2_val, we can compute the distance.\n",
    "        \n",
    "    # 0,0 - v1_val=0 and v2_val=0:\n",
    "    #    1/4[(1-f_ref)(1+1) + (1-f_ref)(1+1)] = 1/4[(1-f_ref)(4)] = (1-f_ref)\n",
    "    \n",
    "    # 0,1 - v1_val=0 and v2_val=1:\n",
    "    #    1/4[(1-f_ref)(1+0) + (1-f_ref)(1+0)] = 1/4[(1-f_ref)(2)] = 1/2(1-f_ref)\n",
    "    \n",
    "    # 0,2 - v1_val=0 and v2_val=2:\n",
    "    #    1/4[(1-f_ref)(0+0) + (1-f_ref)(0+0)] = 0\n",
    "    \n",
    "    # 1,1 - v1_val=1 and v2_val=1:\n",
    "    #    1/4[(1-f_non_ref)(1+0) + (1-f_ref)(1+0)] = 1/4[(1-f_non_ref)+(1-f_ref)]\n",
    "    # note: if f_non_ref+f_ref=1, we get 1/4\n",
    "    \n",
    "    # 1,2 - v1_val=1 and v2_val=2:\n",
    "    #    1/4[(1-f_non_ref)(1+1) + (1-f_ref)(0+0)] = 1/4[(1-f_non_ref)(2)] = 1/2(1-f_non_ref)\n",
    "    \n",
    "    # 2,2 - v1_val=2 and v2_val=2:\n",
    "    #    1/4[(1-f_non_ref)(1+1) + (1-f_non_ref)(1+1)] = 1/4[(1-f_non_ref)(4)] = (1-f_non_ref)\n",
    "    \n",
    "    # Also, note that it is symetric:\n",
    "\n",
    "    # 1,0 - v1_val=1 and v2_val=0:\n",
    "    #    1/4[(1-f_ref)(1+1) + (1-f_non_ref)(0+0)] = 1/4[(1-f_ref)(2)] = 1/2(1-f_ref)\n",
    "    \n",
    "    # 2,1 - v1_val=2 and v2_val=1:\n",
    "    #    1/4[(1-f_non_ref)(1+0) + (1-f_non_ref)(0+1)] = 1/4[(1-f_non_ref)(2)] = 1/2(1-f_non_ref)\n",
    "\n",
    "    # 2,0 - v1_val=2 and v2_val=0:\n",
    "    #    1/4[(1-f_non_ref)(0+0) + (1-f_non_ref)(0+0)] = 0\n",
    "    \n",
    "    # formula to use: \n",
    "    #   (1-f_non_ref)(v1_val*v2_val) + (1-f_ref)((2-v1_val)*(2-v2_val))/4\n",
    "    return ((1-non_ref_freq)*(i1_val*i2_val) + (1-ref_freq)*((2-i1_val)*(2-i2_val)))/4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:40:03.175661Z",
     "start_time": "2021-02-03T08:40:03.145184Z"
    }
   },
   "outputs": [],
   "source": [
    "def _calc_dist_directly(i1_val, i2_val, ref_freq, non_ref_freq):\n",
    "    # set a,b\n",
    "    # both ref\n",
    "    if i1_val == 0:\n",
    "        a='ref'\n",
    "        b='ref'\n",
    "    if i1_val == 1:\n",
    "        a='non_ref'\n",
    "        b='ref'\n",
    "    # both non ref\n",
    "    if i1_val == 2:\n",
    "        a='non_ref'\n",
    "        b='non_ref'\n",
    "    # set c,d\n",
    "    if i2_val == 0:\n",
    "        c='ref'\n",
    "        d='ref'\n",
    "    if i2_val == 1:\n",
    "        c='non_ref'\n",
    "        d='ref'\n",
    "    if i2_val == 2:\n",
    "        c='non_ref'\n",
    "        d='non_ref'\n",
    "    Iac = 1 if a==c else 0\n",
    "    Iad = 1 if a==d else 0\n",
    "    Ibc = 1 if b==c else 0\n",
    "    Ibd = 1 if b==d else 0\n",
    "    f_a = ref_freq if a=='ref' else non_ref_freq\n",
    "    f_b = ref_freq if b=='ref' else non_ref_freq\n",
    "    return (((1-f_a)*(Iac+Iad)) + ((1-f_b)*(Ibc+Ibd)))/4.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:40:41.529598Z",
     "start_time": "2021-02-03T08:40:37.570731Z"
    }
   },
   "outputs": [],
   "source": [
    "for i in range(10000000):\n",
    "    _calc_dist(0,2,f_ref,f_non_ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:40:48.486519Z",
     "start_time": "2021-02-03T08:40:42.616086Z"
    }
   },
   "outputs": [],
   "source": [
    "for i in range(10000000):\n",
    "    _calc_dist_directly(0,2,f_ref,f_non_ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:38:45.425844Z",
     "start_time": "2021-02-03T08:38:45.387211Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 0 0.1 0.8\n",
      "0 1 0.1 0.8\n",
      "2 0 0.1 0.8\n",
      "0 2 0.1 0.8\n",
      "2 1 0.1 0.8\n",
      "1 2 0.1 0.8\n",
      "0 0 0.1 0.8\n",
      "0 1 0.1 0.8\n",
      "0 2 0.1 0.8\n",
      "1 2 0.1 0.8\n",
      "2 2 0.1 0.8\n"
     ]
    }
   ],
   "source": [
    "# validate _calc_dist\n",
    "f_ref = 0.1\n",
    "f_non_ref = 0.8\n",
    "# assert symatry\n",
    "assert _calc_dist(1,0,f_ref,f_non_ref) == _calc_dist(0,1,f_ref,f_non_ref)\n",
    "assert _calc_dist(2,0,f_ref,f_non_ref) == _calc_dist(0,2,f_ref,f_non_ref)\n",
    "assert _calc_dist(2,1,f_ref,f_non_ref) == _calc_dist(1,2,f_ref,f_non_ref)\n",
    "# assert calc\n",
    "assert _calc_dist(0,0,f_ref,f_non_ref) == _calc_dist_directly(0,0,f_ref,f_non_ref)\n",
    "assert _calc_dist(0,1,f_ref,f_non_ref) == _calc_dist_directly(0,1,f_ref,f_non_ref)\n",
    "assert _calc_dist(0,2,f_ref,f_non_ref) == _calc_dist_directly(0,2,f_ref,f_non_ref)\n",
    "assert _calc_dist(1,2,f_ref,f_non_ref) == _calc_dist_directly(1,2,f_ref,f_non_ref)\n",
    "assert _calc_dist(2,2,f_ref,f_non_ref) == _calc_dist_directly(2,2,f_ref,f_non_ref)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T09:43:08.891414Z",
     "start_time": "2021-02-03T09:43:08.868417Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.16666666666666666"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#-1 1 2 2\n",
    "#--rnnnnn\n",
    "nr = 5/6\n",
    "r=1/6\n",
    "r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T09:43:45.802775Z",
     "start_time": "2021-02-03T09:43:45.790766Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.08333333333333331"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_calc_dist(1,2,r,nr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:55:36.625315Z",
     "start_time": "2021-02-03T08:55:36.581326Z"
    }
   },
   "outputs": [],
   "source": [
    "expected_distances_per_colus = {\n",
    "    # locus 1\n",
    "    '1':{\n",
    "        '1':{\n",
    "            '2':0.25,\n",
    "            '3':0.375,\n",
    "            '4':0.375,\n",
    "        },\n",
    "        '2':{\n",
    "            '3':0.375,\n",
    "            '4':0.375,\n",
    "        },\n",
    "        '3':{\n",
    "            '4':0.75,\n",
    "        }\n",
    "    },\n",
    "    # locus 2\n",
    "    '2':{\n",
    "        '1':{\n",
    "            '2':0.75,\n",
    "            '3':0.375,\n",
    "            '4':0.375,\n",
    "        },\n",
    "        '2':{\n",
    "            '3':0.375,\n",
    "            '4':0.375,\n",
    "        },\n",
    "        '3':{\n",
    "            '4':0.25,\n",
    "        }\n",
    "    },\n",
    "    # locus 3\n",
    "    '3':{\n",
    "        '1':{\n",
    "            '2':0.5,\n",
    "            '3':0.0,\n",
    "            '4':0.0,\n",
    "        },\n",
    "        '2':{\n",
    "            '3':0.0,\n",
    "            '4':0.0,\n",
    "        },\n",
    "        '3':{\n",
    "            '4':0.5,\n",
    "        }\n",
    "    },\n",
    "    # locus 4\n",
    "    '4':{\n",
    "        '2':{\n",
    "            '3':0.41667,\n",
    "            '4':0.41667,\n",
    "        },\n",
    "        '3':{\n",
    "            '4':0.83334,\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:55:51.005717Z",
     "start_time": "2021-02-03T08:55:50.990057Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1': {'1': {'2': 0.25, '3': 0.375, '4': 0.375},\n",
       "  '2': {'3': 0.375, '4': 0.375},\n",
       "  '3': {'4': 0.75}},\n",
       " '2': {'1': {'2': 0.75, '3': 0.375, '4': 0.375},\n",
       "  '2': {'3': 0.375, '4': 0.375},\n",
       "  '3': {'4': 0.25}},\n",
       " '3': {'1': {'2': 0.5, '3': 0.0, '4': 0.0},\n",
       "  '2': {'3': 0.0, '4': 0.0},\n",
       "  '3': {'4': 0.5}},\n",
       " '4': {'2': {'3': 0.41667, '4': 0.41667}, '3': {'4': 0.83334}}}"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expected_distances_per_colus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T10:17:45.770869Z",
     "start_time": "2021-01-27T10:17:45.752695Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 (1-f_a)0 + (1-f_r)4\n",
      "0 1 (1-f_a)0 + (1-f_r)2\n",
      "0 2 (1-f_a)0 + (1-f_r)0\n",
      "1 1 (1-f_a)1 + (1-f_r)1\n",
      "1 2 (1-f_a)2 + (1-f_r)0\n",
      "2 2 (1-f_a)4 + (1-f_r)0\n"
     ]
    }
   ],
   "source": [
    "def _test(v1_val, v2_val):\n",
    "    print(v1_val, v2_val, f'(1-f_a){(v1_val*v2_val)} + (1-f_r){(2-v1_val)*(2-v2_val)}')\n",
    "for v1 in range(3):\n",
    "    for v2 in range(v1,3):\n",
    "        _test(v1, v2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Win funcs begin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T07:55:35.422751Z",
     "start_time": "2021-02-03T07:55:35.378112Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "def get_window(classes_folder, mac_maf, class_name, window_index):\n",
    "    class_012_path_template = classes_folder + \"chr{chr_id}\\{class_name}.012\"\n",
    "    windows_indexes_files_folder = classes_folder + r\"windows_indexes/\"\n",
    "    windows_indexes_path_template = windows_indexes_files_folder + \"windows_indexes_for_class_{class_name}.json\"\n",
    "\n",
    "    print(f'Class: {mac_maf}_{class_name}, window index: {window_index}')\n",
    "    # read indexes of window\n",
    "    windows_indexes_path = windows_indexes_path_template.format(class_name=class_name)\n",
    "    with open(windows_indexes_path) as f:\n",
    "        windows_indexes = json.load(f)\n",
    "    window_indexes = windows_indexes[window_index]\n",
    "    print(f'There are {len(window_indexes)} indexes in window indexes list')\n",
    "    chr_id2indexes = dict()\n",
    "    for char_id_index in window_indexes:\n",
    "        chr_id, index = char_id_index.split(';',1)\n",
    "        # if chr_id not in the dict, we get an empty list\n",
    "        indexes = chr_id2indexes.get(chr_id, [])\n",
    "        indexes.append(int(index))\n",
    "        chr_id2indexes[chr_id] = indexes\n",
    "    #sort the indexes so that we can easily read the file\n",
    "    {k:v.sort() for k, v in chr_id2indexes.items()}\n",
    "\n",
    "    # go over the chr_ids, and get from the file the correct columns\n",
    "    class_012_df = pd.DataFrame()\n",
    "    for chr_id, indexes in chr_id2indexes.items():\n",
    "        class_012_path = class_012_path_template.format(chr_id = chr_id, mac_maf = mac_maf, class_name = class_name)\n",
    "        # get number of columns in chr:\n",
    "        with open(class_012_path,'r') as f:\n",
    "            num_columns = len(f.readline().split('\\t'))\n",
    "        print(f'{len(indexes)} / {num_columns-1} sites will be used from file {class_012_path}')\n",
    "        names = [f'chr{chr_id}_idx{i}' for i in range(num_columns)]\n",
    "        # we add one as the csv contains the individual is in the first index\n",
    "        cols_to_uset = [i+1 for i in indexes]\n",
    "        # read file to pandas df\n",
    "        selected_indexes_012_df = pd.read_csv(class_012_path, sep='\\t', names= names, usecols = cols_to_uset)\n",
    "        if class_012_df.empty:\n",
    "            class_012_df = selected_indexes_012_df\n",
    "        else:\n",
    "            class_012_df = class_012_df.join(selected_indexes_012_df)\n",
    "    # validate the number of columns is the number of window indexes\n",
    "    assert class_012_df.shape[1] == len(window_indexes)\n",
    "    print(f'{class_012_df.shape[1]} sites in window')\n",
    "    return class_012_df\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T07:53:43.809850Z",
     "start_time": "2021-02-03T07:53:43.794493Z"
    }
   },
   "outputs": [],
   "source": [
    "def window_calc_pairwise_distances_with_guardrails(window_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected):\n",
    "    # for each column, we calc the pairwise distances and add it to the grand total\n",
    "    # for performance, we use 2 lists of lists, one for distances and one for counts\n",
    "    window_pairwise_counts = _build_pairwise_db(len(window_df), 0)\n",
    "    window_pairwise_dist = _build_pairwise_db(len(window_df), 0.0)\n",
    "    for site_index in range(len(window_df.columns)):\n",
    "        site_calc_pairwise_distances_with_guardrails(window_df, site_index, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected, window_pairwise_counts, window_pairwise_dist)\n",
    "    return window_pairwise_counts, window_pairwise_dist\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T07:57:47.137534Z",
     "start_time": "2021-02-03T07:57:47.121921Z"
    }
   },
   "outputs": [],
   "source": [
    "def site_calc_pairwise_distances_with_guardrails(window_df, site_index, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected, window_pairwise_counts, window_pairwise_dist):\n",
    "    genotypes = window_df.iloc[:,site_index].values\n",
    "    print('genotypes', genotypes)\n",
    "    # get counts\n",
    "    num_individuals = len(genotypes)\n",
    "    print('num_individuals', num_individuals)\n",
    "    # count the amount of not -1 in alleles\n",
    "    num_valid_genotypes = len(genotypes[genotypes!=-1])\n",
    "    print('num_valid_genotypes', num_valid_genotypes)\n",
    "    non_ref_count = sum(genotypes[genotypes>0])\n",
    "    ref_count = 2*num_valid_genotypes-non_ref_count\n",
    "    non_ref_freq = float(non_ref_count)/(2*num_valid_genotypes)\n",
    "    ref_freq = float(ref_count)/(2*num_valid_genotypes)\n",
    "    print(f'Site index: {site_index}, non ref allele frequency: {non_ref_freq}')\n",
    "    print(f'Site index: {site_index}, ref allele frequency: {ref_freq}')\n",
    "    # guardrails \n",
    "    assert abs(ref_freq+non_ref_freq-1)<1e-04\n",
    "    _check_guardrails(num_individuals, num_valid_genotypes, ref_count, non_ref_count, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected)\n",
    "    site_calc_pairwise_distances(genotypes, num_individuals, ref_freq, non_ref_freq, window_pairwise_counts, window_pairwise_dist)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:02:26.569459Z",
     "start_time": "2021-02-03T08:02:26.553919Z"
    }
   },
   "outputs": [],
   "source": [
    "def site_calc_pairwise_distances(genotypes, num_individuals, ref_freq, non_ref_freq, window_pairwise_counts, window_pairwise_dist):\n",
    "    for i1 in range(num_individuals-1):\n",
    "        i1_val = genotypes[i1]\n",
    "        print('i1', i1, 'i1_val', i1_val)\n",
    "        # if this entry is not valid for i1, no need to go over all the others, nothing to add to freq nor counts\n",
    "        if i1%100==0:\n",
    "            print(f'Done with individual {i1}/{len(genotypes)}')\n",
    "        if i1_val == -1:\n",
    "            continue\n",
    "        for i2 in range(i1+1, num_individuals):\n",
    "            i2_val = genotypes[i2]\n",
    "            print('i2', i2, 'i2_val', i2_val)\n",
    "            if i2_val == -1:\n",
    "                continue\n",
    "            else:            \n",
    "                # this is a valid entry, we add 1 to the count\n",
    "                window_pairwise_counts[i1][i2-i1-1] += 1\n",
    "                print('dist',_calc_dist(i1_val, i2_val, ref_freq, non_ref_freq))\n",
    "                window_pairwise_dist[i1][i2-i1-1] += _calc_dist(i1_val, i2_val, ref_freq, non_ref_freq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T07:57:49.611378Z",
     "start_time": "2021-02-03T07:57:49.569510Z"
    }
   },
   "outputs": [],
   "source": [
    "# the output is in couples of <count>,<distance>\n",
    "# the count is the number of valied sites on which the distances is calculated\n",
    "def write_pairwise_distances(output_count_dist_file, window_pairwise_counts, window_pairwise_dist):\n",
    "    with open(output_count_dist_file,'w') as f:\n",
    "        for counts,dists in zip(window_pairwise_counts, window_pairwise_dist):\n",
    "            s = ' '.join(f'{c}{round(d, 5)}' for c,d in zip(counts, dists)) + '\\n'\n",
    "            f.write(s)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process window"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T07:54:43.909690Z",
     "start_time": "2021-02-03T07:54:43.893780Z"
    }
   },
   "outputs": [],
   "source": [
    "class_name='all'\n",
    "# PARAMS\n",
    "# UTILS FOR PARAMS\n",
    "classes_folder = r\"../mock/classes/\"\n",
    "class_012_path_template = classes_folder + \"chr{chr_id}/{class_name}.012\"\n",
    "windows_indexes_files_folder = classes_folder + r\"windows_indexes/\"\n",
    "windows_indexes_path_template = windows_indexes_files_folder + \"windows_indexes_for_class_{class_name}.json\"\n",
    "\n",
    "# if we have less than this which are valid (not -1), site is not included in calc.\n",
    "min_valid_sites_precentage = 0.1\n",
    "# if not in range - will raise a warning\n",
    "mac_maf = 'maf'\n",
    "min_minor_freq_expected = 0.0\n",
    "max_minor_freq_expected = 0.5\n",
    "min_minor_count_expected = -1\n",
    "max_minor_count_expected = -1\n",
    "class_name = 'all'\n",
    "\n",
    "windows_indexes_path = windows_indexes_path_template.format(class_name=class_name)\n",
    "output_dir = f'{classes_folder}windows/{class_name}/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:14:48.473562Z",
     "start_time": "2021-02-03T08:14:48.430953Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class: maf_all, window index: 0\n",
      "There are 1 indexes in window indexes list\n",
      "1 / 4 sites will be used from file ../mock/classes/chr1\\all.012\n",
      "1 sites in window\n",
      "   chr1_idx1\n",
      "0          1\n",
      "1          1\n",
      "2          0\n",
      "3          0\n",
      "Site index: 0, non ref allele frequency: 0.25\n",
      "Site index: 0, ref allele frequency: 0.75\n",
      "Check guardrails\n",
      "Precentage of valid sites: 1.0\n",
      "Minor allele frequency: 0.25\n",
      "Passed guardrails\n",
      "i1 0 i1_val 1\n",
      "Done with individual 0/4\n",
      "i2 1 i2_val 1\n",
      "1 1 0.75 0.25\n",
      "dist 0.25\n",
      "1 1 0.75 0.25\n",
      "i2 2 i2_val 0\n",
      "1 0 0.75 0.25\n",
      "dist 0.125\n",
      "1 0 0.75 0.25\n",
      "i2 3 i2_val 0\n",
      "1 0 0.75 0.25\n",
      "dist 0.125\n",
      "1 0 0.75 0.25\n",
      "i1 1 i1_val 1\n",
      "i2 2 i2_val 0\n",
      "1 0 0.75 0.25\n",
      "dist 0.125\n",
      "1 0 0.75 0.25\n",
      "i2 3 i2_val 0\n",
      "1 0 0.75 0.25\n",
      "dist 0.125\n",
      "1 0 0.75 0.25\n",
      "i1 2 i1_val 0\n",
      "i2 3 i2_val 0\n",
      "0 0 0.75 0.25\n",
      "dist 0.25\n",
      "0 0 0.75 0.25\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "([[1, 1, 1], [1, 1], [1]], [[0.25, 0.125, 0.125], [0.125, 0.125], [0.25]])"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "window_df = get_window(classes_folder, mac_maf, class_name, window_index=0)\n",
    "print(window_df)\n",
    "window_pairwise_counts, window_pairwise_dist = window_calc_pairwise_distances_with_guardrails(window_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected)\n",
    "window_pairwise_counts, window_pairwise_dist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-03T08:03:40.283359Z",
     "start_time": "2021-02-03T08:03:40.252099Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class: maf_all, window index: 0\n",
      "There are 1 indexes in window indexes list\n",
      "1 / 4 sites will be used from file ../mock/classes/chr1\\all.012\n",
      "1 sites in window\n",
      "   chr1_idx1\n",
      "0          1\n",
      "1          1\n",
      "2          0\n",
      "3          0\n",
      "Site index: 0, non ref allele frequency: 0.25\n",
      "Site index: 0, ref allele frequency: 0.75\n",
      "Check guardrails\n",
      "Precentage of valid sites: 1.0\n",
      "Minor allele frequency: 0.25\n",
      "Passed guardrails\n",
      "i1 0 i1_val 1\n",
      "Done with individual 0/4\n",
      "i2 1 i2_val 1\n",
      "dist 0.8125\n",
      "i2 2 i2_val 0\n",
      "dist 0.125\n",
      "i2 3 i2_val 0\n",
      "dist 0.125\n",
      "i1 1 i1_val 1\n",
      "i2 2 i2_val 0\n",
      "dist 0.125\n",
      "i2 3 i2_val 0\n",
      "dist 0.125\n",
      "i1 2 i1_val 0\n",
      "i2 3 i2_val 0\n",
      "dist 0.25\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "([[1, 1, 1], [1, 1], [1]], [[0.8125, 0.125, 0.125], [0.125, 0.125], [0.25]])"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "window_df = get_window(classes_folder, mac_maf, class_name, window_index=0)\n",
    "print(window_df)\n",
    "window_pairwise_counts, window_pairwise_dist = window_calc_pairwise_distances_with_guardrails(window_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected)\n",
    "window_pairwise_counts, window_pairwise_dist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T10:25:12.659345Z",
     "start_time": "2021-01-27T10:20:28.740601Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Site index: 0, non ref allele frequency: 0.4924406047516199\n",
      "Site index: 0, ref allele frequency: 0.5075593952483801\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4924406047516199\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.4924078091106291\n",
      "Site index: 1, ref allele frequency: 0.5075921908893709\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.4924078091106291\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.5037837837837837\n",
      "Site index: 2, ref allele frequency: 0.4962162162162162\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4962162162162162\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.49838187702265374\n",
      "Site index: 3, ref allele frequency: 0.5016181229773463\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49838187702265374\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.49243243243243245\n",
      "Site index: 4, ref allele frequency: 0.5075675675675676\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49243243243243245\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.4983783783783784\n",
      "Site index: 5, ref allele frequency: 0.5016216216216216\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4983783783783784\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.4972972972972973\n",
      "Site index: 6, ref allele frequency: 0.5027027027027027\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4972972972972973\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.4972972972972973\n",
      "Site index: 7, ref allele frequency: 0.5027027027027027\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4972972972972973\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.5081168831168831\n",
      "Site index: 8, ref allele frequency: 0.49188311688311687\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49188311688311687\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.5005494505494505\n",
      "Site index: 9, ref allele frequency: 0.49945054945054945\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9795479009687836\n",
      "Minor allele frequency: 0.49945054945054945\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.4971719457013575\n",
      "Site index: 10, ref allele frequency: 0.5028280542986425\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9515608180839612\n",
      "Minor allele frequency: 0.4971719457013575\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.5054229934924078\n",
      "Site index: 11, ref allele frequency: 0.4945770065075922\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.4945770065075922\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.5048701298701299\n",
      "Site index: 12, ref allele frequency: 0.49512987012987014\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49512987012987014\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.49784250269687164\n",
      "Site index: 13, ref allele frequency: 0.5021574973031284\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49784250269687164\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.5010810810810811\n",
      "Site index: 14, ref allele frequency: 0.49891891891891893\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49891891891891893\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.4967462039045553\n",
      "Site index: 15, ref allele frequency: 0.5032537960954447\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.4967462039045553\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.4989212513484358\n",
      "Site index: 16, ref allele frequency: 0.5010787486515642\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4989212513484358\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.5097402597402597\n",
      "Site index: 17, ref allele frequency: 0.4902597402597403\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.4902597402597403\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.4902597402597403\n",
      "Site index: 18, ref allele frequency: 0.5097402597402597\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.4902597402597403\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.4918918918918919\n",
      "Site index: 19, ref allele frequency: 0.5081081081081081\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4918918918918919\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.49352051835853133\n",
      "Site index: 20, ref allele frequency: 0.5064794816414687\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.49352051835853133\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 21, non ref allele frequency: 0.4973002159827214\n",
      "Site index: 21, ref allele frequency: 0.5026997840172787\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4973002159827214\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 22, non ref allele frequency: 0.49676724137931033\n",
      "Site index: 22, ref allele frequency: 0.5032327586206896\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9989235737351991\n",
      "Minor allele frequency: 0.49676724137931033\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 23, non ref allele frequency: 0.49783783783783786\n",
      "Site index: 23, ref allele frequency: 0.5021621621621621\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49783783783783786\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 24, non ref allele frequency: 0.4967637540453074\n",
      "Site index: 24, ref allele frequency: 0.5032362459546925\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4967637540453074\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 25, non ref allele frequency: 0.5075593952483801\n",
      "Site index: 25, ref allele frequency: 0.4924406047516199\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4924406047516199\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 26, non ref allele frequency: 0.5037919826652221\n",
      "Site index: 26, ref allele frequency: 0.4962080173347779\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.4962080173347779\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 27, non ref allele frequency: 0.4935135135135135\n",
      "Site index: 27, ref allele frequency: 0.5064864864864865\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4935135135135135\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 28, non ref allele frequency: 0.4947429906542056\n",
      "Site index: 28, ref allele frequency: 0.5052570093457944\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9214208826695371\n",
      "Minor allele frequency: 0.4947429906542056\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 29, non ref allele frequency: 0.5005399568034558\n",
      "Site index: 29, ref allele frequency: 0.4994600431965443\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4994600431965443\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 30, non ref allele frequency: 0.5075757575757576\n",
      "Site index: 30, ref allele frequency: 0.49242424242424243\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49242424242424243\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 31, non ref allele frequency: 0.505939524838013\n",
      "Site index: 31, ref allele frequency: 0.49406047516198703\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.49406047516198703\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 32, non ref allele frequency: 0.4964788732394366\n",
      "Site index: 32, ref allele frequency: 0.5035211267605634\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9171151776103337\n",
      "Minor allele frequency: 0.4964788732394366\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 33, non ref allele frequency: 0.5\n",
      "Site index: 33, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 34, non ref allele frequency: 0.49569429494079653\n",
      "Site index: 34, ref allele frequency: 0.5043057050592035\n",
      "Check guardrails\n",
      "Precentage of valid sites: 1.0\n",
      "Minor allele frequency: 0.49569429494079653\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 35, non ref allele frequency: 0.5005405405405405\n",
      "Site index: 35, ref allele frequency: 0.49945945945945946\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49945945945945946\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 36, non ref allele frequency: 0.5021621621621621\n",
      "Site index: 36, ref allele frequency: 0.49783783783783786\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49783783783783786\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 37, non ref allele frequency: 0.5033632286995515\n",
      "Site index: 37, ref allele frequency: 0.4966367713004484\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9601722282023681\n",
      "Minor allele frequency: 0.4966367713004484\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 38, non ref allele frequency: 0.5094866071428571\n",
      "Site index: 38, ref allele frequency: 0.49051339285714285\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9644779332615716\n",
      "Minor allele frequency: 0.49051339285714285\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 39, non ref allele frequency: 0.5037960954446855\n",
      "Site index: 39, ref allele frequency: 0.49620390455531455\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.49620390455531455\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 40, non ref allele frequency: 0.5043243243243243\n",
      "Site index: 40, ref allele frequency: 0.49567567567567566\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49567567567567566\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 41, non ref allele frequency: 0.5037756202804746\n",
      "Site index: 41, ref allele frequency: 0.4962243797195254\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4962243797195254\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 42, non ref allele frequency: 0.49352051835853133\n",
      "Site index: 42, ref allele frequency: 0.5064794816414687\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.49352051835853133\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 43, non ref allele frequency: 0.5005405405405405\n",
      "Site index: 43, ref allele frequency: 0.49945945945945946\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49945945945945946\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 44, non ref allele frequency: 0.5043383947939263\n",
      "Site index: 44, ref allele frequency: 0.4956616052060738\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.4956616052060738\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 45, non ref allele frequency: 0.5053995680345572\n",
      "Site index: 45, ref allele frequency: 0.4946004319654428\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4946004319654428\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 46, non ref allele frequency: 0.4956803455723542\n",
      "Site index: 46, ref allele frequency: 0.5043196544276458\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4956803455723542\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 47, non ref allele frequency: 0.5021598272138229\n",
      "Site index: 47, ref allele frequency: 0.4978401727861771\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4978401727861771\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 48, non ref allele frequency: 0.4967637540453074\n",
      "Site index: 48, ref allele frequency: 0.5032362459546925\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4967637540453074\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 49, non ref allele frequency: 0.5005813953488372\n",
      "Site index: 49, ref allele frequency: 0.4994186046511628\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9257265877287406\n",
      "Minor allele frequency: 0.4994186046511628\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 50, non ref allele frequency: 0.4967637540453074\n",
      "Site index: 50, ref allele frequency: 0.5032362459546925\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4967637540453074\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 51, non ref allele frequency: 0.5048543689320388\n",
      "Site index: 51, ref allele frequency: 0.49514563106796117\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49514563106796117\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 52, non ref allele frequency: 0.49783549783549785\n",
      "Site index: 52, ref allele frequency: 0.5021645021645021\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49783549783549785\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 53, non ref allele frequency: 0.4967637540453074\n",
      "Site index: 53, ref allele frequency: 0.5032362459546925\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4967637540453074\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 54, non ref allele frequency: 0.5021621621621621\n",
      "Site index: 54, ref allele frequency: 0.49783783783783786\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49783783783783786\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 55, non ref allele frequency: 0.5\n",
      "Site index: 55, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9601722282023681\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 56, non ref allele frequency: 0.490280777537797\n",
      "Site index: 56, ref allele frequency: 0.509719222462203\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.490280777537797\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 57, non ref allele frequency: 0.4956803455723542\n",
      "Site index: 57, ref allele frequency: 0.5043196544276458\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4956803455723542\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 58, non ref allele frequency: 0.4946004319654428\n",
      "Site index: 58, ref allele frequency: 0.5053995680345572\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4946004319654428\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 59, non ref allele frequency: 0.5075757575757576\n",
      "Site index: 59, ref allele frequency: 0.49242424242424243\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49242424242424243\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 60, non ref allele frequency: 0.4967497291440953\n",
      "Site index: 60, ref allele frequency: 0.5032502708559047\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.4967497291440953\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 61, non ref allele frequency: 0.5063437139561707\n",
      "Site index: 61, ref allele frequency: 0.4936562860438293\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9332615715823466\n",
      "Minor allele frequency: 0.4936562860438293\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 62, non ref allele frequency: 0.49191810344827586\n",
      "Site index: 62, ref allele frequency: 0.5080818965517241\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9989235737351991\n",
      "Minor allele frequency: 0.49191810344827586\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 63, non ref allele frequency: 0.5064724919093851\n",
      "Site index: 63, ref allele frequency: 0.4935275080906149\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4935275080906149\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 64, non ref allele frequency: 0.5\n",
      "Site index: 64, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 1.0\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 65, non ref allele frequency: 0.4951403887688985\n",
      "Site index: 65, ref allele frequency: 0.5048596112311015\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4951403887688985\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 66, non ref allele frequency: 0.5043243243243243\n",
      "Site index: 66, ref allele frequency: 0.49567567567567566\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49567567567567566\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 67, non ref allele frequency: 0.5070270270270271\n",
      "Site index: 67, ref allele frequency: 0.492972972972973\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.492972972972973\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 68, non ref allele frequency: 0.5087051142546246\n",
      "Site index: 68, ref allele frequency: 0.4912948857453754\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9892357373519914\n",
      "Minor allele frequency: 0.4912948857453754\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 69, non ref allele frequency: 0.49187432286023836\n",
      "Site index: 69, ref allele frequency: 0.5081256771397616\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.49187432286023836\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 70, non ref allele frequency: 0.49838187702265374\n",
      "Site index: 70, ref allele frequency: 0.5016181229773463\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49838187702265374\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 71, non ref allele frequency: 0.49943757030371205\n",
      "Site index: 71, ref allele frequency: 0.500562429696288\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9569429494079655\n",
      "Minor allele frequency: 0.49943757030371205\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 72, non ref allele frequency: 0.49945945945945946\n",
      "Site index: 72, ref allele frequency: 0.5005405405405405\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49945945945945946\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 73, non ref allele frequency: 0.500544069640914\n",
      "Site index: 73, ref allele frequency: 0.499455930359086\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9892357373519914\n",
      "Minor allele frequency: 0.499455930359086\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 74, non ref allele frequency: 0.49891186071817195\n",
      "Site index: 74, ref allele frequency: 0.501088139281828\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9892357373519914\n",
      "Minor allele frequency: 0.49891186071817195\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 75, non ref allele frequency: 0.494028230184582\n",
      "Site index: 75, ref allele frequency: 0.505971769815418\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9913885898815931\n",
      "Minor allele frequency: 0.494028230184582\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 76, non ref allele frequency: 0.5048543689320388\n",
      "Site index: 76, ref allele frequency: 0.49514563106796117\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49514563106796117\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 77, non ref allele frequency: 0.509719222462203\n",
      "Site index: 77, ref allele frequency: 0.490280777537797\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.490280777537797\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 78, non ref allele frequency: 0.49837486457204766\n",
      "Site index: 78, ref allele frequency: 0.5016251354279523\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.49837486457204766\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 79, non ref allele frequency: 0.49406047516198703\n",
      "Site index: 79, ref allele frequency: 0.505939524838013\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.49406047516198703\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 80, non ref allele frequency: 0.49083063646170444\n",
      "Site index: 80, ref allele frequency: 0.5091693635382956\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49083063646170444\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 81, non ref allele frequency: 0.4956521739130435\n",
      "Site index: 81, ref allele frequency: 0.5043478260869565\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9903121636167922\n",
      "Minor allele frequency: 0.4956521739130435\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 82, non ref allele frequency: 0.5082056892778993\n",
      "Site index: 82, ref allele frequency: 0.49179431072210067\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9838536060279871\n",
      "Minor allele frequency: 0.49179431072210067\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 83, non ref allele frequency: 0.503767491926803\n",
      "Site index: 83, ref allele frequency: 0.49623250807319697\n",
      "Check guardrails\n",
      "Precentage of valid sites: 1.0\n",
      "Minor allele frequency: 0.49623250807319697\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 84, non ref allele frequency: 0.5080993520518359\n",
      "Site index: 84, ref allele frequency: 0.4919006479481641\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4919006479481641\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 85, non ref allele frequency: 0.5037878787878788\n",
      "Site index: 85, ref allele frequency: 0.4962121212121212\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.4962121212121212\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 86, non ref allele frequency: 0.5070194384449244\n",
      "Site index: 86, ref allele frequency: 0.4929805615550756\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4929805615550756\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 87, non ref allele frequency: 0.49129930394431554\n",
      "Site index: 87, ref allele frequency: 0.5087006960556845\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9278794402583423\n",
      "Minor allele frequency: 0.49129930394431554\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 88, non ref allele frequency: 0.5070422535211268\n",
      "Site index: 88, ref allele frequency: 0.49295774647887325\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.49295774647887325\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 89, non ref allele frequency: 0.5037878787878788\n",
      "Site index: 89, ref allele frequency: 0.4962121212121212\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.4962121212121212\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 90, non ref allele frequency: 0.49079089924160346\n",
      "Site index: 90, ref allele frequency: 0.5092091007583965\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.49079089924160346\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 91, non ref allele frequency: 0.5075512405609492\n",
      "Site index: 91, ref allele frequency: 0.4924487594390507\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.4924487594390507\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 92, non ref allele frequency: 0.5086206896551724\n",
      "Site index: 92, ref allele frequency: 0.49137931034482757\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9989235737351991\n",
      "Minor allele frequency: 0.49137931034482757\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 93, non ref allele frequency: 0.5075839653304443\n",
      "Site index: 93, ref allele frequency: 0.4924160346695558\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9935414424111948\n",
      "Minor allele frequency: 0.4924160346695558\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 94, non ref allele frequency: 0.499457111834962\n",
      "Site index: 94, ref allele frequency: 0.500542888165038\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9913885898815931\n",
      "Minor allele frequency: 0.499457111834962\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 95, non ref allele frequency: 0.5016233766233766\n",
      "Site index: 95, ref allele frequency: 0.49837662337662336\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.49837662337662336\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 96, non ref allele frequency: 0.49457111834962\n",
      "Site index: 96, ref allele frequency: 0.50542888165038\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9913885898815931\n",
      "Minor allele frequency: 0.49457111834962\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 97, non ref allele frequency: 0.5048543689320388\n",
      "Site index: 97, ref allele frequency: 0.49514563106796117\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49514563106796117\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 98, non ref allele frequency: 0.5010810810810811\n",
      "Site index: 98, ref allele frequency: 0.49891891891891893\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.49891891891891893\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 99, non ref allele frequency: 0.4908008658008658\n",
      "Site index: 99, ref allele frequency: 0.5091991341991342\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9946178686759957\n",
      "Minor allele frequency: 0.4908008658008658\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n"
     ]
    }
   ],
   "source": [
    "# PARAMS\n",
    "# if we have less than this which are valid (not -1), site is not included in calc.\n",
    "min_valid_sites_precentage = 0.1\n",
    "\n",
    "# if not in range - will raise a warning\n",
    "min_minor_freq_expected = 0.49\n",
    "max_minor_freq_expected = 0.5\n",
    "min_minor_count_expected = -1\n",
    "max_minor_count_expected = -1\n",
    "\n",
    "mac_maf = 'maf'\n",
    "class_name = '0.49'\n",
    "window_index = 0\n",
    "classes_folder = r\"C:\\Data\\HUJI\\hgdp\\classes/\"\n",
    "\n",
    "# FUNCTION\n",
    "OUTPUT_PATTERN_DIST_FILE = 'count_dist_window_{window_index}.tsv'\n",
    "output_dir = f'{classes_folder}windows/{mac_maf}_{class_name}/'\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "output_count_dist_file = output_dir + OUTPUT_PATTERN_DIST_FILE.format(window_index=window_index)\n",
    "       \n",
    "window_df = get_window(classes_folder, mac_maf, class_name, window_index)\n",
    "\n",
    "# 100 indexes takes ~5 minutes \n",
    "# using percision of 5 decimals we generates a file of ~5MB\n",
    "# we have 322,483 windows of 100\n",
    "# will take over 3 years to process on one machine\n",
    "# output is about 12 TB\n",
    "window_pairwise_counts, window_pairwise_dist = window_calc_pairwise_distances_with_guardrails(window_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T13:03:35.777681Z",
     "start_time": "2021-01-27T13:03:20.494758Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Site index: 0, non ref allele frequency: 0.4924406047516199\n",
      "Site index: 0, ref allele frequency: 0.5075593952483801\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9967707212055974\n",
      "Minor allele frequency: 0.4924406047516199\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.4924078091106291\n",
      "Site index: 1, ref allele frequency: 0.5075921908893709\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9924650161463939\n",
      "Minor allele frequency: 0.4924078091106291\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.5037837837837837\n",
      "Site index: 2, ref allele frequency: 0.4962162162162162\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9956942949407965\n",
      "Minor allele frequency: 0.4962162162162162\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.49838187702265374\n",
      "Site index: 3, ref allele frequency: 0.5016181229773463\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9978471474703983\n",
      "Minor allele frequency: 0.49838187702265374\n",
      "Passed guardrails\n",
      "Done with individual 0/929\n",
      "Done with individual 100/929\n",
      "Done with individual 200/929\n",
      "Done with individual 300/929\n",
      "Done with individual 400/929\n",
      "Done with individual 500/929\n",
      "Done with individual 600/929\n",
      "Done with individual 700/929\n",
      "Done with individual 800/929\n",
      "Done with individual 900/929\n"
     ]
    }
   ],
   "source": [
    "dummy_df = window_df[window_df.columns[:4]]\n",
    "window_pairwise_counts, window_pairwise_dist = window_calc_pairwise_distances_with_guardrails(dummy_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, min_minor_count_expected, max_minor_count_expected)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T11:00:21.573613Z",
     "start_time": "2021-01-27T11:00:19.309418Z"
    }
   },
   "outputs": [],
   "source": [
    "write_pairwise_distances(output_count_dist_file, window_pairwise_counts, window_pairwise_dist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-06T08:32:26.719837Z",
     "start_time": "2021-01-06T08:32:26.698822Z"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "def build_per_slice_distances_matrix(\n",
    "    path_to_012_files, \n",
    "    base_output_path, \n",
    "    desired_num_of_sites_per_output,\n",
    "    min_minor_freq_expected, \n",
    "    max_minor_freq_expected, \n",
    "    min_valid_sites_precentage):\n",
    "    # outputs:\n",
    "    # for each group i in 1,...,k:\n",
    "    # 1. count_dist : space seperated pairwise distances and count of valid sites.\n",
    "    #    This is the sum over the valid sites used, and the format in each entry is <sum>,<l'>.\n",
    "    # dummy example for 3 individuals, over l=5, where individual number 3 has 2 invalid entries (-1 -1 1 0 2)\n",
    "    # 0.9,5 0.7,3\n",
    "    # 0.8,3 \n",
    "    # the ',3' indicates that there are only 3 valied sited for individuals 1 and 3 and 2 and 3\n",
    "    # 2. {i}_sites_ids : <site_id> <number of non -1 entris>\n",
    "\n",
    "\n",
    "    # Validate inputs\n",
    "    path_012 = f'{path_to_012_files}{SUFFIX_012}'\n",
    "    path_pos = f'{path_to_012_files}{SUFFIX_POS}'\n",
    "    chr_name = _get_chr_name(path_pos)\n",
    "\n",
    "    _set_logger(chr_name, min_minor_freq_expected)\n",
    "\n",
    "    _validate_file_exist(path_012)\n",
    "    _validate_file_exist(path_pos)\n",
    "\n",
    "    # prepare output folder\n",
    "    _log(f'chr_name is {chr_name}')\n",
    "    output_dir = _prepare_output_folder(base_output_path, chr_name)\n",
    "\n",
    "    # load the data\n",
    "    # get sites positions to use as column names\n",
    "    with open(path_pos, 'r') as f:\n",
    "        sites_pos = [l.replace('\\t','_').replace('\\n','') for l in f.readlines()]\n",
    "    all_df = pd.read_csv(path_012, sep='\\t', names = ['ind_id'] + sites_pos)\n",
    "    # drop first column - individual ids\n",
    "    all_df = all_df.drop(columns=['ind_id'])\n",
    "    num_sites = len(all_df.columns)\n",
    "    num_of_slices = int(float(num_sites)/desired_num_of_sites_per_output)\n",
    "    size_of_last_slice = num_sites % desired_num_of_sites_per_output\n",
    "    _log(f'Total num of sites: {num_sites}')\n",
    "    _log(f'Desired size of slice: {desired_num_of_sites_per_output}')\n",
    "    _log(f'Number of slices: {num_of_slices+1}')\n",
    "    _log(f'Size of last slice: {size_of_last_slice}')\n",
    "\n",
    "    for slice_index in range(num_of_slices+1):\n",
    "        _log(f'{chr_name}: slice {slice_index}/{num_of_slices}')\n",
    "\n",
    "        min_col_index = slice_index*desired_num_of_sites_per_output\n",
    "        # in case we have less than desired_num_of_sites_per_output, pandas will only take those we have, nice!\n",
    "        max_col_index = min_col_index + desired_num_of_sites_per_output\n",
    "        slice_df = all_df.iloc[:,min_col_index:max_col_index]\n",
    "        num_of_sites = len(slice_df.columns)\n",
    "        _log(f'\\n#############\\n')\n",
    "        _log(f'Slice index: {slice_index}. Slice size: {num_of_sites} ')\n",
    "        output_count_dist_file = output_dir + OUTPUT_PATTERN_DIST_FILE.format(slice_index=slice_index)\n",
    "        output_sites_file = output_dir + OUTPUT_PATTERN_SITES_FILE.format(slice_index=slice_index)\n",
    "        # todo - maybe write to this file also the number of -1, 0, 1 and 2?\n",
    "        _write_output_sites_file(output_sites_file, slice_df)\n",
    "        _log(f'Output files: {output_count_dist_file} {output_sites_file}')\n",
    "        window_pairwise_counts, window_pairwise_dist = _slice_calc_pairwise_distances_with_guardrails(slice_index, num_of_slices, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected)\n",
    "        _write_pairwise_distances(output_count_dist_file, window_pairwise_counts, window_pairwise_dist)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-27T10:59:00.150440Z",
     "start_time": "2021-01-27T10:59:00.134817Z"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Call func"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-06T08:38:33.346080Z",
     "start_time": "2021-01-06T08:33:49.787888Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr_name is chr9\n",
      "output_dir is C:\\Data\\HUJI\\hgdp\\mock/chr9/\n",
      "Total num of sites: 178\n",
      "Desired size of slice: 100\n",
      "Number of slices: 2\n",
      "Size of last slice: 78\n",
      "chr9: slice 0/1\n",
      "\n",
      "#############\n",
      "\n",
      "Slice index: 0. Slice size: 100 \n",
      "Output files: C:\\Data\\HUJI\\hgdp\\mock/chr9/count_dist_slice_0.tsv C:\\Data\\HUJI\\hgdp\\mock/chr9/sites_ids_slice_0.tsv\n",
      "Site index: 0, non ref allele frequency: 0.5\n",
      "Site index: 0, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4951560818083961\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 0. Done with individual 0/929\n",
      "Slice: 0/1. Site: 0. Done with individual 100/929\n",
      "Slice: 0/1. Site: 0. Done with individual 200/929\n",
      "Slice: 0/1. Site: 0. Done with individual 300/929\n",
      "Slice: 0/1. Site: 0. Done with individual 400/929\n",
      "Slice: 0/1. Site: 0. Done with individual 500/929\n",
      "Slice: 0/1. Site: 0. Done with individual 600/929\n",
      "Slice: 0/1. Site: 0. Done with individual 700/929\n",
      "Slice: 0/1. Site: 0. Done with individual 800/929\n",
      "Slice: 0/1. Site: 0. Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.5\n",
      "Site index: 1, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4908503767491927\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 1. Done with individual 0/929\n",
      "Slice: 0/1. Site: 1. Done with individual 100/929\n",
      "Slice: 0/1. Site: 1. Done with individual 200/929\n",
      "Slice: 0/1. Site: 1. Done with individual 300/929\n",
      "Slice: 0/1. Site: 1. Done with individual 400/929\n",
      "Slice: 0/1. Site: 1. Done with individual 500/929\n",
      "Slice: 0/1. Site: 1. Done with individual 600/929\n",
      "Slice: 0/1. Site: 1. Done with individual 700/929\n",
      "Slice: 0/1. Site: 1. Done with individual 800/929\n",
      "Slice: 0/1. Site: 1. Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.49746192893401014\n",
      "Site index: 2, ref allele frequency: 0.5025380710659898\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4241119483315393\n",
      "Minor allele frequency: 0.49746192893401014\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 2. Done with individual 0/929\n",
      "Slice: 0/1. Site: 2. Done with individual 100/929\n",
      "Slice: 0/1. Site: 2. Done with individual 200/929\n",
      "Slice: 0/1. Site: 2. Done with individual 300/929\n",
      "Slice: 0/1. Site: 2. Done with individual 400/929\n",
      "Slice: 0/1. Site: 2. Done with individual 500/929\n",
      "Slice: 0/1. Site: 2. Done with individual 600/929\n",
      "Slice: 0/1. Site: 2. Done with individual 700/929\n",
      "Slice: 0/1. Site: 2. Done with individual 800/929\n",
      "Slice: 0/1. Site: 2. Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.5\n",
      "Site index: 3, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.30893433799784714\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 3. Done with individual 0/929\n",
      "Slice: 0/1. Site: 3. Done with individual 100/929\n",
      "Slice: 0/1. Site: 3. Done with individual 200/929\n",
      "Slice: 0/1. Site: 3. Done with individual 300/929\n",
      "Slice: 0/1. Site: 3. Done with individual 400/929\n",
      "Slice: 0/1. Site: 3. Done with individual 500/929\n",
      "Slice: 0/1. Site: 3. Done with individual 600/929\n",
      "Slice: 0/1. Site: 3. Done with individual 700/929\n",
      "Slice: 0/1. Site: 3. Done with individual 800/929\n",
      "Slice: 0/1. Site: 3. Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.4969040247678019\n",
      "Site index: 4, ref allele frequency: 0.5030959752321982\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34768568353067814\n",
      "Minor allele frequency: 0.4969040247678019\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 4. Done with individual 0/929\n",
      "Slice: 0/1. Site: 4. Done with individual 100/929\n",
      "Slice: 0/1. Site: 4. Done with individual 200/929\n",
      "Slice: 0/1. Site: 4. Done with individual 300/929\n",
      "Slice: 0/1. Site: 4. Done with individual 400/929\n",
      "Slice: 0/1. Site: 4. Done with individual 500/929\n",
      "Slice: 0/1. Site: 4. Done with individual 600/929\n",
      "Slice: 0/1. Site: 4. Done with individual 700/929\n",
      "Slice: 0/1. Site: 4. Done with individual 800/929\n",
      "Slice: 0/1. Site: 4. Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.49534883720930234\n",
      "Site index: 5, ref allele frequency: 0.5046511627906977\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.23143164693218515\n",
      "Minor allele frequency: 0.49534883720930234\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 5. Done with individual 0/929\n",
      "Slice: 0/1. Site: 5. Done with individual 100/929\n",
      "Slice: 0/1. Site: 5. Done with individual 200/929\n",
      "Slice: 0/1. Site: 5. Done with individual 300/929\n",
      "Slice: 0/1. Site: 5. Done with individual 400/929\n",
      "Slice: 0/1. Site: 5. Done with individual 500/929\n",
      "Slice: 0/1. Site: 5. Done with individual 600/929\n",
      "Slice: 0/1. Site: 5. Done with individual 700/929\n",
      "Slice: 0/1. Site: 5. Done with individual 800/929\n",
      "Slice: 0/1. Site: 5. Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.4935897435897436\n",
      "Site index: 6, ref allele frequency: 0.5064102564102564\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.16792249730893433\n",
      "Minor allele frequency: 0.4935897435897436\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 6. Done with individual 0/929\n",
      "Slice: 0/1. Site: 6. Done with individual 100/929\n",
      "Slice: 0/1. Site: 6. Done with individual 200/929\n",
      "Slice: 0/1. Site: 6. Done with individual 300/929\n",
      "Slice: 0/1. Site: 6. Done with individual 400/929\n",
      "Slice: 0/1. Site: 6. Done with individual 500/929\n",
      "Slice: 0/1. Site: 6. Done with individual 600/929\n",
      "Slice: 0/1. Site: 6. Done with individual 700/929\n",
      "Slice: 0/1. Site: 6. Done with individual 800/929\n",
      "Slice: 0/1. Site: 6. Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.4966442953020134\n",
      "Site index: 7, ref allele frequency: 0.5033557046979866\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.48116254036598494\n",
      "Minor allele frequency: 0.4966442953020134\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 7. Done with individual 0/929\n",
      "Slice: 0/1. Site: 7. Done with individual 100/929\n",
      "Slice: 0/1. Site: 7. Done with individual 200/929\n",
      "Slice: 0/1. Site: 7. Done with individual 300/929\n",
      "Slice: 0/1. Site: 7. Done with individual 400/929\n",
      "Slice: 0/1. Site: 7. Done with individual 500/929\n",
      "Slice: 0/1. Site: 7. Done with individual 600/929\n",
      "Slice: 0/1. Site: 7. Done with individual 700/929\n",
      "Slice: 0/1. Site: 7. Done with individual 800/929\n",
      "Slice: 0/1. Site: 7. Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.49230769230769234\n",
      "Site index: 8, ref allele frequency: 0.5076923076923077\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34983853606027987\n",
      "Minor allele frequency: 0.49230769230769234\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 8. Done with individual 0/929\n",
      "Slice: 0/1. Site: 8. Done with individual 100/929\n",
      "Slice: 0/1. Site: 8. Done with individual 200/929\n",
      "Slice: 0/1. Site: 8. Done with individual 300/929\n",
      "Slice: 0/1. Site: 8. Done with individual 400/929\n",
      "Slice: 0/1. Site: 8. Done with individual 500/929\n",
      "Slice: 0/1. Site: 8. Done with individual 600/929\n",
      "Slice: 0/1. Site: 8. Done with individual 700/929\n",
      "Slice: 0/1. Site: 8. Done with individual 800/929\n",
      "Slice: 0/1. Site: 8. Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.494475138121547\n",
      "Site index: 9, ref allele frequency: 0.505524861878453\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.19483315392895587\n",
      "Minor allele frequency: 0.494475138121547\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 9. Done with individual 0/929\n",
      "Slice: 0/1. Site: 9. Done with individual 100/929\n",
      "Slice: 0/1. Site: 9. Done with individual 200/929\n",
      "Slice: 0/1. Site: 9. Done with individual 300/929\n",
      "Slice: 0/1. Site: 9. Done with individual 400/929\n",
      "Slice: 0/1. Site: 9. Done with individual 500/929\n",
      "Slice: 0/1. Site: 9. Done with individual 600/929\n",
      "Slice: 0/1. Site: 9. Done with individual 700/929\n",
      "Slice: 0/1. Site: 9. Done with individual 800/929\n",
      "Slice: 0/1. Site: 9. Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.4948630136986301\n",
      "Site index: 10, ref allele frequency: 0.5051369863013698\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.628632938643703\n",
      "Minor allele frequency: 0.4948630136986301\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 10. Done with individual 0/929\n",
      "Slice: 0/1. Site: 10. Done with individual 100/929\n",
      "Slice: 0/1. Site: 10. Done with individual 200/929\n",
      "Slice: 0/1. Site: 10. Done with individual 300/929\n",
      "Slice: 0/1. Site: 10. Done with individual 400/929\n",
      "Slice: 0/1. Site: 10. Done with individual 500/929\n",
      "Slice: 0/1. Site: 10. Done with individual 600/929\n",
      "Slice: 0/1. Site: 10. Done with individual 700/929\n",
      "Slice: 0/1. Site: 10. Done with individual 800/929\n",
      "Slice: 0/1. Site: 10. Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.4942434210526316\n",
      "Site index: 11, ref allele frequency: 0.5057565789473685\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6544671689989235\n",
      "Minor allele frequency: 0.4942434210526316\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 11. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 11. Done with individual 100/929\n",
      "Slice: 0/1. Site: 11. Done with individual 200/929\n",
      "Slice: 0/1. Site: 11. Done with individual 300/929\n",
      "Slice: 0/1. Site: 11. Done with individual 400/929\n",
      "Slice: 0/1. Site: 11. Done with individual 500/929\n",
      "Slice: 0/1. Site: 11. Done with individual 600/929\n",
      "Slice: 0/1. Site: 11. Done with individual 700/929\n",
      "Slice: 0/1. Site: 11. Done with individual 800/929\n",
      "Slice: 0/1. Site: 11. Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.4919786096256685\n",
      "Site index: 12, ref allele frequency: 0.5080213903743316\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.20129171151776104\n",
      "Minor allele frequency: 0.4919786096256685\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 12. Done with individual 0/929\n",
      "Slice: 0/1. Site: 12. Done with individual 100/929\n",
      "Slice: 0/1. Site: 12. Done with individual 200/929\n",
      "Slice: 0/1. Site: 12. Done with individual 300/929\n",
      "Slice: 0/1. Site: 12. Done with individual 400/929\n",
      "Slice: 0/1. Site: 12. Done with individual 500/929\n",
      "Slice: 0/1. Site: 12. Done with individual 600/929\n",
      "Slice: 0/1. Site: 12. Done with individual 700/929\n",
      "Slice: 0/1. Site: 12. Done with individual 800/929\n",
      "Slice: 0/1. Site: 12. Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.4974025974025974\n",
      "Site index: 13, ref allele frequency: 0.5025974025974026\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8288482238966631\n",
      "Minor allele frequency: 0.4974025974025974\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 13. Done with individual 0/929\n",
      "Slice: 0/1. Site: 13. Done with individual 100/929\n",
      "Slice: 0/1. Site: 13. Done with individual 200/929\n",
      "Slice: 0/1. Site: 13. Done with individual 300/929\n",
      "Slice: 0/1. Site: 13. Done with individual 400/929\n",
      "Slice: 0/1. Site: 13. Done with individual 500/929\n",
      "Slice: 0/1. Site: 13. Done with individual 600/929\n",
      "Slice: 0/1. Site: 13. Done with individual 700/929\n",
      "Slice: 0/1. Site: 13. Done with individual 800/929\n",
      "Slice: 0/1. Site: 13. Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.5\n",
      "Site index: 14, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.48869752421959095\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 14. Done with individual 0/929\n",
      "Slice: 0/1. Site: 14. Done with individual 100/929\n",
      "Slice: 0/1. Site: 14. Done with individual 200/929\n",
      "Slice: 0/1. Site: 14. Done with individual 300/929\n",
      "Slice: 0/1. Site: 14. Done with individual 400/929\n",
      "Slice: 0/1. Site: 14. Done with individual 500/929\n",
      "Slice: 0/1. Site: 14. Done with individual 600/929\n",
      "Slice: 0/1. Site: 14. Done with individual 700/929\n",
      "Slice: 0/1. Site: 14. Done with individual 800/929\n",
      "Slice: 0/1. Site: 14. Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.5\n",
      "Site index: 15, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6275565123789021\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 15. Done with individual 0/929\n",
      "Slice: 0/1. Site: 15. Done with individual 100/929\n",
      "Slice: 0/1. Site: 15. Done with individual 200/929\n",
      "Slice: 0/1. Site: 15. Done with individual 300/929\n",
      "Slice: 0/1. Site: 15. Done with individual 400/929\n",
      "Slice: 0/1. Site: 15. Done with individual 500/929\n",
      "Slice: 0/1. Site: 15. Done with individual 600/929\n",
      "Slice: 0/1. Site: 15. Done with individual 700/929\n",
      "Slice: 0/1. Site: 15. Done with individual 800/929\n",
      "Slice: 0/1. Site: 15. Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.5\n",
      "Site index: 16, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5608180839612487\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 16. Done with individual 0/929\n",
      "Slice: 0/1. Site: 16. Done with individual 100/929\n",
      "Slice: 0/1. Site: 16. Done with individual 200/929\n",
      "Slice: 0/1. Site: 16. Done with individual 300/929\n",
      "Slice: 0/1. Site: 16. Done with individual 400/929\n",
      "Slice: 0/1. Site: 16. Done with individual 500/929\n",
      "Slice: 0/1. Site: 16. Done with individual 600/929\n",
      "Slice: 0/1. Site: 16. Done with individual 700/929\n",
      "Slice: 0/1. Site: 16. Done with individual 800/929\n",
      "Slice: 0/1. Site: 16. Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.49280575539568344\n",
      "Site index: 17, ref allele frequency: 0.5071942446043165\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.1496232508073197\n",
      "Minor allele frequency: 0.49280575539568344\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 17. Done with individual 0/929\n",
      "Slice: 0/1. Site: 17. Done with individual 100/929\n",
      "Slice: 0/1. Site: 17. Done with individual 200/929\n",
      "Slice: 0/1. Site: 17. Done with individual 300/929\n",
      "Slice: 0/1. Site: 17. Done with individual 400/929\n",
      "Slice: 0/1. Site: 17. Done with individual 500/929\n",
      "Slice: 0/1. Site: 17. Done with individual 600/929\n",
      "Slice: 0/1. Site: 17. Done with individual 700/929\n",
      "Slice: 0/1. Site: 17. Done with individual 800/929\n",
      "Slice: 0/1. Site: 17. Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.5\n",
      "Site index: 18, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5059203444564048\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 18. Done with individual 0/929\n",
      "Slice: 0/1. Site: 18. Done with individual 100/929\n",
      "Slice: 0/1. Site: 18. Done with individual 200/929\n",
      "Slice: 0/1. Site: 18. Done with individual 300/929\n",
      "Slice: 0/1. Site: 18. Done with individual 400/929\n",
      "Slice: 0/1. Site: 18. Done with individual 500/929\n",
      "Slice: 0/1. Site: 18. Done with individual 600/929\n",
      "Slice: 0/1. Site: 18. Done with individual 700/929\n",
      "Slice: 0/1. Site: 18. Done with individual 800/929\n",
      "Slice: 0/1. Site: 18. Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.5\n",
      "Site index: 19, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.37782561894510225\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 19. Done with individual 0/929\n",
      "Slice: 0/1. Site: 19. Done with individual 100/929\n",
      "Slice: 0/1. Site: 19. Done with individual 200/929\n",
      "Slice: 0/1. Site: 19. Done with individual 300/929\n",
      "Slice: 0/1. Site: 19. Done with individual 400/929\n",
      "Slice: 0/1. Site: 19. Done with individual 500/929\n",
      "Slice: 0/1. Site: 19. Done with individual 600/929\n",
      "Slice: 0/1. Site: 19. Done with individual 700/929\n",
      "Slice: 0/1. Site: 19. Done with individual 800/929\n",
      "Slice: 0/1. Site: 19. Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.5\n",
      "Site index: 20, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.47255113024757806\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 20. Done with individual 0/929\n",
      "Slice: 0/1. Site: 20. Done with individual 100/929\n",
      "Slice: 0/1. Site: 20. Done with individual 200/929\n",
      "Slice: 0/1. Site: 20. Done with individual 300/929\n",
      "Slice: 0/1. Site: 20. Done with individual 400/929\n",
      "Slice: 0/1. Site: 20. Done with individual 500/929\n",
      "Slice: 0/1. Site: 20. Done with individual 600/929\n",
      "Slice: 0/1. Site: 20. Done with individual 700/929\n",
      "Slice: 0/1. Site: 20. Done with individual 800/929\n",
      "Slice: 0/1. Site: 20. Done with individual 900/929\n",
      "Site index: 21, non ref allele frequency: 0.5\n",
      "Site index: 21, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7265877287405813\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 21. Done with individual 0/929\n",
      "Slice: 0/1. Site: 21. Done with individual 100/929\n",
      "Slice: 0/1. Site: 21. Done with individual 200/929\n",
      "Slice: 0/1. Site: 21. Done with individual 300/929\n",
      "Slice: 0/1. Site: 21. Done with individual 400/929\n",
      "Slice: 0/1. Site: 21. Done with individual 500/929\n",
      "Slice: 0/1. Site: 21. Done with individual 600/929\n",
      "Slice: 0/1. Site: 21. Done with individual 700/929\n",
      "Slice: 0/1. Site: 21. Done with individual 800/929\n",
      "Slice: 0/1. Site: 21. Done with individual 900/929\n",
      "Site index: 22, non ref allele frequency: 0.5\n",
      "Site index: 22, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.25834230355220666\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 22. Done with individual 0/929\n",
      "Slice: 0/1. Site: 22. Done with individual 100/929\n",
      "Slice: 0/1. Site: 22. Done with individual 200/929\n",
      "Slice: 0/1. Site: 22. Done with individual 300/929\n",
      "Slice: 0/1. Site: 22. Done with individual 400/929\n",
      "Slice: 0/1. Site: 22. Done with individual 500/929\n",
      "Slice: 0/1. Site: 22. Done with individual 600/929\n",
      "Slice: 0/1. Site: 22. Done with individual 700/929\n",
      "Slice: 0/1. Site: 22. Done with individual 800/929\n",
      "Slice: 0/1. Site: 22. Done with individual 900/929\n",
      "Site index: 23, non ref allele frequency: 0.49099099099099097\n",
      "Site index: 23, ref allele frequency: 0.509009009009009\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.11948331539289558\n",
      "Minor allele frequency: 0.49099099099099097\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 23. Done with individual 0/929\n",
      "Slice: 0/1. Site: 23. Done with individual 100/929\n",
      "Slice: 0/1. Site: 23. Done with individual 200/929\n",
      "Slice: 0/1. Site: 23. Done with individual 300/929\n",
      "Slice: 0/1. Site: 23. Done with individual 400/929\n",
      "Slice: 0/1. Site: 23. Done with individual 500/929\n",
      "Slice: 0/1. Site: 23. Done with individual 600/929\n",
      "Slice: 0/1. Site: 23. Done with individual 700/929\n",
      "Slice: 0/1. Site: 23. Done with individual 800/929\n",
      "Slice: 0/1. Site: 23. Done with individual 900/929\n",
      "Site index: 24, non ref allele frequency: 0.4976190476190476\n",
      "Site index: 24, ref allele frequency: 0.5023809523809524\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22604951560818085\n",
      "Minor allele frequency: 0.4976190476190476\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 24. Done with individual 0/929\n",
      "Slice: 0/1. Site: 24. Done with individual 100/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 24. Done with individual 200/929\n",
      "Slice: 0/1. Site: 24. Done with individual 300/929\n",
      "Slice: 0/1. Site: 24. Done with individual 400/929\n",
      "Slice: 0/1. Site: 24. Done with individual 500/929\n",
      "Slice: 0/1. Site: 24. Done with individual 600/929\n",
      "Slice: 0/1. Site: 24. Done with individual 700/929\n",
      "Slice: 0/1. Site: 24. Done with individual 800/929\n",
      "Slice: 0/1. Site: 24. Done with individual 900/929\n",
      "Site index: 25, non ref allele frequency: 0.5\n",
      "Site index: 25, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8460710441334769\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 25. Done with individual 0/929\n",
      "Slice: 0/1. Site: 25. Done with individual 100/929\n",
      "Slice: 0/1. Site: 25. Done with individual 200/929\n",
      "Slice: 0/1. Site: 25. Done with individual 300/929\n",
      "Slice: 0/1. Site: 25. Done with individual 400/929\n",
      "Slice: 0/1. Site: 25. Done with individual 500/929\n",
      "Slice: 0/1. Site: 25. Done with individual 600/929\n",
      "Slice: 0/1. Site: 25. Done with individual 700/929\n",
      "Slice: 0/1. Site: 25. Done with individual 800/929\n",
      "Slice: 0/1. Site: 25. Done with individual 900/929\n",
      "Site index: 26, non ref allele frequency: 0.5\n",
      "Site index: 26, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8385360602798708\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 26. Done with individual 0/929\n",
      "Slice: 0/1. Site: 26. Done with individual 100/929\n",
      "Slice: 0/1. Site: 26. Done with individual 200/929\n",
      "Slice: 0/1. Site: 26. Done with individual 300/929\n",
      "Slice: 0/1. Site: 26. Done with individual 400/929\n",
      "Slice: 0/1. Site: 26. Done with individual 500/929\n",
      "Slice: 0/1. Site: 26. Done with individual 600/929\n",
      "Slice: 0/1. Site: 26. Done with individual 700/929\n",
      "Slice: 0/1. Site: 26. Done with individual 800/929\n",
      "Slice: 0/1. Site: 26. Done with individual 900/929\n",
      "Site index: 27, non ref allele frequency: 0.49473684210526314\n",
      "Site index: 27, ref allele frequency: 0.5052631578947369\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.40904198062432723\n",
      "Minor allele frequency: 0.49473684210526314\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 27. Done with individual 0/929\n",
      "Slice: 0/1. Site: 27. Done with individual 100/929\n",
      "Slice: 0/1. Site: 27. Done with individual 200/929\n",
      "Slice: 0/1. Site: 27. Done with individual 300/929\n",
      "Slice: 0/1. Site: 27. Done with individual 400/929\n",
      "Slice: 0/1. Site: 27. Done with individual 500/929\n",
      "Slice: 0/1. Site: 27. Done with individual 600/929\n",
      "Slice: 0/1. Site: 27. Done with individual 700/929\n",
      "Slice: 0/1. Site: 27. Done with individual 800/929\n",
      "Slice: 0/1. Site: 27. Done with individual 900/929\n",
      "Site index: 28, non ref allele frequency: 0.5\n",
      "Site index: 28, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5317545748116254\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 28. Done with individual 0/929\n",
      "Slice: 0/1. Site: 28. Done with individual 100/929\n",
      "Slice: 0/1. Site: 28. Done with individual 200/929\n",
      "Slice: 0/1. Site: 28. Done with individual 300/929\n",
      "Slice: 0/1. Site: 28. Done with individual 400/929\n",
      "Slice: 0/1. Site: 28. Done with individual 500/929\n",
      "Slice: 0/1. Site: 28. Done with individual 600/929\n",
      "Slice: 0/1. Site: 28. Done with individual 700/929\n",
      "Slice: 0/1. Site: 28. Done with individual 800/929\n",
      "Slice: 0/1. Site: 28. Done with individual 900/929\n",
      "Site index: 29, non ref allele frequency: 0.5\n",
      "Site index: 29, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.24865446716899892\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 29. Done with individual 0/929\n",
      "Slice: 0/1. Site: 29. Done with individual 100/929\n",
      "Slice: 0/1. Site: 29. Done with individual 200/929\n",
      "Slice: 0/1. Site: 29. Done with individual 300/929\n",
      "Slice: 0/1. Site: 29. Done with individual 400/929\n",
      "Slice: 0/1. Site: 29. Done with individual 500/929\n",
      "Slice: 0/1. Site: 29. Done with individual 600/929\n",
      "Slice: 0/1. Site: 29. Done with individual 700/929\n",
      "Slice: 0/1. Site: 29. Done with individual 800/929\n",
      "Slice: 0/1. Site: 29. Done with individual 900/929\n",
      "Site index: 30, non ref allele frequency: 0.5\n",
      "Site index: 30, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.11410118406889128\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 30. Done with individual 0/929\n",
      "Slice: 0/1. Site: 30. Done with individual 100/929\n",
      "Slice: 0/1. Site: 30. Done with individual 200/929\n",
      "Slice: 0/1. Site: 30. Done with individual 300/929\n",
      "Slice: 0/1. Site: 30. Done with individual 400/929\n",
      "Slice: 0/1. Site: 30. Done with individual 500/929\n",
      "Slice: 0/1. Site: 30. Done with individual 600/929\n",
      "Slice: 0/1. Site: 30. Done with individual 700/929\n",
      "Slice: 0/1. Site: 30. Done with individual 800/929\n",
      "Slice: 0/1. Site: 30. Done with individual 900/929\n",
      "Site index: 31, non ref allele frequency: 0.49330357142857145\n",
      "Site index: 31, ref allele frequency: 0.5066964285714286\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4822389666307858\n",
      "Minor allele frequency: 0.49330357142857145\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 31. Done with individual 0/929\n",
      "Slice: 0/1. Site: 31. Done with individual 100/929\n",
      "Slice: 0/1. Site: 31. Done with individual 200/929\n",
      "Slice: 0/1. Site: 31. Done with individual 300/929\n",
      "Slice: 0/1. Site: 31. Done with individual 400/929\n",
      "Slice: 0/1. Site: 31. Done with individual 500/929\n",
      "Slice: 0/1. Site: 31. Done with individual 600/929\n",
      "Slice: 0/1. Site: 31. Done with individual 700/929\n",
      "Slice: 0/1. Site: 31. Done with individual 800/929\n",
      "Slice: 0/1. Site: 31. Done with individual 900/929\n",
      "Site index: 32, non ref allele frequency: 0.4931506849315068\n",
      "Site index: 32, ref allele frequency: 0.5068493150684932\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.55005382131324\n",
      "Minor allele frequency: 0.4931506849315068\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 32. Done with individual 0/929\n",
      "Slice: 0/1. Site: 32. Done with individual 100/929\n",
      "Slice: 0/1. Site: 32. Done with individual 200/929\n",
      "Slice: 0/1. Site: 32. Done with individual 300/929\n",
      "Slice: 0/1. Site: 32. Done with individual 400/929\n",
      "Slice: 0/1. Site: 32. Done with individual 500/929\n",
      "Slice: 0/1. Site: 32. Done with individual 600/929\n",
      "Slice: 0/1. Site: 32. Done with individual 700/929\n",
      "Slice: 0/1. Site: 32. Done with individual 800/929\n",
      "Slice: 0/1. Site: 32. Done with individual 900/929\n",
      "Site index: 33, non ref allele frequency: 0.49580712788259956\n",
      "Site index: 33, ref allele frequency: 0.5041928721174004\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5134553283100107\n",
      "Minor allele frequency: 0.49580712788259956\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 33. Done with individual 0/929\n",
      "Slice: 0/1. Site: 33. Done with individual 100/929\n",
      "Slice: 0/1. Site: 33. Done with individual 200/929\n",
      "Slice: 0/1. Site: 33. Done with individual 300/929\n",
      "Slice: 0/1. Site: 33. Done with individual 400/929\n",
      "Slice: 0/1. Site: 33. Done with individual 500/929\n",
      "Slice: 0/1. Site: 33. Done with individual 600/929\n",
      "Slice: 0/1. Site: 33. Done with individual 700/929\n",
      "Slice: 0/1. Site: 33. Done with individual 800/929\n",
      "Slice: 0/1. Site: 33. Done with individual 900/929\n",
      "Site index: 34, non ref allele frequency: 0.506578947368421\n",
      "Site index: 34, ref allele frequency: 0.4934210526315789\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.40904198062432723\n",
      "Minor allele frequency: 0.4934210526315789\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 34. Done with individual 0/929\n",
      "Slice: 0/1. Site: 34. Done with individual 100/929\n",
      "Slice: 0/1. Site: 34. Done with individual 200/929\n",
      "Slice: 0/1. Site: 34. Done with individual 300/929\n",
      "Slice: 0/1. Site: 34. Done with individual 400/929\n",
      "Slice: 0/1. Site: 34. Done with individual 500/929\n",
      "Slice: 0/1. Site: 34. Done with individual 600/929\n",
      "Slice: 0/1. Site: 34. Done with individual 700/929\n",
      "Slice: 0/1. Site: 34. Done with individual 800/929\n",
      "Slice: 0/1. Site: 34. Done with individual 900/929\n",
      "Site index: 35, non ref allele frequency: 0.49012158054711247\n",
      "Site index: 35, ref allele frequency: 0.5098784194528876\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7082884822389667\n",
      "Minor allele frequency: 0.49012158054711247\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 35. Done with individual 0/929\n",
      "Slice: 0/1. Site: 35. Done with individual 100/929\n",
      "Slice: 0/1. Site: 35. Done with individual 200/929\n",
      "Slice: 0/1. Site: 35. Done with individual 300/929\n",
      "Slice: 0/1. Site: 35. Done with individual 400/929\n",
      "Slice: 0/1. Site: 35. Done with individual 500/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 35. Done with individual 600/929\n",
      "Slice: 0/1. Site: 35. Done with individual 700/929\n",
      "Slice: 0/1. Site: 35. Done with individual 800/929\n",
      "Slice: 0/1. Site: 35. Done with individual 900/929\n",
      "Site index: 36, non ref allele frequency: 0.5015948963317385\n",
      "Site index: 36, ref allele frequency: 0.4984051036682616\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6749192680301399\n",
      "Minor allele frequency: 0.4984051036682616\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 36. Done with individual 0/929\n",
      "Slice: 0/1. Site: 36. Done with individual 100/929\n",
      "Slice: 0/1. Site: 36. Done with individual 200/929\n",
      "Slice: 0/1. Site: 36. Done with individual 300/929\n",
      "Slice: 0/1. Site: 36. Done with individual 400/929\n",
      "Slice: 0/1. Site: 36. Done with individual 500/929\n",
      "Slice: 0/1. Site: 36. Done with individual 600/929\n",
      "Slice: 0/1. Site: 36. Done with individual 700/929\n",
      "Slice: 0/1. Site: 36. Done with individual 800/929\n",
      "Slice: 0/1. Site: 36. Done with individual 900/929\n",
      "Site index: 37, non ref allele frequency: 0.5\n",
      "Site index: 37, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7319698600645855\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 37. Done with individual 0/929\n",
      "Slice: 0/1. Site: 37. Done with individual 100/929\n",
      "Slice: 0/1. Site: 37. Done with individual 200/929\n",
      "Slice: 0/1. Site: 37. Done with individual 300/929\n",
      "Slice: 0/1. Site: 37. Done with individual 400/929\n",
      "Slice: 0/1. Site: 37. Done with individual 500/929\n",
      "Slice: 0/1. Site: 37. Done with individual 600/929\n",
      "Slice: 0/1. Site: 37. Done with individual 700/929\n",
      "Slice: 0/1. Site: 37. Done with individual 800/929\n",
      "Slice: 0/1. Site: 37. Done with individual 900/929\n",
      "Site index: 38, non ref allele frequency: 0.5\n",
      "Site index: 38, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3627556512378902\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 38. Done with individual 0/929\n",
      "Slice: 0/1. Site: 38. Done with individual 100/929\n",
      "Slice: 0/1. Site: 38. Done with individual 200/929\n",
      "Slice: 0/1. Site: 38. Done with individual 300/929\n",
      "Slice: 0/1. Site: 38. Done with individual 400/929\n",
      "Slice: 0/1. Site: 38. Done with individual 500/929\n",
      "Slice: 0/1. Site: 38. Done with individual 600/929\n",
      "Slice: 0/1. Site: 38. Done with individual 700/929\n",
      "Slice: 0/1. Site: 38. Done with individual 800/929\n",
      "Slice: 0/1. Site: 38. Done with individual 900/929\n",
      "Site index: 39, non ref allele frequency: 0.4951923076923077\n",
      "Site index: 39, ref allele frequency: 0.5048076923076923\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2238966630785791\n",
      "Minor allele frequency: 0.4951923076923077\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 39. Done with individual 0/929\n",
      "Slice: 0/1. Site: 39. Done with individual 100/929\n",
      "Slice: 0/1. Site: 39. Done with individual 200/929\n",
      "Slice: 0/1. Site: 39. Done with individual 300/929\n",
      "Slice: 0/1. Site: 39. Done with individual 400/929\n",
      "Slice: 0/1. Site: 39. Done with individual 500/929\n",
      "Slice: 0/1. Site: 39. Done with individual 600/929\n",
      "Slice: 0/1. Site: 39. Done with individual 700/929\n",
      "Slice: 0/1. Site: 39. Done with individual 800/929\n",
      "Slice: 0/1. Site: 39. Done with individual 900/929\n",
      "Site index: 40, non ref allele frequency: 0.49707602339181284\n",
      "Site index: 40, ref allele frequency: 0.5029239766081871\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.18406889128094725\n",
      "Minor allele frequency: 0.49707602339181284\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 40. Done with individual 0/929\n",
      "Slice: 0/1. Site: 40. Done with individual 100/929\n",
      "Slice: 0/1. Site: 40. Done with individual 200/929\n",
      "Slice: 0/1. Site: 40. Done with individual 300/929\n",
      "Slice: 0/1. Site: 40. Done with individual 400/929\n",
      "Slice: 0/1. Site: 40. Done with individual 500/929\n",
      "Slice: 0/1. Site: 40. Done with individual 600/929\n",
      "Slice: 0/1. Site: 40. Done with individual 700/929\n",
      "Slice: 0/1. Site: 40. Done with individual 800/929\n",
      "Slice: 0/1. Site: 40. Done with individual 900/929\n",
      "Site index: 41, non ref allele frequency: 0.4909090909090909\n",
      "Site index: 41, ref allele frequency: 0.509090909090909\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7104413347685683\n",
      "Minor allele frequency: 0.4909090909090909\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 41. Done with individual 0/929\n",
      "Slice: 0/1. Site: 41. Done with individual 100/929\n",
      "Slice: 0/1. Site: 41. Done with individual 200/929\n",
      "Slice: 0/1. Site: 41. Done with individual 300/929\n",
      "Slice: 0/1. Site: 41. Done with individual 400/929\n",
      "Slice: 0/1. Site: 41. Done with individual 500/929\n",
      "Slice: 0/1. Site: 41. Done with individual 600/929\n",
      "Slice: 0/1. Site: 41. Done with individual 700/929\n",
      "Slice: 0/1. Site: 41. Done with individual 800/929\n",
      "Slice: 0/1. Site: 41. Done with individual 900/929\n",
      "Site index: 42, non ref allele frequency: 0.4985549132947977\n",
      "Site index: 42, ref allele frequency: 0.5014450867052023\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7448869752421959\n",
      "Minor allele frequency: 0.4985549132947977\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 42. Done with individual 0/929\n",
      "Slice: 0/1. Site: 42. Done with individual 100/929\n",
      "Slice: 0/1. Site: 42. Done with individual 200/929\n",
      "Slice: 0/1. Site: 42. Done with individual 300/929\n",
      "Slice: 0/1. Site: 42. Done with individual 400/929\n",
      "Slice: 0/1. Site: 42. Done with individual 500/929\n",
      "Slice: 0/1. Site: 42. Done with individual 600/929\n",
      "Slice: 0/1. Site: 42. Done with individual 700/929\n",
      "Slice: 0/1. Site: 42. Done with individual 800/929\n",
      "Slice: 0/1. Site: 42. Done with individual 900/929\n",
      "Site index: 43, non ref allele frequency: 0.5094339622641509\n",
      "Site index: 43, ref allele frequency: 0.49056603773584906\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5705059203444564\n",
      "Minor allele frequency: 0.49056603773584906\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 43. Done with individual 0/929\n",
      "Slice: 0/1. Site: 43. Done with individual 100/929\n",
      "Slice: 0/1. Site: 43. Done with individual 200/929\n",
      "Slice: 0/1. Site: 43. Done with individual 300/929\n",
      "Slice: 0/1. Site: 43. Done with individual 400/929\n",
      "Slice: 0/1. Site: 43. Done with individual 500/929\n",
      "Slice: 0/1. Site: 43. Done with individual 600/929\n",
      "Slice: 0/1. Site: 43. Done with individual 700/929\n",
      "Slice: 0/1. Site: 43. Done with individual 800/929\n",
      "Slice: 0/1. Site: 43. Done with individual 900/929\n",
      "Site index: 44, non ref allele frequency: 0.49887133182844245\n",
      "Site index: 44, ref allele frequency: 0.5011286681715575\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.47685683530678147\n",
      "Minor allele frequency: 0.49887133182844245\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 44. Done with individual 0/929\n",
      "Slice: 0/1. Site: 44. Done with individual 100/929\n",
      "Slice: 0/1. Site: 44. Done with individual 200/929\n",
      "Slice: 0/1. Site: 44. Done with individual 300/929\n",
      "Slice: 0/1. Site: 44. Done with individual 400/929\n",
      "Slice: 0/1. Site: 44. Done with individual 500/929\n",
      "Slice: 0/1. Site: 44. Done with individual 600/929\n",
      "Slice: 0/1. Site: 44. Done with individual 700/929\n",
      "Slice: 0/1. Site: 44. Done with individual 800/929\n",
      "Slice: 0/1. Site: 44. Done with individual 900/929\n",
      "Site index: 45, non ref allele frequency: 0.4919678714859438\n",
      "Site index: 45, ref allele frequency: 0.5080321285140562\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2680301399354144\n",
      "Minor allele frequency: 0.4919678714859438\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 45. Done with individual 0/929\n",
      "Slice: 0/1. Site: 45. Done with individual 100/929\n",
      "Slice: 0/1. Site: 45. Done with individual 200/929\n",
      "Slice: 0/1. Site: 45. Done with individual 300/929\n",
      "Slice: 0/1. Site: 45. Done with individual 400/929\n",
      "Slice: 0/1. Site: 45. Done with individual 500/929\n",
      "Slice: 0/1. Site: 45. Done with individual 600/929\n",
      "Slice: 0/1. Site: 45. Done with individual 700/929\n",
      "Slice: 0/1. Site: 45. Done with individual 800/929\n",
      "Slice: 0/1. Site: 45. Done with individual 900/929\n",
      "Site index: 46, non ref allele frequency: 0.4929742388758782\n",
      "Site index: 46, ref allele frequency: 0.5070257611241218\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4596340150699677\n",
      "Minor allele frequency: 0.4929742388758782\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 46. Done with individual 0/929\n",
      "Slice: 0/1. Site: 46. Done with individual 100/929\n",
      "Slice: 0/1. Site: 46. Done with individual 200/929\n",
      "Slice: 0/1. Site: 46. Done with individual 300/929\n",
      "Slice: 0/1. Site: 46. Done with individual 400/929\n",
      "Slice: 0/1. Site: 46. Done with individual 500/929\n",
      "Slice: 0/1. Site: 46. Done with individual 600/929\n",
      "Slice: 0/1. Site: 46. Done with individual 700/929\n",
      "Slice: 0/1. Site: 46. Done with individual 800/929\n",
      "Slice: 0/1. Site: 46. Done with individual 900/929\n",
      "Site index: 47, non ref allele frequency: 0.4991103202846975\n",
      "Site index: 47, ref allele frequency: 0.5008896797153025\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.604951560818084\n",
      "Minor allele frequency: 0.4991103202846975\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 47. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 47. Done with individual 100/929\n",
      "Slice: 0/1. Site: 47. Done with individual 200/929\n",
      "Slice: 0/1. Site: 47. Done with individual 300/929\n",
      "Slice: 0/1. Site: 47. Done with individual 400/929\n",
      "Slice: 0/1. Site: 47. Done with individual 500/929\n",
      "Slice: 0/1. Site: 47. Done with individual 600/929\n",
      "Slice: 0/1. Site: 47. Done with individual 700/929\n",
      "Slice: 0/1. Site: 47. Done with individual 800/929\n",
      "Slice: 0/1. Site: 47. Done with individual 900/929\n",
      "Site index: 48, non ref allele frequency: 0.5\n",
      "Site index: 48, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.620021528525296\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 48. Done with individual 0/929\n",
      "Slice: 0/1. Site: 48. Done with individual 100/929\n",
      "Slice: 0/1. Site: 48. Done with individual 200/929\n",
      "Slice: 0/1. Site: 48. Done with individual 300/929\n",
      "Slice: 0/1. Site: 48. Done with individual 400/929\n",
      "Slice: 0/1. Site: 48. Done with individual 500/929\n",
      "Slice: 0/1. Site: 48. Done with individual 600/929\n",
      "Slice: 0/1. Site: 48. Done with individual 700/929\n",
      "Slice: 0/1. Site: 48. Done with individual 800/929\n",
      "Slice: 0/1. Site: 48. Done with individual 900/929\n",
      "Site index: 49, non ref allele frequency: 0.5\n",
      "Site index: 49, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9364908503767492\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 49. Done with individual 0/929\n",
      "Slice: 0/1. Site: 49. Done with individual 100/929\n",
      "Slice: 0/1. Site: 49. Done with individual 200/929\n",
      "Slice: 0/1. Site: 49. Done with individual 300/929\n",
      "Slice: 0/1. Site: 49. Done with individual 400/929\n",
      "Slice: 0/1. Site: 49. Done with individual 500/929\n",
      "Slice: 0/1. Site: 49. Done with individual 600/929\n",
      "Slice: 0/1. Site: 49. Done with individual 700/929\n",
      "Slice: 0/1. Site: 49. Done with individual 800/929\n",
      "Slice: 0/1. Site: 49. Done with individual 900/929\n",
      "Site index: 50, non ref allele frequency: 0.5\n",
      "Site index: 50, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9117330462863293\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 50. Done with individual 0/929\n",
      "Slice: 0/1. Site: 50. Done with individual 100/929\n",
      "Slice: 0/1. Site: 50. Done with individual 200/929\n",
      "Slice: 0/1. Site: 50. Done with individual 300/929\n",
      "Slice: 0/1. Site: 50. Done with individual 400/929\n",
      "Slice: 0/1. Site: 50. Done with individual 500/929\n",
      "Slice: 0/1. Site: 50. Done with individual 600/929\n",
      "Slice: 0/1. Site: 50. Done with individual 700/929\n",
      "Slice: 0/1. Site: 50. Done with individual 800/929\n",
      "Slice: 0/1. Site: 50. Done with individual 900/929\n",
      "Site index: 51, non ref allele frequency: 0.5\n",
      "Site index: 51, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8708288482238966\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 51. Done with individual 0/929\n",
      "Slice: 0/1. Site: 51. Done with individual 100/929\n",
      "Slice: 0/1. Site: 51. Done with individual 200/929\n",
      "Slice: 0/1. Site: 51. Done with individual 300/929\n",
      "Slice: 0/1. Site: 51. Done with individual 400/929\n",
      "Slice: 0/1. Site: 51. Done with individual 500/929\n",
      "Slice: 0/1. Site: 51. Done with individual 600/929\n",
      "Slice: 0/1. Site: 51. Done with individual 700/929\n",
      "Slice: 0/1. Site: 51. Done with individual 800/929\n",
      "Slice: 0/1. Site: 51. Done with individual 900/929\n",
      "Site index: 52, non ref allele frequency: 0.5\n",
      "Site index: 52, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8783638320775027\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 52. Done with individual 0/929\n",
      "Slice: 0/1. Site: 52. Done with individual 100/929\n",
      "Slice: 0/1. Site: 52. Done with individual 200/929\n",
      "Slice: 0/1. Site: 52. Done with individual 300/929\n",
      "Slice: 0/1. Site: 52. Done with individual 400/929\n",
      "Slice: 0/1. Site: 52. Done with individual 500/929\n",
      "Slice: 0/1. Site: 52. Done with individual 600/929\n",
      "Slice: 0/1. Site: 52. Done with individual 700/929\n",
      "Slice: 0/1. Site: 52. Done with individual 800/929\n",
      "Slice: 0/1. Site: 52. Done with individual 900/929\n",
      "Site index: 53, non ref allele frequency: 0.49823321554770317\n",
      "Site index: 53, ref allele frequency: 0.5017667844522968\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6092572658772875\n",
      "Minor allele frequency: 0.49823321554770317\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 53. Done with individual 0/929\n",
      "Slice: 0/1. Site: 53. Done with individual 100/929\n",
      "Slice: 0/1. Site: 53. Done with individual 200/929\n",
      "Slice: 0/1. Site: 53. Done with individual 300/929\n",
      "Slice: 0/1. Site: 53. Done with individual 400/929\n",
      "Slice: 0/1. Site: 53. Done with individual 500/929\n",
      "Slice: 0/1. Site: 53. Done with individual 600/929\n",
      "Slice: 0/1. Site: 53. Done with individual 700/929\n",
      "Slice: 0/1. Site: 53. Done with individual 800/929\n",
      "Slice: 0/1. Site: 53. Done with individual 900/929\n",
      "Site index: 54, non ref allele frequency: 0.5\n",
      "Site index: 54, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.13885898815931108\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 54. Done with individual 0/929\n",
      "Slice: 0/1. Site: 54. Done with individual 100/929\n",
      "Slice: 0/1. Site: 54. Done with individual 200/929\n",
      "Slice: 0/1. Site: 54. Done with individual 300/929\n",
      "Slice: 0/1. Site: 54. Done with individual 400/929\n",
      "Slice: 0/1. Site: 54. Done with individual 500/929\n",
      "Slice: 0/1. Site: 54. Done with individual 600/929\n",
      "Slice: 0/1. Site: 54. Done with individual 700/929\n",
      "Slice: 0/1. Site: 54. Done with individual 800/929\n",
      "Slice: 0/1. Site: 54. Done with individual 900/929\n",
      "Site index: 55, non ref allele frequency: 0.4964622641509434\n",
      "Site index: 55, ref allele frequency: 0.5035377358490566\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4564047362755651\n",
      "Minor allele frequency: 0.4964622641509434\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 55. Done with individual 0/929\n",
      "Slice: 0/1. Site: 55. Done with individual 100/929\n",
      "Slice: 0/1. Site: 55. Done with individual 200/929\n",
      "Slice: 0/1. Site: 55. Done with individual 300/929\n",
      "Slice: 0/1. Site: 55. Done with individual 400/929\n",
      "Slice: 0/1. Site: 55. Done with individual 500/929\n",
      "Slice: 0/1. Site: 55. Done with individual 600/929\n",
      "Slice: 0/1. Site: 55. Done with individual 700/929\n",
      "Slice: 0/1. Site: 55. Done with individual 800/929\n",
      "Slice: 0/1. Site: 55. Done with individual 900/929\n",
      "Site index: 56, non ref allele frequency: 0.5\n",
      "Site index: 56, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6921420882669537\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 56. Done with individual 0/929\n",
      "Slice: 0/1. Site: 56. Done with individual 100/929\n",
      "Slice: 0/1. Site: 56. Done with individual 200/929\n",
      "Slice: 0/1. Site: 56. Done with individual 300/929\n",
      "Slice: 0/1. Site: 56. Done with individual 400/929\n",
      "Slice: 0/1. Site: 56. Done with individual 500/929\n",
      "Slice: 0/1. Site: 56. Done with individual 600/929\n",
      "Slice: 0/1. Site: 56. Done with individual 700/929\n",
      "Slice: 0/1. Site: 56. Done with individual 800/929\n",
      "Slice: 0/1. Site: 56. Done with individual 900/929\n",
      "Site index: 57, non ref allele frequency: 0.5\n",
      "Site index: 57, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5640473627556513\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 57. Done with individual 0/929\n",
      "Slice: 0/1. Site: 57. Done with individual 100/929\n",
      "Slice: 0/1. Site: 57. Done with individual 200/929\n",
      "Slice: 0/1. Site: 57. Done with individual 300/929\n",
      "Slice: 0/1. Site: 57. Done with individual 400/929\n",
      "Slice: 0/1. Site: 57. Done with individual 500/929\n",
      "Slice: 0/1. Site: 57. Done with individual 600/929\n",
      "Slice: 0/1. Site: 57. Done with individual 700/929\n",
      "Slice: 0/1. Site: 57. Done with individual 800/929\n",
      "Slice: 0/1. Site: 57. Done with individual 900/929\n",
      "Site index: 58, non ref allele frequency: 0.502127659574468\n",
      "Site index: 58, ref allele frequency: 0.4978723404255319\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5059203444564048\n",
      "Minor allele frequency: 0.4978723404255319\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 58. Done with individual 0/929\n",
      "Slice: 0/1. Site: 58. Done with individual 100/929\n",
      "Slice: 0/1. Site: 58. Done with individual 200/929\n",
      "Slice: 0/1. Site: 58. Done with individual 300/929\n",
      "Slice: 0/1. Site: 58. Done with individual 400/929\n",
      "Slice: 0/1. Site: 58. Done with individual 500/929\n",
      "Slice: 0/1. Site: 58. Done with individual 600/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 58. Done with individual 700/929\n",
      "Slice: 0/1. Site: 58. Done with individual 800/929\n",
      "Slice: 0/1. Site: 58. Done with individual 900/929\n",
      "Site index: 59, non ref allele frequency: 0.49029126213592233\n",
      "Site index: 59, ref allele frequency: 0.5097087378640777\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.1108719052744887\n",
      "Minor allele frequency: 0.49029126213592233\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 59. Done with individual 0/929\n",
      "Slice: 0/1. Site: 59. Done with individual 100/929\n",
      "Slice: 0/1. Site: 59. Done with individual 200/929\n",
      "Slice: 0/1. Site: 59. Done with individual 300/929\n",
      "Slice: 0/1. Site: 59. Done with individual 400/929\n",
      "Slice: 0/1. Site: 59. Done with individual 500/929\n",
      "Slice: 0/1. Site: 59. Done with individual 600/929\n",
      "Slice: 0/1. Site: 59. Done with individual 700/929\n",
      "Slice: 0/1. Site: 59. Done with individual 800/929\n",
      "Slice: 0/1. Site: 59. Done with individual 900/929\n",
      "Site index: 60, non ref allele frequency: 0.49917627677100496\n",
      "Site index: 60, ref allele frequency: 0.500823723228995\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6533907427341227\n",
      "Minor allele frequency: 0.49917627677100496\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 60. Done with individual 0/929\n",
      "Slice: 0/1. Site: 60. Done with individual 100/929\n",
      "Slice: 0/1. Site: 60. Done with individual 200/929\n",
      "Slice: 0/1. Site: 60. Done with individual 300/929\n",
      "Slice: 0/1. Site: 60. Done with individual 400/929\n",
      "Slice: 0/1. Site: 60. Done with individual 500/929\n",
      "Slice: 0/1. Site: 60. Done with individual 600/929\n",
      "Slice: 0/1. Site: 60. Done with individual 700/929\n",
      "Slice: 0/1. Site: 60. Done with individual 800/929\n",
      "Slice: 0/1. Site: 60. Done with individual 900/929\n",
      "Site index: 61, non ref allele frequency: 0.502851711026616\n",
      "Site index: 61, ref allele frequency: 0.49714828897338403\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.566200215285253\n",
      "Minor allele frequency: 0.49714828897338403\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 61. Done with individual 0/929\n",
      "Slice: 0/1. Site: 61. Done with individual 100/929\n",
      "Slice: 0/1. Site: 61. Done with individual 200/929\n",
      "Slice: 0/1. Site: 61. Done with individual 300/929\n",
      "Slice: 0/1. Site: 61. Done with individual 400/929\n",
      "Slice: 0/1. Site: 61. Done with individual 500/929\n",
      "Slice: 0/1. Site: 61. Done with individual 600/929\n",
      "Slice: 0/1. Site: 61. Done with individual 700/929\n",
      "Slice: 0/1. Site: 61. Done with individual 800/929\n",
      "Slice: 0/1. Site: 61. Done with individual 900/929\n",
      "Site index: 62, non ref allele frequency: 0.4957627118644068\n",
      "Site index: 62, ref allele frequency: 0.5042372881355932\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.38105489773950485\n",
      "Minor allele frequency: 0.4957627118644068\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 62. Done with individual 0/929\n",
      "Slice: 0/1. Site: 62. Done with individual 100/929\n",
      "Slice: 0/1. Site: 62. Done with individual 200/929\n",
      "Slice: 0/1. Site: 62. Done with individual 300/929\n",
      "Slice: 0/1. Site: 62. Done with individual 400/929\n",
      "Slice: 0/1. Site: 62. Done with individual 500/929\n",
      "Slice: 0/1. Site: 62. Done with individual 600/929\n",
      "Slice: 0/1. Site: 62. Done with individual 700/929\n",
      "Slice: 0/1. Site: 62. Done with individual 800/929\n",
      "Slice: 0/1. Site: 62. Done with individual 900/929\n",
      "Site index: 63, non ref allele frequency: 0.49019607843137253\n",
      "Site index: 63, ref allele frequency: 0.5098039215686274\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.10979547900968784\n",
      "Minor allele frequency: 0.49019607843137253\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 63. Done with individual 0/929\n",
      "Slice: 0/1. Site: 63. Done with individual 100/929\n",
      "Slice: 0/1. Site: 63. Done with individual 200/929\n",
      "Slice: 0/1. Site: 63. Done with individual 300/929\n",
      "Slice: 0/1. Site: 63. Done with individual 400/929\n",
      "Slice: 0/1. Site: 63. Done with individual 500/929\n",
      "Slice: 0/1. Site: 63. Done with individual 600/929\n",
      "Slice: 0/1. Site: 63. Done with individual 700/929\n",
      "Slice: 0/1. Site: 63. Done with individual 800/929\n",
      "Slice: 0/1. Site: 63. Done with individual 900/929\n",
      "Site index: 64, non ref allele frequency: 0.4974025974025974\n",
      "Site index: 64, ref allele frequency: 0.5025974025974026\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.41442411194833156\n",
      "Minor allele frequency: 0.4974025974025974\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 64. Done with individual 0/929\n",
      "Slice: 0/1. Site: 64. Done with individual 100/929\n",
      "Slice: 0/1. Site: 64. Done with individual 200/929\n",
      "Slice: 0/1. Site: 64. Done with individual 300/929\n",
      "Slice: 0/1. Site: 64. Done with individual 400/929\n",
      "Slice: 0/1. Site: 64. Done with individual 500/929\n",
      "Slice: 0/1. Site: 64. Done with individual 600/929\n",
      "Slice: 0/1. Site: 64. Done with individual 700/929\n",
      "Slice: 0/1. Site: 64. Done with individual 800/929\n",
      "Slice: 0/1. Site: 64. Done with individual 900/929\n",
      "Site index: 65, non ref allele frequency: 0.49567099567099565\n",
      "Site index: 65, ref allele frequency: 0.5043290043290043\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.49730893433799783\n",
      "Minor allele frequency: 0.49567099567099565\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 65. Done with individual 0/929\n",
      "Slice: 0/1. Site: 65. Done with individual 100/929\n",
      "Slice: 0/1. Site: 65. Done with individual 200/929\n",
      "Slice: 0/1. Site: 65. Done with individual 300/929\n",
      "Slice: 0/1. Site: 65. Done with individual 400/929\n",
      "Slice: 0/1. Site: 65. Done with individual 500/929\n",
      "Slice: 0/1. Site: 65. Done with individual 600/929\n",
      "Slice: 0/1. Site: 65. Done with individual 700/929\n",
      "Slice: 0/1. Site: 65. Done with individual 800/929\n",
      "Slice: 0/1. Site: 65. Done with individual 900/929\n",
      "Site index: 66, non ref allele frequency: 0.5\n",
      "Site index: 66, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8073196986006459\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 66. Done with individual 0/929\n",
      "Slice: 0/1. Site: 66. Done with individual 100/929\n",
      "Slice: 0/1. Site: 66. Done with individual 200/929\n",
      "Slice: 0/1. Site: 66. Done with individual 300/929\n",
      "Slice: 0/1. Site: 66. Done with individual 400/929\n",
      "Slice: 0/1. Site: 66. Done with individual 500/929\n",
      "Slice: 0/1. Site: 66. Done with individual 600/929\n",
      "Slice: 0/1. Site: 66. Done with individual 700/929\n",
      "Slice: 0/1. Site: 66. Done with individual 800/929\n",
      "Slice: 0/1. Site: 66. Done with individual 900/929\n",
      "Site index: 67, non ref allele frequency: 0.5\n",
      "Site index: 67, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8127018299246501\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 67. Done with individual 0/929\n",
      "Slice: 0/1. Site: 67. Done with individual 100/929\n",
      "Slice: 0/1. Site: 67. Done with individual 200/929\n",
      "Slice: 0/1. Site: 67. Done with individual 300/929\n",
      "Slice: 0/1. Site: 67. Done with individual 400/929\n",
      "Slice: 0/1. Site: 67. Done with individual 500/929\n",
      "Slice: 0/1. Site: 67. Done with individual 600/929\n",
      "Slice: 0/1. Site: 67. Done with individual 700/929\n",
      "Slice: 0/1. Site: 67. Done with individual 800/929\n",
      "Slice: 0/1. Site: 67. Done with individual 900/929\n",
      "Site index: 68, non ref allele frequency: 0.5\n",
      "Site index: 68, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5339074273412271\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 68. Done with individual 0/929\n",
      "Slice: 0/1. Site: 68. Done with individual 100/929\n",
      "Slice: 0/1. Site: 68. Done with individual 200/929\n",
      "Slice: 0/1. Site: 68. Done with individual 300/929\n",
      "Slice: 0/1. Site: 68. Done with individual 400/929\n",
      "Slice: 0/1. Site: 68. Done with individual 500/929\n",
      "Slice: 0/1. Site: 68. Done with individual 600/929\n",
      "Slice: 0/1. Site: 68. Done with individual 700/929\n",
      "Slice: 0/1. Site: 68. Done with individual 800/929\n",
      "Slice: 0/1. Site: 68. Done with individual 900/929\n",
      "Site index: 69, non ref allele frequency: 0.49377593360995853\n",
      "Site index: 69, ref allele frequency: 0.5062240663900415\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.25941872981700753\n",
      "Minor allele frequency: 0.49377593360995853\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 69. Done with individual 0/929\n",
      "Slice: 0/1. Site: 69. Done with individual 100/929\n",
      "Slice: 0/1. Site: 69. Done with individual 200/929\n",
      "Slice: 0/1. Site: 69. Done with individual 300/929\n",
      "Slice: 0/1. Site: 69. Done with individual 400/929\n",
      "Slice: 0/1. Site: 69. Done with individual 500/929\n",
      "Slice: 0/1. Site: 69. Done with individual 600/929\n",
      "Slice: 0/1. Site: 69. Done with individual 700/929\n",
      "Slice: 0/1. Site: 69. Done with individual 800/929\n",
      "Slice: 0/1. Site: 69. Done with individual 900/929\n",
      "Site index: 70, non ref allele frequency: 0.5\n",
      "Site index: 70, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7072120559741658\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 70. Done with individual 0/929\n",
      "Slice: 0/1. Site: 70. Done with individual 100/929\n",
      "Slice: 0/1. Site: 70. Done with individual 200/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 70. Done with individual 300/929\n",
      "Slice: 0/1. Site: 70. Done with individual 400/929\n",
      "Slice: 0/1. Site: 70. Done with individual 500/929\n",
      "Slice: 0/1. Site: 70. Done with individual 600/929\n",
      "Slice: 0/1. Site: 70. Done with individual 700/929\n",
      "Slice: 0/1. Site: 70. Done with individual 800/929\n",
      "Slice: 0/1. Site: 70. Done with individual 900/929\n",
      "Site index: 71, non ref allele frequency: 0.5\n",
      "Site index: 71, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7190527448869752\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 71. Done with individual 0/929\n",
      "Slice: 0/1. Site: 71. Done with individual 100/929\n",
      "Slice: 0/1. Site: 71. Done with individual 200/929\n",
      "Slice: 0/1. Site: 71. Done with individual 300/929\n",
      "Slice: 0/1. Site: 71. Done with individual 400/929\n",
      "Slice: 0/1. Site: 71. Done with individual 500/929\n",
      "Slice: 0/1. Site: 71. Done with individual 600/929\n",
      "Slice: 0/1. Site: 71. Done with individual 700/929\n",
      "Slice: 0/1. Site: 71. Done with individual 800/929\n",
      "Slice: 0/1. Site: 71. Done with individual 900/929\n",
      "Site index: 72, non ref allele frequency: 0.5\n",
      "Site index: 72, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4251883745963401\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 72. Done with individual 0/929\n",
      "Slice: 0/1. Site: 72. Done with individual 100/929\n",
      "Slice: 0/1. Site: 72. Done with individual 200/929\n",
      "Slice: 0/1. Site: 72. Done with individual 300/929\n",
      "Slice: 0/1. Site: 72. Done with individual 400/929\n",
      "Slice: 0/1. Site: 72. Done with individual 500/929\n",
      "Slice: 0/1. Site: 72. Done with individual 600/929\n",
      "Slice: 0/1. Site: 72. Done with individual 700/929\n",
      "Slice: 0/1. Site: 72. Done with individual 800/929\n",
      "Slice: 0/1. Site: 72. Done with individual 900/929\n",
      "Site index: 73, non ref allele frequency: 0.5\n",
      "Site index: 73, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3261571582346609\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 73. Done with individual 0/929\n",
      "Slice: 0/1. Site: 73. Done with individual 100/929\n",
      "Slice: 0/1. Site: 73. Done with individual 200/929\n",
      "Slice: 0/1. Site: 73. Done with individual 300/929\n",
      "Slice: 0/1. Site: 73. Done with individual 400/929\n",
      "Slice: 0/1. Site: 73. Done with individual 500/929\n",
      "Slice: 0/1. Site: 73. Done with individual 600/929\n",
      "Slice: 0/1. Site: 73. Done with individual 700/929\n",
      "Slice: 0/1. Site: 73. Done with individual 800/929\n",
      "Slice: 0/1. Site: 73. Done with individual 900/929\n",
      "Site index: 74, non ref allele frequency: 0.49458204334365324\n",
      "Site index: 74, ref allele frequency: 0.5054179566563467\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6953713670613563\n",
      "Minor allele frequency: 0.49458204334365324\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 74. Done with individual 0/929\n",
      "Slice: 0/1. Site: 74. Done with individual 100/929\n",
      "Slice: 0/1. Site: 74. Done with individual 200/929\n",
      "Slice: 0/1. Site: 74. Done with individual 300/929\n",
      "Slice: 0/1. Site: 74. Done with individual 400/929\n",
      "Slice: 0/1. Site: 74. Done with individual 500/929\n",
      "Slice: 0/1. Site: 74. Done with individual 600/929\n",
      "Slice: 0/1. Site: 74. Done with individual 700/929\n",
      "Slice: 0/1. Site: 74. Done with individual 800/929\n",
      "Slice: 0/1. Site: 74. Done with individual 900/929\n",
      "Site index: 75, non ref allele frequency: 0.49611801242236025\n",
      "Site index: 75, ref allele frequency: 0.5038819875776398\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6932185145317545\n",
      "Minor allele frequency: 0.49611801242236025\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 75. Done with individual 0/929\n",
      "Slice: 0/1. Site: 75. Done with individual 100/929\n",
      "Slice: 0/1. Site: 75. Done with individual 200/929\n",
      "Slice: 0/1. Site: 75. Done with individual 300/929\n",
      "Slice: 0/1. Site: 75. Done with individual 400/929\n",
      "Slice: 0/1. Site: 75. Done with individual 500/929\n",
      "Slice: 0/1. Site: 75. Done with individual 600/929\n",
      "Slice: 0/1. Site: 75. Done with individual 700/929\n",
      "Slice: 0/1. Site: 75. Done with individual 800/929\n",
      "Slice: 0/1. Site: 75. Done with individual 900/929\n",
      "Site index: 76, non ref allele frequency: 0.5024232633279483\n",
      "Site index: 76, ref allele frequency: 0.4975767366720517\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6663078579117331\n",
      "Minor allele frequency: 0.4975767366720517\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 76. Done with individual 0/929\n",
      "Slice: 0/1. Site: 76. Done with individual 100/929\n",
      "Slice: 0/1. Site: 76. Done with individual 200/929\n",
      "Slice: 0/1. Site: 76. Done with individual 300/929\n",
      "Slice: 0/1. Site: 76. Done with individual 400/929\n",
      "Slice: 0/1. Site: 76. Done with individual 500/929\n",
      "Slice: 0/1. Site: 76. Done with individual 600/929\n",
      "Slice: 0/1. Site: 76. Done with individual 700/929\n",
      "Slice: 0/1. Site: 76. Done with individual 800/929\n",
      "Slice: 0/1. Site: 76. Done with individual 900/929\n",
      "Site index: 77, non ref allele frequency: 0.4985507246376812\n",
      "Site index: 77, ref allele frequency: 0.5014492753623189\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3713670613562971\n",
      "Minor allele frequency: 0.4985507246376812\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 77. Done with individual 0/929\n",
      "Slice: 0/1. Site: 77. Done with individual 100/929\n",
      "Slice: 0/1. Site: 77. Done with individual 200/929\n",
      "Slice: 0/1. Site: 77. Done with individual 300/929\n",
      "Slice: 0/1. Site: 77. Done with individual 400/929\n",
      "Slice: 0/1. Site: 77. Done with individual 500/929\n",
      "Slice: 0/1. Site: 77. Done with individual 600/929\n",
      "Slice: 0/1. Site: 77. Done with individual 700/929\n",
      "Slice: 0/1. Site: 77. Done with individual 800/929\n",
      "Slice: 0/1. Site: 77. Done with individual 900/929\n",
      "Site index: 78, non ref allele frequency: 0.5073746312684366\n",
      "Site index: 78, ref allele frequency: 0.49262536873156343\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7298170075349839\n",
      "Minor allele frequency: 0.49262536873156343\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 78. Done with individual 0/929\n",
      "Slice: 0/1. Site: 78. Done with individual 100/929\n",
      "Slice: 0/1. Site: 78. Done with individual 200/929\n",
      "Slice: 0/1. Site: 78. Done with individual 300/929\n",
      "Slice: 0/1. Site: 78. Done with individual 400/929\n",
      "Slice: 0/1. Site: 78. Done with individual 500/929\n",
      "Slice: 0/1. Site: 78. Done with individual 600/929\n",
      "Slice: 0/1. Site: 78. Done with individual 700/929\n",
      "Slice: 0/1. Site: 78. Done with individual 800/929\n",
      "Slice: 0/1. Site: 78. Done with individual 900/929\n",
      "Site index: 79, non ref allele frequency: 0.494750656167979\n",
      "Site index: 79, ref allele frequency: 0.505249343832021\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4101184068891281\n",
      "Minor allele frequency: 0.494750656167979\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 79. Done with individual 0/929\n",
      "Slice: 0/1. Site: 79. Done with individual 100/929\n",
      "Slice: 0/1. Site: 79. Done with individual 200/929\n",
      "Slice: 0/1. Site: 79. Done with individual 300/929\n",
      "Slice: 0/1. Site: 79. Done with individual 400/929\n",
      "Slice: 0/1. Site: 79. Done with individual 500/929\n",
      "Slice: 0/1. Site: 79. Done with individual 600/929\n",
      "Slice: 0/1. Site: 79. Done with individual 700/929\n",
      "Slice: 0/1. Site: 79. Done with individual 800/929\n",
      "Slice: 0/1. Site: 79. Done with individual 900/929\n",
      "Site index: 80, non ref allele frequency: 0.5067415730337078\n",
      "Site index: 80, ref allele frequency: 0.49325842696629213\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4790096878363832\n",
      "Minor allele frequency: 0.49325842696629213\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 80. Done with individual 0/929\n",
      "Slice: 0/1. Site: 80. Done with individual 100/929\n",
      "Slice: 0/1. Site: 80. Done with individual 200/929\n",
      "Slice: 0/1. Site: 80. Done with individual 300/929\n",
      "Slice: 0/1. Site: 80. Done with individual 400/929\n",
      "Slice: 0/1. Site: 80. Done with individual 500/929\n",
      "Slice: 0/1. Site: 80. Done with individual 600/929\n",
      "Slice: 0/1. Site: 80. Done with individual 700/929\n",
      "Slice: 0/1. Site: 80. Done with individual 800/929\n",
      "Slice: 0/1. Site: 80. Done with individual 900/929\n",
      "Site index: 81, non ref allele frequency: 0.5043668122270742\n",
      "Site index: 81, ref allele frequency: 0.49563318777292575\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2465016146393972\n",
      "Minor allele frequency: 0.49563318777292575\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 81. Done with individual 0/929\n",
      "Slice: 0/1. Site: 81. Done with individual 100/929\n",
      "Slice: 0/1. Site: 81. Done with individual 200/929\n",
      "Slice: 0/1. Site: 81. Done with individual 300/929\n",
      "Slice: 0/1. Site: 81. Done with individual 400/929\n",
      "Slice: 0/1. Site: 81. Done with individual 500/929\n",
      "Slice: 0/1. Site: 81. Done with individual 600/929\n",
      "Slice: 0/1. Site: 81. Done with individual 700/929\n",
      "Slice: 0/1. Site: 81. Done with individual 800/929\n",
      "Slice: 0/1. Site: 81. Done with individual 900/929\n",
      "Site index: 82, non ref allele frequency: 0.4967948717948718\n",
      "Site index: 82, ref allele frequency: 0.5032051282051282\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.33584499461786865\n",
      "Minor allele frequency: 0.4967948717948718\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 82. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 82. Done with individual 100/929\n",
      "Slice: 0/1. Site: 82. Done with individual 200/929\n",
      "Slice: 0/1. Site: 82. Done with individual 300/929\n",
      "Slice: 0/1. Site: 82. Done with individual 400/929\n",
      "Slice: 0/1. Site: 82. Done with individual 500/929\n",
      "Slice: 0/1. Site: 82. Done with individual 600/929\n",
      "Slice: 0/1. Site: 82. Done with individual 700/929\n",
      "Slice: 0/1. Site: 82. Done with individual 800/929\n",
      "Slice: 0/1. Site: 82. Done with individual 900/929\n",
      "Site index: 83, non ref allele frequency: 0.49634146341463414\n",
      "Site index: 83, ref allele frequency: 0.5036585365853659\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4413347685683531\n",
      "Minor allele frequency: 0.49634146341463414\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 83. Done with individual 0/929\n",
      "Slice: 0/1. Site: 83. Done with individual 100/929\n",
      "Slice: 0/1. Site: 83. Done with individual 200/929\n",
      "Slice: 0/1. Site: 83. Done with individual 300/929\n",
      "Slice: 0/1. Site: 83. Done with individual 400/929\n",
      "Slice: 0/1. Site: 83. Done with individual 500/929\n",
      "Slice: 0/1. Site: 83. Done with individual 600/929\n",
      "Slice: 0/1. Site: 83. Done with individual 700/929\n",
      "Slice: 0/1. Site: 83. Done with individual 800/929\n",
      "Slice: 0/1. Site: 83. Done with individual 900/929\n",
      "Site index: 84, non ref allele frequency: 0.49107142857142855\n",
      "Site index: 84, ref allele frequency: 0.5089285714285714\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.12055974165769645\n",
      "Minor allele frequency: 0.49107142857142855\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 84. Done with individual 0/929\n",
      "Slice: 0/1. Site: 84. Done with individual 100/929\n",
      "Slice: 0/1. Site: 84. Done with individual 200/929\n",
      "Slice: 0/1. Site: 84. Done with individual 300/929\n",
      "Slice: 0/1. Site: 84. Done with individual 400/929\n",
      "Slice: 0/1. Site: 84. Done with individual 500/929\n",
      "Slice: 0/1. Site: 84. Done with individual 600/929\n",
      "Slice: 0/1. Site: 84. Done with individual 700/929\n",
      "Slice: 0/1. Site: 84. Done with individual 800/929\n",
      "Slice: 0/1. Site: 84. Done with individual 900/929\n",
      "Site index: 85, non ref allele frequency: 0.5\n",
      "Site index: 85, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.10118406889128095\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 85. Done with individual 0/929\n",
      "Slice: 0/1. Site: 85. Done with individual 100/929\n",
      "Slice: 0/1. Site: 85. Done with individual 200/929\n",
      "Slice: 0/1. Site: 85. Done with individual 300/929\n",
      "Slice: 0/1. Site: 85. Done with individual 400/929\n",
      "Slice: 0/1. Site: 85. Done with individual 500/929\n",
      "Slice: 0/1. Site: 85. Done with individual 600/929\n",
      "Slice: 0/1. Site: 85. Done with individual 700/929\n",
      "Slice: 0/1. Site: 85. Done with individual 800/929\n",
      "Slice: 0/1. Site: 85. Done with individual 900/929\n",
      "Site index: 86, non ref allele frequency: 0.4954337899543379\n",
      "Site index: 86, ref allele frequency: 0.5045662100456622\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2357373519913886\n",
      "Minor allele frequency: 0.4954337899543379\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 86. Done with individual 0/929\n",
      "Slice: 0/1. Site: 86. Done with individual 100/929\n",
      "Slice: 0/1. Site: 86. Done with individual 200/929\n",
      "Slice: 0/1. Site: 86. Done with individual 300/929\n",
      "Slice: 0/1. Site: 86. Done with individual 400/929\n",
      "Slice: 0/1. Site: 86. Done with individual 500/929\n",
      "Slice: 0/1. Site: 86. Done with individual 600/929\n",
      "Slice: 0/1. Site: 86. Done with individual 700/929\n",
      "Slice: 0/1. Site: 86. Done with individual 800/929\n",
      "Slice: 0/1. Site: 86. Done with individual 900/929\n",
      "Site index: 87, non ref allele frequency: 0.5\n",
      "Site index: 87, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3293864370290635\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 87. Done with individual 0/929\n",
      "Slice: 0/1. Site: 87. Done with individual 100/929\n",
      "Slice: 0/1. Site: 87. Done with individual 200/929\n",
      "Slice: 0/1. Site: 87. Done with individual 300/929\n",
      "Slice: 0/1. Site: 87. Done with individual 400/929\n",
      "Slice: 0/1. Site: 87. Done with individual 500/929\n",
      "Slice: 0/1. Site: 87. Done with individual 600/929\n",
      "Slice: 0/1. Site: 87. Done with individual 700/929\n",
      "Slice: 0/1. Site: 87. Done with individual 800/929\n",
      "Slice: 0/1. Site: 87. Done with individual 900/929\n",
      "Site index: 88, non ref allele frequency: 0.49523809523809526\n",
      "Site index: 88, ref allele frequency: 0.5047619047619047\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5651237890204521\n",
      "Minor allele frequency: 0.49523809523809526\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 88. Done with individual 0/929\n",
      "Slice: 0/1. Site: 88. Done with individual 100/929\n",
      "Slice: 0/1. Site: 88. Done with individual 200/929\n",
      "Slice: 0/1. Site: 88. Done with individual 300/929\n",
      "Slice: 0/1. Site: 88. Done with individual 400/929\n",
      "Slice: 0/1. Site: 88. Done with individual 500/929\n",
      "Slice: 0/1. Site: 88. Done with individual 600/929\n",
      "Slice: 0/1. Site: 88. Done with individual 700/929\n",
      "Slice: 0/1. Site: 88. Done with individual 800/929\n",
      "Slice: 0/1. Site: 88. Done with individual 900/929\n",
      "Site index: 89, non ref allele frequency: 0.4961832061068702\n",
      "Site index: 89, ref allele frequency: 0.5038167938931297\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.14101184068891282\n",
      "Minor allele frequency: 0.4961832061068702\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 89. Done with individual 0/929\n",
      "Slice: 0/1. Site: 89. Done with individual 100/929\n",
      "Slice: 0/1. Site: 89. Done with individual 200/929\n",
      "Slice: 0/1. Site: 89. Done with individual 300/929\n",
      "Slice: 0/1. Site: 89. Done with individual 400/929\n",
      "Slice: 0/1. Site: 89. Done with individual 500/929\n",
      "Slice: 0/1. Site: 89. Done with individual 600/929\n",
      "Slice: 0/1. Site: 89. Done with individual 700/929\n",
      "Slice: 0/1. Site: 89. Done with individual 800/929\n",
      "Slice: 0/1. Site: 89. Done with individual 900/929\n",
      "Site index: 90, non ref allele frequency: 0.49056603773584906\n",
      "Site index: 90, ref allele frequency: 0.5094339622641509\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22820236813778255\n",
      "Minor allele frequency: 0.49056603773584906\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 90. Done with individual 0/929\n",
      "Slice: 0/1. Site: 90. Done with individual 100/929\n",
      "Slice: 0/1. Site: 90. Done with individual 200/929\n",
      "Slice: 0/1. Site: 90. Done with individual 300/929\n",
      "Slice: 0/1. Site: 90. Done with individual 400/929\n",
      "Slice: 0/1. Site: 90. Done with individual 500/929\n",
      "Slice: 0/1. Site: 90. Done with individual 600/929\n",
      "Slice: 0/1. Site: 90. Done with individual 700/929\n",
      "Slice: 0/1. Site: 90. Done with individual 800/929\n",
      "Slice: 0/1. Site: 90. Done with individual 900/929\n",
      "Site index: 91, non ref allele frequency: 0.49193548387096775\n",
      "Site index: 91, ref allele frequency: 0.5080645161290323\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.26695371367061355\n",
      "Minor allele frequency: 0.49193548387096775\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 91. Done with individual 0/929\n",
      "Slice: 0/1. Site: 91. Done with individual 100/929\n",
      "Slice: 0/1. Site: 91. Done with individual 200/929\n",
      "Slice: 0/1. Site: 91. Done with individual 300/929\n",
      "Slice: 0/1. Site: 91. Done with individual 400/929\n",
      "Slice: 0/1. Site: 91. Done with individual 500/929\n",
      "Slice: 0/1. Site: 91. Done with individual 600/929\n",
      "Slice: 0/1. Site: 91. Done with individual 700/929\n",
      "Slice: 0/1. Site: 91. Done with individual 800/929\n",
      "Slice: 0/1. Site: 91. Done with individual 900/929\n",
      "Site index: 92, non ref allele frequency: 0.4934210526315789\n",
      "Site index: 92, ref allele frequency: 0.506578947368421\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.32723358449946177\n",
      "Minor allele frequency: 0.4934210526315789\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 92. Done with individual 0/929\n",
      "Slice: 0/1. Site: 92. Done with individual 100/929\n",
      "Slice: 0/1. Site: 92. Done with individual 200/929\n",
      "Slice: 0/1. Site: 92. Done with individual 300/929\n",
      "Slice: 0/1. Site: 92. Done with individual 400/929\n",
      "Slice: 0/1. Site: 92. Done with individual 500/929\n",
      "Slice: 0/1. Site: 92. Done with individual 600/929\n",
      "Slice: 0/1. Site: 92. Done with individual 700/929\n",
      "Slice: 0/1. Site: 92. Done with individual 800/929\n",
      "Slice: 0/1. Site: 92. Done with individual 900/929\n",
      "Site index: 93, non ref allele frequency: 0.5041551246537396\n",
      "Site index: 93, ref allele frequency: 0.49584487534626037\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.38858988159311086\n",
      "Minor allele frequency: 0.49584487534626037\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 93. Done with individual 0/929\n",
      "Slice: 0/1. Site: 93. Done with individual 100/929\n",
      "Slice: 0/1. Site: 93. Done with individual 200/929\n",
      "Slice: 0/1. Site: 93. Done with individual 300/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/1. Site: 93. Done with individual 400/929\n",
      "Slice: 0/1. Site: 93. Done with individual 500/929\n",
      "Slice: 0/1. Site: 93. Done with individual 600/929\n",
      "Slice: 0/1. Site: 93. Done with individual 700/929\n",
      "Slice: 0/1. Site: 93. Done with individual 800/929\n",
      "Slice: 0/1. Site: 93. Done with individual 900/929\n",
      "Site index: 94, non ref allele frequency: 0.5023584905660378\n",
      "Site index: 94, ref allele frequency: 0.49764150943396224\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22820236813778255\n",
      "Minor allele frequency: 0.49764150943396224\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 94. Done with individual 0/929\n",
      "Slice: 0/1. Site: 94. Done with individual 100/929\n",
      "Slice: 0/1. Site: 94. Done with individual 200/929\n",
      "Slice: 0/1. Site: 94. Done with individual 300/929\n",
      "Slice: 0/1. Site: 94. Done with individual 400/929\n",
      "Slice: 0/1. Site: 94. Done with individual 500/929\n",
      "Slice: 0/1. Site: 94. Done with individual 600/929\n",
      "Slice: 0/1. Site: 94. Done with individual 700/929\n",
      "Slice: 0/1. Site: 94. Done with individual 800/929\n",
      "Slice: 0/1. Site: 94. Done with individual 900/929\n",
      "Site index: 95, non ref allele frequency: 0.49850299401197606\n",
      "Site index: 95, ref allele frequency: 0.5014970059880239\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3595263724434876\n",
      "Minor allele frequency: 0.49850299401197606\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 95. Done with individual 0/929\n",
      "Slice: 0/1. Site: 95. Done with individual 100/929\n",
      "Slice: 0/1. Site: 95. Done with individual 200/929\n",
      "Slice: 0/1. Site: 95. Done with individual 300/929\n",
      "Slice: 0/1. Site: 95. Done with individual 400/929\n",
      "Slice: 0/1. Site: 95. Done with individual 500/929\n",
      "Slice: 0/1. Site: 95. Done with individual 600/929\n",
      "Slice: 0/1. Site: 95. Done with individual 700/929\n",
      "Slice: 0/1. Site: 95. Done with individual 800/929\n",
      "Slice: 0/1. Site: 95. Done with individual 900/929\n",
      "Site index: 96, non ref allele frequency: 0.5049019607843137\n",
      "Site index: 96, ref allele frequency: 0.4950980392156863\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.21959095801937567\n",
      "Minor allele frequency: 0.4950980392156863\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 96. Done with individual 0/929\n",
      "Slice: 0/1. Site: 96. Done with individual 100/929\n",
      "Slice: 0/1. Site: 96. Done with individual 200/929\n",
      "Slice: 0/1. Site: 96. Done with individual 300/929\n",
      "Slice: 0/1. Site: 96. Done with individual 400/929\n",
      "Slice: 0/1. Site: 96. Done with individual 500/929\n",
      "Slice: 0/1. Site: 96. Done with individual 600/929\n",
      "Slice: 0/1. Site: 96. Done with individual 700/929\n",
      "Slice: 0/1. Site: 96. Done with individual 800/929\n",
      "Slice: 0/1. Site: 96. Done with individual 900/929\n",
      "Site index: 97, non ref allele frequency: 0.4935064935064935\n",
      "Site index: 97, ref allele frequency: 0.5064935064935064\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.16576964477933262\n",
      "Minor allele frequency: 0.4935064935064935\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 97. Done with individual 0/929\n",
      "Slice: 0/1. Site: 97. Done with individual 100/929\n",
      "Slice: 0/1. Site: 97. Done with individual 200/929\n",
      "Slice: 0/1. Site: 97. Done with individual 300/929\n",
      "Slice: 0/1. Site: 97. Done with individual 400/929\n",
      "Slice: 0/1. Site: 97. Done with individual 500/929\n",
      "Slice: 0/1. Site: 97. Done with individual 600/929\n",
      "Slice: 0/1. Site: 97. Done with individual 700/929\n",
      "Slice: 0/1. Site: 97. Done with individual 800/929\n",
      "Slice: 0/1. Site: 97. Done with individual 900/929\n",
      "Site index: 98, non ref allele frequency: 0.5024390243902439\n",
      "Site index: 98, ref allele frequency: 0.4975609756097561\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22066738428417654\n",
      "Minor allele frequency: 0.4975609756097561\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 98. Done with individual 0/929\n",
      "Slice: 0/1. Site: 98. Done with individual 100/929\n",
      "Slice: 0/1. Site: 98. Done with individual 200/929\n",
      "Slice: 0/1. Site: 98. Done with individual 300/929\n",
      "Slice: 0/1. Site: 98. Done with individual 400/929\n",
      "Slice: 0/1. Site: 98. Done with individual 500/929\n",
      "Slice: 0/1. Site: 98. Done with individual 600/929\n",
      "Slice: 0/1. Site: 98. Done with individual 700/929\n",
      "Slice: 0/1. Site: 98. Done with individual 800/929\n",
      "Slice: 0/1. Site: 98. Done with individual 900/929\n",
      "Site index: 99, non ref allele frequency: 0.49049429657794674\n",
      "Site index: 99, ref allele frequency: 0.5095057034220533\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2831001076426265\n",
      "Minor allele frequency: 0.49049429657794674\n",
      "Passed guardrails\n",
      "Slice: 0/1. Site: 99. Done with individual 0/929\n",
      "Slice: 0/1. Site: 99. Done with individual 100/929\n",
      "Slice: 0/1. Site: 99. Done with individual 200/929\n",
      "Slice: 0/1. Site: 99. Done with individual 300/929\n",
      "Slice: 0/1. Site: 99. Done with individual 400/929\n",
      "Slice: 0/1. Site: 99. Done with individual 500/929\n",
      "Slice: 0/1. Site: 99. Done with individual 600/929\n",
      "Slice: 0/1. Site: 99. Done with individual 700/929\n",
      "Slice: 0/1. Site: 99. Done with individual 800/929\n",
      "Slice: 0/1. Site: 99. Done with individual 900/929\n",
      "chr9: slice 1/1\n",
      "\n",
      "#############\n",
      "\n",
      "Slice index: 1. Slice size: 78 \n",
      "Output files: C:\\Data\\HUJI\\hgdp\\mock/chr9/count_dist_slice_1.tsv C:\\Data\\HUJI\\hgdp\\mock/chr9/sites_ids_slice_1.tsv\n",
      "Site index: 0, non ref allele frequency: 0.496309963099631\n",
      "Site index: 0, ref allele frequency: 0.503690036900369\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2917115177610334\n",
      "Minor allele frequency: 0.496309963099631\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 0. Done with individual 0/929\n",
      "Slice: 1/1. Site: 0. Done with individual 100/929\n",
      "Slice: 1/1. Site: 0. Done with individual 200/929\n",
      "Slice: 1/1. Site: 0. Done with individual 300/929\n",
      "Slice: 1/1. Site: 0. Done with individual 400/929\n",
      "Slice: 1/1. Site: 0. Done with individual 500/929\n",
      "Slice: 1/1. Site: 0. Done with individual 600/929\n",
      "Slice: 1/1. Site: 0. Done with individual 700/929\n",
      "Slice: 1/1. Site: 0. Done with individual 800/929\n",
      "Slice: 1/1. Site: 0. Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.49570200573065903\n",
      "Site index: 1, ref allele frequency: 0.504297994269341\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3756727664155005\n",
      "Minor allele frequency: 0.49570200573065903\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 1. Done with individual 0/929\n",
      "Slice: 1/1. Site: 1. Done with individual 100/929\n",
      "Slice: 1/1. Site: 1. Done with individual 200/929\n",
      "Slice: 1/1. Site: 1. Done with individual 300/929\n",
      "Slice: 1/1. Site: 1. Done with individual 400/929\n",
      "Slice: 1/1. Site: 1. Done with individual 500/929\n",
      "Slice: 1/1. Site: 1. Done with individual 600/929\n",
      "Slice: 1/1. Site: 1. Done with individual 700/929\n",
      "Slice: 1/1. Site: 1. Done with individual 800/929\n",
      "Slice: 1/1. Site: 1. Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.4969230769230769\n",
      "Site index: 2, ref allele frequency: 0.5030769230769231\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34983853606027987\n",
      "Minor allele frequency: 0.4969230769230769\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 2. Done with individual 0/929\n",
      "Slice: 1/1. Site: 2. Done with individual 100/929\n",
      "Slice: 1/1. Site: 2. Done with individual 200/929\n",
      "Slice: 1/1. Site: 2. Done with individual 300/929\n",
      "Slice: 1/1. Site: 2. Done with individual 400/929\n",
      "Slice: 1/1. Site: 2. Done with individual 500/929\n",
      "Slice: 1/1. Site: 2. Done with individual 600/929\n",
      "Slice: 1/1. Site: 2. Done with individual 700/929\n",
      "Slice: 1/1. Site: 2. Done with individual 800/929\n",
      "Slice: 1/1. Site: 2. Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.49572649572649574\n",
      "Site index: 3, ref allele frequency: 0.5042735042735043\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2518837459634015\n",
      "Minor allele frequency: 0.49572649572649574\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 3. Done with individual 0/929\n",
      "Slice: 1/1. Site: 3. Done with individual 100/929\n",
      "Slice: 1/1. Site: 3. Done with individual 200/929\n",
      "Slice: 1/1. Site: 3. Done with individual 300/929\n",
      "Slice: 1/1. Site: 3. Done with individual 400/929\n",
      "Slice: 1/1. Site: 3. Done with individual 500/929\n",
      "Slice: 1/1. Site: 3. Done with individual 600/929\n",
      "Slice: 1/1. Site: 3. Done with individual 700/929\n",
      "Slice: 1/1. Site: 3. Done with individual 800/929\n",
      "Slice: 1/1. Site: 3. Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.5094850948509485\n",
      "Site index: 4, ref allele frequency: 0.4905149051490515\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.39720129171151775\n",
      "Minor allele frequency: 0.4905149051490515\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 4. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 4. Done with individual 100/929\n",
      "Slice: 1/1. Site: 4. Done with individual 200/929\n",
      "Slice: 1/1. Site: 4. Done with individual 300/929\n",
      "Slice: 1/1. Site: 4. Done with individual 400/929\n",
      "Slice: 1/1. Site: 4. Done with individual 500/929\n",
      "Slice: 1/1. Site: 4. Done with individual 600/929\n",
      "Slice: 1/1. Site: 4. Done with individual 700/929\n",
      "Slice: 1/1. Site: 4. Done with individual 800/929\n",
      "Slice: 1/1. Site: 4. Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.49375866851595007\n",
      "Site index: 5, ref allele frequency: 0.5062413314840499\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7761033369214209\n",
      "Minor allele frequency: 0.49375866851595007\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 5. Done with individual 0/929\n",
      "Slice: 1/1. Site: 5. Done with individual 100/929\n",
      "Slice: 1/1. Site: 5. Done with individual 200/929\n",
      "Slice: 1/1. Site: 5. Done with individual 300/929\n",
      "Slice: 1/1. Site: 5. Done with individual 400/929\n",
      "Slice: 1/1. Site: 5. Done with individual 500/929\n",
      "Slice: 1/1. Site: 5. Done with individual 600/929\n",
      "Slice: 1/1. Site: 5. Done with individual 700/929\n",
      "Slice: 1/1. Site: 5. Done with individual 800/929\n",
      "Slice: 1/1. Site: 5. Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.5\n",
      "Site index: 6, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.36598493003229277\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 6. Done with individual 0/929\n",
      "Slice: 1/1. Site: 6. Done with individual 100/929\n",
      "Slice: 1/1. Site: 6. Done with individual 200/929\n",
      "Slice: 1/1. Site: 6. Done with individual 300/929\n",
      "Slice: 1/1. Site: 6. Done with individual 400/929\n",
      "Slice: 1/1. Site: 6. Done with individual 500/929\n",
      "Slice: 1/1. Site: 6. Done with individual 600/929\n",
      "Slice: 1/1. Site: 6. Done with individual 700/929\n",
      "Slice: 1/1. Site: 6. Done with individual 800/929\n",
      "Slice: 1/1. Site: 6. Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.5087579617834395\n",
      "Site index: 7, ref allele frequency: 0.4912420382165605\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6759956942949408\n",
      "Minor allele frequency: 0.4912420382165605\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 7. Done with individual 0/929\n",
      "Slice: 1/1. Site: 7. Done with individual 100/929\n",
      "Slice: 1/1. Site: 7. Done with individual 200/929\n",
      "Slice: 1/1. Site: 7. Done with individual 300/929\n",
      "Slice: 1/1. Site: 7. Done with individual 400/929\n",
      "Slice: 1/1. Site: 7. Done with individual 500/929\n",
      "Slice: 1/1. Site: 7. Done with individual 600/929\n",
      "Slice: 1/1. Site: 7. Done with individual 700/929\n",
      "Slice: 1/1. Site: 7. Done with individual 800/929\n",
      "Slice: 1/1. Site: 7. Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.49920886075949367\n",
      "Site index: 8, ref allele frequency: 0.5007911392405063\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6803013993541442\n",
      "Minor allele frequency: 0.49920886075949367\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 8. Done with individual 0/929\n",
      "Slice: 1/1. Site: 8. Done with individual 100/929\n",
      "Slice: 1/1. Site: 8. Done with individual 200/929\n",
      "Slice: 1/1. Site: 8. Done with individual 300/929\n",
      "Slice: 1/1. Site: 8. Done with individual 400/929\n",
      "Slice: 1/1. Site: 8. Done with individual 500/929\n",
      "Slice: 1/1. Site: 8. Done with individual 600/929\n",
      "Slice: 1/1. Site: 8. Done with individual 700/929\n",
      "Slice: 1/1. Site: 8. Done with individual 800/929\n",
      "Slice: 1/1. Site: 8. Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.5083333333333333\n",
      "Site index: 9, ref allele frequency: 0.49166666666666664\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.77502691065662\n",
      "Minor allele frequency: 0.49166666666666664\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 9. Done with individual 0/929\n",
      "Slice: 1/1. Site: 9. Done with individual 100/929\n",
      "Slice: 1/1. Site: 9. Done with individual 200/929\n",
      "Slice: 1/1. Site: 9. Done with individual 300/929\n",
      "Slice: 1/1. Site: 9. Done with individual 400/929\n",
      "Slice: 1/1. Site: 9. Done with individual 500/929\n",
      "Slice: 1/1. Site: 9. Done with individual 600/929\n",
      "Slice: 1/1. Site: 9. Done with individual 700/929\n",
      "Slice: 1/1. Site: 9. Done with individual 800/929\n",
      "Slice: 1/1. Site: 9. Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.5085106382978724\n",
      "Site index: 10, ref allele frequency: 0.49148936170212765\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7588805166846071\n",
      "Minor allele frequency: 0.49148936170212765\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 10. Done with individual 0/929\n",
      "Slice: 1/1. Site: 10. Done with individual 100/929\n",
      "Slice: 1/1. Site: 10. Done with individual 200/929\n",
      "Slice: 1/1. Site: 10. Done with individual 300/929\n",
      "Slice: 1/1. Site: 10. Done with individual 400/929\n",
      "Slice: 1/1. Site: 10. Done with individual 500/929\n",
      "Slice: 1/1. Site: 10. Done with individual 600/929\n",
      "Slice: 1/1. Site: 10. Done with individual 700/929\n",
      "Slice: 1/1. Site: 10. Done with individual 800/929\n",
      "Slice: 1/1. Site: 10. Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.5063191153238547\n",
      "Site index: 11, ref allele frequency: 0.49368088467614535\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6813778256189451\n",
      "Minor allele frequency: 0.49368088467614535\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 11. Done with individual 0/929\n",
      "Slice: 1/1. Site: 11. Done with individual 100/929\n",
      "Slice: 1/1. Site: 11. Done with individual 200/929\n",
      "Slice: 1/1. Site: 11. Done with individual 300/929\n",
      "Slice: 1/1. Site: 11. Done with individual 400/929\n",
      "Slice: 1/1. Site: 11. Done with individual 500/929\n",
      "Slice: 1/1. Site: 11. Done with individual 600/929\n",
      "Slice: 1/1. Site: 11. Done with individual 700/929\n",
      "Slice: 1/1. Site: 11. Done with individual 800/929\n",
      "Slice: 1/1. Site: 11. Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.5097629009762901\n",
      "Site index: 12, ref allele frequency: 0.4902370990237099\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7717976318622174\n",
      "Minor allele frequency: 0.4902370990237099\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 12. Done with individual 0/929\n",
      "Slice: 1/1. Site: 12. Done with individual 100/929\n",
      "Slice: 1/1. Site: 12. Done with individual 200/929\n",
      "Slice: 1/1. Site: 12. Done with individual 300/929\n",
      "Slice: 1/1. Site: 12. Done with individual 400/929\n",
      "Slice: 1/1. Site: 12. Done with individual 500/929\n",
      "Slice: 1/1. Site: 12. Done with individual 600/929\n",
      "Slice: 1/1. Site: 12. Done with individual 700/929\n",
      "Slice: 1/1. Site: 12. Done with individual 800/929\n",
      "Slice: 1/1. Site: 12. Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.5021582733812949\n",
      "Site index: 13, ref allele frequency: 0.497841726618705\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7481162540365985\n",
      "Minor allele frequency: 0.497841726618705\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 13. Done with individual 0/929\n",
      "Slice: 1/1. Site: 13. Done with individual 100/929\n",
      "Slice: 1/1. Site: 13. Done with individual 200/929\n",
      "Slice: 1/1. Site: 13. Done with individual 300/929\n",
      "Slice: 1/1. Site: 13. Done with individual 400/929\n",
      "Slice: 1/1. Site: 13. Done with individual 500/929\n",
      "Slice: 1/1. Site: 13. Done with individual 600/929\n",
      "Slice: 1/1. Site: 13. Done with individual 700/929\n",
      "Slice: 1/1. Site: 13. Done with individual 800/929\n",
      "Slice: 1/1. Site: 13. Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.5053929121725732\n",
      "Site index: 14, ref allele frequency: 0.4946070878274268\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6986006458557589\n",
      "Minor allele frequency: 0.4946070878274268\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 14. Done with individual 0/929\n",
      "Slice: 1/1. Site: 14. Done with individual 100/929\n",
      "Slice: 1/1. Site: 14. Done with individual 200/929\n",
      "Slice: 1/1. Site: 14. Done with individual 300/929\n",
      "Slice: 1/1. Site: 14. Done with individual 400/929\n",
      "Slice: 1/1. Site: 14. Done with individual 500/929\n",
      "Slice: 1/1. Site: 14. Done with individual 600/929\n",
      "Slice: 1/1. Site: 14. Done with individual 700/929\n",
      "Slice: 1/1. Site: 14. Done with individual 800/929\n",
      "Slice: 1/1. Site: 14. Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.5078125\n",
      "Site index: 15, ref allele frequency: 0.4921875\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6889128094725512\n",
      "Minor allele frequency: 0.4921875\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 15. Done with individual 0/929\n",
      "Slice: 1/1. Site: 15. Done with individual 100/929\n",
      "Slice: 1/1. Site: 15. Done with individual 200/929\n",
      "Slice: 1/1. Site: 15. Done with individual 300/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 15. Done with individual 400/929\n",
      "Slice: 1/1. Site: 15. Done with individual 500/929\n",
      "Slice: 1/1. Site: 15. Done with individual 600/929\n",
      "Slice: 1/1. Site: 15. Done with individual 700/929\n",
      "Slice: 1/1. Site: 15. Done with individual 800/929\n",
      "Slice: 1/1. Site: 15. Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.5\n",
      "Site index: 16, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7707212055974165\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 16. Done with individual 0/929\n",
      "Slice: 1/1. Site: 16. Done with individual 100/929\n",
      "Slice: 1/1. Site: 16. Done with individual 200/929\n",
      "Slice: 1/1. Site: 16. Done with individual 300/929\n",
      "Slice: 1/1. Site: 16. Done with individual 400/929\n",
      "Slice: 1/1. Site: 16. Done with individual 500/929\n",
      "Slice: 1/1. Site: 16. Done with individual 600/929\n",
      "Slice: 1/1. Site: 16. Done with individual 700/929\n",
      "Slice: 1/1. Site: 16. Done with individual 800/929\n",
      "Slice: 1/1. Site: 16. Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.49122807017543857\n",
      "Site index: 17, ref allele frequency: 0.5087719298245614\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3681377825618945\n",
      "Minor allele frequency: 0.49122807017543857\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 17. Done with individual 0/929\n",
      "Slice: 1/1. Site: 17. Done with individual 100/929\n",
      "Slice: 1/1. Site: 17. Done with individual 200/929\n",
      "Slice: 1/1. Site: 17. Done with individual 300/929\n",
      "Slice: 1/1. Site: 17. Done with individual 400/929\n",
      "Slice: 1/1. Site: 17. Done with individual 500/929\n",
      "Slice: 1/1. Site: 17. Done with individual 600/929\n",
      "Slice: 1/1. Site: 17. Done with individual 700/929\n",
      "Slice: 1/1. Site: 17. Done with individual 800/929\n",
      "Slice: 1/1. Site: 17. Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.5068762278978389\n",
      "Site index: 18, ref allele frequency: 0.4931237721021611\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5479009687836384\n",
      "Minor allele frequency: 0.4931237721021611\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 18. Done with individual 0/929\n",
      "Slice: 1/1. Site: 18. Done with individual 100/929\n",
      "Slice: 1/1. Site: 18. Done with individual 200/929\n",
      "Slice: 1/1. Site: 18. Done with individual 300/929\n",
      "Slice: 1/1. Site: 18. Done with individual 400/929\n",
      "Slice: 1/1. Site: 18. Done with individual 500/929\n",
      "Slice: 1/1. Site: 18. Done with individual 600/929\n",
      "Slice: 1/1. Site: 18. Done with individual 700/929\n",
      "Slice: 1/1. Site: 18. Done with individual 800/929\n",
      "Slice: 1/1. Site: 18. Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.5007246376811594\n",
      "Site index: 19, ref allele frequency: 0.49927536231884057\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7427341227125942\n",
      "Minor allele frequency: 0.49927536231884057\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 19. Done with individual 0/929\n",
      "Slice: 1/1. Site: 19. Done with individual 100/929\n",
      "Slice: 1/1. Site: 19. Done with individual 200/929\n",
      "Slice: 1/1. Site: 19. Done with individual 300/929\n",
      "Slice: 1/1. Site: 19. Done with individual 400/929\n",
      "Slice: 1/1. Site: 19. Done with individual 500/929\n",
      "Slice: 1/1. Site: 19. Done with individual 600/929\n",
      "Slice: 1/1. Site: 19. Done with individual 700/929\n",
      "Slice: 1/1. Site: 19. Done with individual 800/929\n",
      "Slice: 1/1. Site: 19. Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.5\n",
      "Site index: 20, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7405812701829925\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 20. Done with individual 0/929\n",
      "Slice: 1/1. Site: 20. Done with individual 100/929\n",
      "Slice: 1/1. Site: 20. Done with individual 200/929\n",
      "Slice: 1/1. Site: 20. Done with individual 300/929\n",
      "Slice: 1/1. Site: 20. Done with individual 400/929\n",
      "Slice: 1/1. Site: 20. Done with individual 500/929\n",
      "Slice: 1/1. Site: 20. Done with individual 600/929\n",
      "Slice: 1/1. Site: 20. Done with individual 700/929\n",
      "Slice: 1/1. Site: 20. Done with individual 800/929\n",
      "Slice: 1/1. Site: 20. Done with individual 900/929\n",
      "Site index: 21, non ref allele frequency: 0.49393939393939396\n",
      "Site index: 21, ref allele frequency: 0.5060606060606061\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7104413347685683\n",
      "Minor allele frequency: 0.49393939393939396\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 21. Done with individual 0/929\n",
      "Slice: 1/1. Site: 21. Done with individual 100/929\n",
      "Slice: 1/1. Site: 21. Done with individual 200/929\n",
      "Slice: 1/1. Site: 21. Done with individual 300/929\n",
      "Slice: 1/1. Site: 21. Done with individual 400/929\n",
      "Slice: 1/1. Site: 21. Done with individual 500/929\n",
      "Slice: 1/1. Site: 21. Done with individual 600/929\n",
      "Slice: 1/1. Site: 21. Done with individual 700/929\n",
      "Slice: 1/1. Site: 21. Done with individual 800/929\n",
      "Slice: 1/1. Site: 21. Done with individual 900/929\n",
      "Site index: 22, non ref allele frequency: 0.4925249169435216\n",
      "Site index: 22, ref allele frequency: 0.5074750830564784\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6480086114101185\n",
      "Minor allele frequency: 0.4925249169435216\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 22. Done with individual 0/929\n",
      "Slice: 1/1. Site: 22. Done with individual 100/929\n",
      "Slice: 1/1. Site: 22. Done with individual 200/929\n",
      "Slice: 1/1. Site: 22. Done with individual 300/929\n",
      "Slice: 1/1. Site: 22. Done with individual 400/929\n",
      "Slice: 1/1. Site: 22. Done with individual 500/929\n",
      "Slice: 1/1. Site: 22. Done with individual 600/929\n",
      "Slice: 1/1. Site: 22. Done with individual 700/929\n",
      "Slice: 1/1. Site: 22. Done with individual 800/929\n",
      "Slice: 1/1. Site: 22. Done with individual 900/929\n",
      "Site index: 23, non ref allele frequency: 0.5044642857142857\n",
      "Site index: 23, ref allele frequency: 0.4955357142857143\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.12055974165769645\n",
      "Minor allele frequency: 0.4955357142857143\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 23. Done with individual 0/929\n",
      "Slice: 1/1. Site: 23. Done with individual 100/929\n",
      "Slice: 1/1. Site: 23. Done with individual 200/929\n",
      "Slice: 1/1. Site: 23. Done with individual 300/929\n",
      "Slice: 1/1. Site: 23. Done with individual 400/929\n",
      "Slice: 1/1. Site: 23. Done with individual 500/929\n",
      "Slice: 1/1. Site: 23. Done with individual 600/929\n",
      "Slice: 1/1. Site: 23. Done with individual 700/929\n",
      "Slice: 1/1. Site: 23. Done with individual 800/929\n",
      "Slice: 1/1. Site: 23. Done with individual 900/929\n",
      "Site index: 24, non ref allele frequency: 0.49704579025110784\n",
      "Site index: 24, ref allele frequency: 0.5029542097488922\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.728740581270183\n",
      "Minor allele frequency: 0.49704579025110784\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 24. Done with individual 0/929\n",
      "Slice: 1/1. Site: 24. Done with individual 100/929\n",
      "Slice: 1/1. Site: 24. Done with individual 200/929\n",
      "Slice: 1/1. Site: 24. Done with individual 300/929\n",
      "Slice: 1/1. Site: 24. Done with individual 400/929\n",
      "Slice: 1/1. Site: 24. Done with individual 500/929\n",
      "Slice: 1/1. Site: 24. Done with individual 600/929\n",
      "Slice: 1/1. Site: 24. Done with individual 700/929\n",
      "Slice: 1/1. Site: 24. Done with individual 800/929\n",
      "Slice: 1/1. Site: 24. Done with individual 900/929\n",
      "Site index: 25, non ref allele frequency: 0.5033167495854063\n",
      "Site index: 25, ref allele frequency: 0.4966832504145937\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6490850376749193\n",
      "Minor allele frequency: 0.4966832504145937\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 25. Done with individual 0/929\n",
      "Slice: 1/1. Site: 25. Done with individual 100/929\n",
      "Slice: 1/1. Site: 25. Done with individual 200/929\n",
      "Slice: 1/1. Site: 25. Done with individual 300/929\n",
      "Slice: 1/1. Site: 25. Done with individual 400/929\n",
      "Slice: 1/1. Site: 25. Done with individual 500/929\n",
      "Slice: 1/1. Site: 25. Done with individual 600/929\n",
      "Slice: 1/1. Site: 25. Done with individual 700/929\n",
      "Slice: 1/1. Site: 25. Done with individual 800/929\n",
      "Slice: 1/1. Site: 25. Done with individual 900/929\n",
      "Site index: 26, non ref allele frequency: 0.49683544303797467\n",
      "Site index: 26, ref allele frequency: 0.5031645569620253\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.17007534983853606\n",
      "Minor allele frequency: 0.49683544303797467\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 26. Done with individual 0/929\n",
      "Slice: 1/1. Site: 26. Done with individual 100/929\n",
      "Slice: 1/1. Site: 26. Done with individual 200/929\n",
      "Slice: 1/1. Site: 26. Done with individual 300/929\n",
      "Slice: 1/1. Site: 26. Done with individual 400/929\n",
      "Slice: 1/1. Site: 26. Done with individual 500/929\n",
      "Slice: 1/1. Site: 26. Done with individual 600/929\n",
      "Slice: 1/1. Site: 26. Done with individual 700/929\n",
      "Slice: 1/1. Site: 26. Done with individual 800/929\n",
      "Slice: 1/1. Site: 26. Done with individual 900/929\n",
      "Site index: 27, non ref allele frequency: 0.5047169811320755\n",
      "Site index: 27, ref allele frequency: 0.49528301886792453\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5705059203444564\n",
      "Minor allele frequency: 0.49528301886792453\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 27. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 27. Done with individual 100/929\n",
      "Slice: 1/1. Site: 27. Done with individual 200/929\n",
      "Slice: 1/1. Site: 27. Done with individual 300/929\n",
      "Slice: 1/1. Site: 27. Done with individual 400/929\n",
      "Slice: 1/1. Site: 27. Done with individual 500/929\n",
      "Slice: 1/1. Site: 27. Done with individual 600/929\n",
      "Slice: 1/1. Site: 27. Done with individual 700/929\n",
      "Slice: 1/1. Site: 27. Done with individual 800/929\n",
      "Slice: 1/1. Site: 27. Done with individual 900/929\n",
      "Site index: 28, non ref allele frequency: 0.5024752475247525\n",
      "Site index: 28, ref allele frequency: 0.4975247524752475\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6523143164693218\n",
      "Minor allele frequency: 0.4975247524752475\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 28. Done with individual 0/929\n",
      "Slice: 1/1. Site: 28. Done with individual 100/929\n",
      "Slice: 1/1. Site: 28. Done with individual 200/929\n",
      "Slice: 1/1. Site: 28. Done with individual 300/929\n",
      "Slice: 1/1. Site: 28. Done with individual 400/929\n",
      "Slice: 1/1. Site: 28. Done with individual 500/929\n",
      "Slice: 1/1. Site: 28. Done with individual 600/929\n",
      "Slice: 1/1. Site: 28. Done with individual 700/929\n",
      "Slice: 1/1. Site: 28. Done with individual 800/929\n",
      "Slice: 1/1. Site: 28. Done with individual 900/929\n",
      "Site index: 29, non ref allele frequency: 0.4969230769230769\n",
      "Site index: 29, ref allele frequency: 0.5030769230769231\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6996770721205597\n",
      "Minor allele frequency: 0.4969230769230769\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 29. Done with individual 0/929\n",
      "Slice: 1/1. Site: 29. Done with individual 100/929\n",
      "Slice: 1/1. Site: 29. Done with individual 200/929\n",
      "Slice: 1/1. Site: 29. Done with individual 300/929\n",
      "Slice: 1/1. Site: 29. Done with individual 400/929\n",
      "Slice: 1/1. Site: 29. Done with individual 500/929\n",
      "Slice: 1/1. Site: 29. Done with individual 600/929\n",
      "Slice: 1/1. Site: 29. Done with individual 700/929\n",
      "Slice: 1/1. Site: 29. Done with individual 800/929\n",
      "Slice: 1/1. Site: 29. Done with individual 900/929\n",
      "Site index: 30, non ref allele frequency: 0.5\n",
      "Site index: 30, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.12055974165769645\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 30. Done with individual 0/929\n",
      "Slice: 1/1. Site: 30. Done with individual 100/929\n",
      "Slice: 1/1. Site: 30. Done with individual 200/929\n",
      "Slice: 1/1. Site: 30. Done with individual 300/929\n",
      "Slice: 1/1. Site: 30. Done with individual 400/929\n",
      "Slice: 1/1. Site: 30. Done with individual 500/929\n",
      "Slice: 1/1. Site: 30. Done with individual 600/929\n",
      "Slice: 1/1. Site: 30. Done with individual 700/929\n",
      "Slice: 1/1. Site: 30. Done with individual 800/929\n",
      "Slice: 1/1. Site: 30. Done with individual 900/929\n",
      "Site index: 31, non ref allele frequency: 0.5040983606557377\n",
      "Site index: 31, ref allele frequency: 0.4959016393442623\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.13132400430570507\n",
      "Minor allele frequency: 0.4959016393442623\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 31. Done with individual 0/929\n",
      "Slice: 1/1. Site: 31. Done with individual 100/929\n",
      "Slice: 1/1. Site: 31. Done with individual 200/929\n",
      "Slice: 1/1. Site: 31. Done with individual 300/929\n",
      "Slice: 1/1. Site: 31. Done with individual 400/929\n",
      "Slice: 1/1. Site: 31. Done with individual 500/929\n",
      "Slice: 1/1. Site: 31. Done with individual 600/929\n",
      "Slice: 1/1. Site: 31. Done with individual 700/929\n",
      "Slice: 1/1. Site: 31. Done with individual 800/929\n",
      "Slice: 1/1. Site: 31. Done with individual 900/929\n",
      "Site index: 32, non ref allele frequency: 0.4922737306843267\n",
      "Site index: 32, ref allele frequency: 0.5077262693156733\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4876210979547901\n",
      "Minor allele frequency: 0.4922737306843267\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 32. Done with individual 0/929\n",
      "Slice: 1/1. Site: 32. Done with individual 100/929\n",
      "Slice: 1/1. Site: 32. Done with individual 200/929\n",
      "Slice: 1/1. Site: 32. Done with individual 300/929\n",
      "Slice: 1/1. Site: 32. Done with individual 400/929\n",
      "Slice: 1/1. Site: 32. Done with individual 500/929\n",
      "Slice: 1/1. Site: 32. Done with individual 600/929\n",
      "Slice: 1/1. Site: 32. Done with individual 700/929\n",
      "Slice: 1/1. Site: 32. Done with individual 800/929\n",
      "Slice: 1/1. Site: 32. Done with individual 900/929\n",
      "Site index: 33, non ref allele frequency: 0.4956140350877193\n",
      "Site index: 33, ref allele frequency: 0.5043859649122807\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4908503767491927\n",
      "Minor allele frequency: 0.4956140350877193\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 33. Done with individual 0/929\n",
      "Slice: 1/1. Site: 33. Done with individual 100/929\n",
      "Slice: 1/1. Site: 33. Done with individual 200/929\n",
      "Slice: 1/1. Site: 33. Done with individual 300/929\n",
      "Slice: 1/1. Site: 33. Done with individual 400/929\n",
      "Slice: 1/1. Site: 33. Done with individual 500/929\n",
      "Slice: 1/1. Site: 33. Done with individual 600/929\n",
      "Slice: 1/1. Site: 33. Done with individual 700/929\n",
      "Slice: 1/1. Site: 33. Done with individual 800/929\n",
      "Slice: 1/1. Site: 33. Done with individual 900/929\n",
      "Site index: 34, non ref allele frequency: 0.4946808510638298\n",
      "Site index: 34, ref allele frequency: 0.5053191489361702\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7082884822389667\n",
      "Minor allele frequency: 0.4946808510638298\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 34. Done with individual 0/929\n",
      "Slice: 1/1. Site: 34. Done with individual 100/929\n",
      "Slice: 1/1. Site: 34. Done with individual 200/929\n",
      "Slice: 1/1. Site: 34. Done with individual 300/929\n",
      "Slice: 1/1. Site: 34. Done with individual 400/929\n",
      "Slice: 1/1. Site: 34. Done with individual 500/929\n",
      "Slice: 1/1. Site: 34. Done with individual 600/929\n",
      "Slice: 1/1. Site: 34. Done with individual 700/929\n",
      "Slice: 1/1. Site: 34. Done with individual 800/929\n",
      "Slice: 1/1. Site: 34. Done with individual 900/929\n",
      "Site index: 35, non ref allele frequency: 0.4900662251655629\n",
      "Site index: 35, ref allele frequency: 0.5099337748344371\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6501614639397201\n",
      "Minor allele frequency: 0.4900662251655629\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 35. Done with individual 0/929\n",
      "Slice: 1/1. Site: 35. Done with individual 100/929\n",
      "Slice: 1/1. Site: 35. Done with individual 200/929\n",
      "Slice: 1/1. Site: 35. Done with individual 300/929\n",
      "Slice: 1/1. Site: 35. Done with individual 400/929\n",
      "Slice: 1/1. Site: 35. Done with individual 500/929\n",
      "Slice: 1/1. Site: 35. Done with individual 600/929\n",
      "Slice: 1/1. Site: 35. Done with individual 700/929\n",
      "Slice: 1/1. Site: 35. Done with individual 800/929\n",
      "Slice: 1/1. Site: 35. Done with individual 900/929\n",
      "Site index: 36, non ref allele frequency: 0.4936708860759494\n",
      "Site index: 36, ref allele frequency: 0.5063291139240507\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.25511302475780406\n",
      "Minor allele frequency: 0.4936708860759494\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 36. Done with individual 0/929\n",
      "Slice: 1/1. Site: 36. Done with individual 100/929\n",
      "Slice: 1/1. Site: 36. Done with individual 200/929\n",
      "Slice: 1/1. Site: 36. Done with individual 300/929\n",
      "Slice: 1/1. Site: 36. Done with individual 400/929\n",
      "Slice: 1/1. Site: 36. Done with individual 500/929\n",
      "Slice: 1/1. Site: 36. Done with individual 600/929\n",
      "Slice: 1/1. Site: 36. Done with individual 700/929\n",
      "Slice: 1/1. Site: 36. Done with individual 800/929\n",
      "Slice: 1/1. Site: 36. Done with individual 900/929\n",
      "Site index: 37, non ref allele frequency: 0.5072463768115942\n",
      "Site index: 37, ref allele frequency: 0.4927536231884058\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.14854682454251883\n",
      "Minor allele frequency: 0.4927536231884058\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 37. Done with individual 0/929\n",
      "Slice: 1/1. Site: 37. Done with individual 100/929\n",
      "Slice: 1/1. Site: 37. Done with individual 200/929\n",
      "Slice: 1/1. Site: 37. Done with individual 300/929\n",
      "Slice: 1/1. Site: 37. Done with individual 400/929\n",
      "Slice: 1/1. Site: 37. Done with individual 500/929\n",
      "Slice: 1/1. Site: 37. Done with individual 600/929\n",
      "Slice: 1/1. Site: 37. Done with individual 700/929\n",
      "Slice: 1/1. Site: 37. Done with individual 800/929\n",
      "Slice: 1/1. Site: 37. Done with individual 900/929\n",
      "Site index: 38, non ref allele frequency: 0.5092592592592593\n",
      "Site index: 38, ref allele frequency: 0.49074074074074076\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.697524219590958\n",
      "Minor allele frequency: 0.49074074074074076\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 38. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 38. Done with individual 100/929\n",
      "Slice: 1/1. Site: 38. Done with individual 200/929\n",
      "Slice: 1/1. Site: 38. Done with individual 300/929\n",
      "Slice: 1/1. Site: 38. Done with individual 400/929\n",
      "Slice: 1/1. Site: 38. Done with individual 500/929\n",
      "Slice: 1/1. Site: 38. Done with individual 600/929\n",
      "Slice: 1/1. Site: 38. Done with individual 700/929\n",
      "Slice: 1/1. Site: 38. Done with individual 800/929\n",
      "Slice: 1/1. Site: 38. Done with individual 900/929\n",
      "Site index: 39, non ref allele frequency: 0.5059523809523809\n",
      "Site index: 39, ref allele frequency: 0.49404761904761907\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.36167922497308935\n",
      "Minor allele frequency: 0.49404761904761907\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 39. Done with individual 0/929\n",
      "Slice: 1/1. Site: 39. Done with individual 100/929\n",
      "Slice: 1/1. Site: 39. Done with individual 200/929\n",
      "Slice: 1/1. Site: 39. Done with individual 300/929\n",
      "Slice: 1/1. Site: 39. Done with individual 400/929\n",
      "Slice: 1/1. Site: 39. Done with individual 500/929\n",
      "Slice: 1/1. Site: 39. Done with individual 600/929\n",
      "Slice: 1/1. Site: 39. Done with individual 700/929\n",
      "Slice: 1/1. Site: 39. Done with individual 800/929\n",
      "Slice: 1/1. Site: 39. Done with individual 900/929\n",
      "Site index: 40, non ref allele frequency: 0.49083333333333334\n",
      "Site index: 40, ref allele frequency: 0.5091666666666667\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6458557588805167\n",
      "Minor allele frequency: 0.49083333333333334\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 40. Done with individual 0/929\n",
      "Slice: 1/1. Site: 40. Done with individual 100/929\n",
      "Slice: 1/1. Site: 40. Done with individual 200/929\n",
      "Slice: 1/1. Site: 40. Done with individual 300/929\n",
      "Slice: 1/1. Site: 40. Done with individual 400/929\n",
      "Slice: 1/1. Site: 40. Done with individual 500/929\n",
      "Slice: 1/1. Site: 40. Done with individual 600/929\n",
      "Slice: 1/1. Site: 40. Done with individual 700/929\n",
      "Slice: 1/1. Site: 40. Done with individual 800/929\n",
      "Slice: 1/1. Site: 40. Done with individual 900/929\n",
      "Site index: 41, non ref allele frequency: 0.49078341013824883\n",
      "Site index: 41, ref allele frequency: 0.5092165898617511\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4671689989235737\n",
      "Minor allele frequency: 0.49078341013824883\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 41. Done with individual 0/929\n",
      "Slice: 1/1. Site: 41. Done with individual 100/929\n",
      "Slice: 1/1. Site: 41. Done with individual 200/929\n",
      "Slice: 1/1. Site: 41. Done with individual 300/929\n",
      "Slice: 1/1. Site: 41. Done with individual 400/929\n",
      "Slice: 1/1. Site: 41. Done with individual 500/929\n",
      "Slice: 1/1. Site: 41. Done with individual 600/929\n",
      "Slice: 1/1. Site: 41. Done with individual 700/929\n",
      "Slice: 1/1. Site: 41. Done with individual 800/929\n",
      "Slice: 1/1. Site: 41. Done with individual 900/929\n",
      "Site index: 42, non ref allele frequency: 0.4924812030075188\n",
      "Site index: 42, ref allele frequency: 0.5075187969924813\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7158234660925726\n",
      "Minor allele frequency: 0.4924812030075188\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 42. Done with individual 0/929\n",
      "Slice: 1/1. Site: 42. Done with individual 100/929\n",
      "Slice: 1/1. Site: 42. Done with individual 200/929\n",
      "Slice: 1/1. Site: 42. Done with individual 300/929\n",
      "Slice: 1/1. Site: 42. Done with individual 400/929\n",
      "Slice: 1/1. Site: 42. Done with individual 500/929\n",
      "Slice: 1/1. Site: 42. Done with individual 600/929\n",
      "Slice: 1/1. Site: 42. Done with individual 700/929\n",
      "Slice: 1/1. Site: 42. Done with individual 800/929\n",
      "Slice: 1/1. Site: 42. Done with individual 900/929\n",
      "Site index: 43, non ref allele frequency: 0.49250374812593706\n",
      "Site index: 43, ref allele frequency: 0.507496251874063\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7179763186221744\n",
      "Minor allele frequency: 0.49250374812593706\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 43. Done with individual 0/929\n",
      "Slice: 1/1. Site: 43. Done with individual 100/929\n",
      "Slice: 1/1. Site: 43. Done with individual 200/929\n",
      "Slice: 1/1. Site: 43. Done with individual 300/929\n",
      "Slice: 1/1. Site: 43. Done with individual 400/929\n",
      "Slice: 1/1. Site: 43. Done with individual 500/929\n",
      "Slice: 1/1. Site: 43. Done with individual 600/929\n",
      "Slice: 1/1. Site: 43. Done with individual 700/929\n",
      "Slice: 1/1. Site: 43. Done with individual 800/929\n",
      "Slice: 1/1. Site: 43. Done with individual 900/929\n",
      "Site index: 44, non ref allele frequency: 0.5043478260869565\n",
      "Site index: 44, ref allele frequency: 0.4956521739130435\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6189451022604952\n",
      "Minor allele frequency: 0.4956521739130435\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 44. Done with individual 0/929\n",
      "Slice: 1/1. Site: 44. Done with individual 100/929\n",
      "Slice: 1/1. Site: 44. Done with individual 200/929\n",
      "Slice: 1/1. Site: 44. Done with individual 300/929\n",
      "Slice: 1/1. Site: 44. Done with individual 400/929\n",
      "Slice: 1/1. Site: 44. Done with individual 500/929\n",
      "Slice: 1/1. Site: 44. Done with individual 600/929\n",
      "Slice: 1/1. Site: 44. Done with individual 700/929\n",
      "Slice: 1/1. Site: 44. Done with individual 800/929\n",
      "Slice: 1/1. Site: 44. Done with individual 900/929\n",
      "Site index: 45, non ref allele frequency: 0.4924670433145009\n",
      "Site index: 45, ref allele frequency: 0.507532956685499\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5715823466092572\n",
      "Minor allele frequency: 0.4924670433145009\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 45. Done with individual 0/929\n",
      "Slice: 1/1. Site: 45. Done with individual 100/929\n",
      "Slice: 1/1. Site: 45. Done with individual 200/929\n",
      "Slice: 1/1. Site: 45. Done with individual 300/929\n",
      "Slice: 1/1. Site: 45. Done with individual 400/929\n",
      "Slice: 1/1. Site: 45. Done with individual 500/929\n",
      "Slice: 1/1. Site: 45. Done with individual 600/929\n",
      "Slice: 1/1. Site: 45. Done with individual 700/929\n",
      "Slice: 1/1. Site: 45. Done with individual 800/929\n",
      "Slice: 1/1. Site: 45. Done with individual 900/929\n",
      "Site index: 46, non ref allele frequency: 0.5027624309392266\n",
      "Site index: 46, ref allele frequency: 0.4972375690607735\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5844994617868676\n",
      "Minor allele frequency: 0.4972375690607735\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 46. Done with individual 0/929\n",
      "Slice: 1/1. Site: 46. Done with individual 100/929\n",
      "Slice: 1/1. Site: 46. Done with individual 200/929\n",
      "Slice: 1/1. Site: 46. Done with individual 300/929\n",
      "Slice: 1/1. Site: 46. Done with individual 400/929\n",
      "Slice: 1/1. Site: 46. Done with individual 500/929\n",
      "Slice: 1/1. Site: 46. Done with individual 600/929\n",
      "Slice: 1/1. Site: 46. Done with individual 700/929\n",
      "Slice: 1/1. Site: 46. Done with individual 800/929\n",
      "Slice: 1/1. Site: 46. Done with individual 900/929\n",
      "Site index: 47, non ref allele frequency: 0.5\n",
      "Site index: 47, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.581270182992465\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 47. Done with individual 0/929\n",
      "Slice: 1/1. Site: 47. Done with individual 100/929\n",
      "Slice: 1/1. Site: 47. Done with individual 200/929\n",
      "Slice: 1/1. Site: 47. Done with individual 300/929\n",
      "Slice: 1/1. Site: 47. Done with individual 400/929\n",
      "Slice: 1/1. Site: 47. Done with individual 500/929\n",
      "Slice: 1/1. Site: 47. Done with individual 600/929\n",
      "Slice: 1/1. Site: 47. Done with individual 700/929\n",
      "Slice: 1/1. Site: 47. Done with individual 800/929\n",
      "Slice: 1/1. Site: 47. Done with individual 900/929\n",
      "Site index: 48, non ref allele frequency: 0.500788643533123\n",
      "Site index: 48, ref allele frequency: 0.499211356466877\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.682454251883746\n",
      "Minor allele frequency: 0.499211356466877\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 48. Done with individual 0/929\n",
      "Slice: 1/1. Site: 48. Done with individual 100/929\n",
      "Slice: 1/1. Site: 48. Done with individual 200/929\n",
      "Slice: 1/1. Site: 48. Done with individual 300/929\n",
      "Slice: 1/1. Site: 48. Done with individual 400/929\n",
      "Slice: 1/1. Site: 48. Done with individual 500/929\n",
      "Slice: 1/1. Site: 48. Done with individual 600/929\n",
      "Slice: 1/1. Site: 48. Done with individual 700/929\n",
      "Slice: 1/1. Site: 48. Done with individual 800/929\n",
      "Slice: 1/1. Site: 48. Done with individual 900/929\n",
      "Site index: 49, non ref allele frequency: 0.508\n",
      "Site index: 49, ref allele frequency: 0.492\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6727664155005382\n",
      "Minor allele frequency: 0.492\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 49. Done with individual 0/929\n",
      "Slice: 1/1. Site: 49. Done with individual 100/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 49. Done with individual 200/929\n",
      "Slice: 1/1. Site: 49. Done with individual 300/929\n",
      "Slice: 1/1. Site: 49. Done with individual 400/929\n",
      "Slice: 1/1. Site: 49. Done with individual 500/929\n",
      "Slice: 1/1. Site: 49. Done with individual 600/929\n",
      "Slice: 1/1. Site: 49. Done with individual 700/929\n",
      "Slice: 1/1. Site: 49. Done with individual 800/929\n",
      "Slice: 1/1. Site: 49. Done with individual 900/929\n",
      "Site index: 50, non ref allele frequency: 0.49700598802395207\n",
      "Site index: 50, ref allele frequency: 0.5029940119760479\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7190527448869752\n",
      "Minor allele frequency: 0.49700598802395207\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 50. Done with individual 0/929\n",
      "Slice: 1/1. Site: 50. Done with individual 100/929\n",
      "Slice: 1/1. Site: 50. Done with individual 200/929\n",
      "Slice: 1/1. Site: 50. Done with individual 300/929\n",
      "Slice: 1/1. Site: 50. Done with individual 400/929\n",
      "Slice: 1/1. Site: 50. Done with individual 500/929\n",
      "Slice: 1/1. Site: 50. Done with individual 600/929\n",
      "Slice: 1/1. Site: 50. Done with individual 700/929\n",
      "Slice: 1/1. Site: 50. Done with individual 800/929\n",
      "Slice: 1/1. Site: 50. Done with individual 900/929\n",
      "Site index: 51, non ref allele frequency: 0.49327731092436977\n",
      "Site index: 51, ref allele frequency: 0.5067226890756302\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6404736275565124\n",
      "Minor allele frequency: 0.49327731092436977\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 51. Done with individual 0/929\n",
      "Slice: 1/1. Site: 51. Done with individual 100/929\n",
      "Slice: 1/1. Site: 51. Done with individual 200/929\n",
      "Slice: 1/1. Site: 51. Done with individual 300/929\n",
      "Slice: 1/1. Site: 51. Done with individual 400/929\n",
      "Slice: 1/1. Site: 51. Done with individual 500/929\n",
      "Slice: 1/1. Site: 51. Done with individual 600/929\n",
      "Slice: 1/1. Site: 51. Done with individual 700/929\n",
      "Slice: 1/1. Site: 51. Done with individual 800/929\n",
      "Slice: 1/1. Site: 51. Done with individual 900/929\n",
      "Site index: 52, non ref allele frequency: 0.5050335570469798\n",
      "Site index: 52, ref allele frequency: 0.4949664429530201\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6415500538213132\n",
      "Minor allele frequency: 0.4949664429530201\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 52. Done with individual 0/929\n",
      "Slice: 1/1. Site: 52. Done with individual 100/929\n",
      "Slice: 1/1. Site: 52. Done with individual 200/929\n",
      "Slice: 1/1. Site: 52. Done with individual 300/929\n",
      "Slice: 1/1. Site: 52. Done with individual 400/929\n",
      "Slice: 1/1. Site: 52. Done with individual 500/929\n",
      "Slice: 1/1. Site: 52. Done with individual 600/929\n",
      "Slice: 1/1. Site: 52. Done with individual 700/929\n",
      "Slice: 1/1. Site: 52. Done with individual 800/929\n",
      "Slice: 1/1. Site: 52. Done with individual 900/929\n",
      "Site index: 53, non ref allele frequency: 0.4913644214162349\n",
      "Site index: 53, ref allele frequency: 0.5086355785837651\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6232508073196986\n",
      "Minor allele frequency: 0.4913644214162349\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 53. Done with individual 0/929\n",
      "Slice: 1/1. Site: 53. Done with individual 100/929\n",
      "Slice: 1/1. Site: 53. Done with individual 200/929\n",
      "Slice: 1/1. Site: 53. Done with individual 300/929\n",
      "Slice: 1/1. Site: 53. Done with individual 400/929\n",
      "Slice: 1/1. Site: 53. Done with individual 500/929\n",
      "Slice: 1/1. Site: 53. Done with individual 600/929\n",
      "Slice: 1/1. Site: 53. Done with individual 700/929\n",
      "Slice: 1/1. Site: 53. Done with individual 800/929\n",
      "Slice: 1/1. Site: 53. Done with individual 900/929\n",
      "Site index: 54, non ref allele frequency: 0.5098199672667758\n",
      "Site index: 54, ref allele frequency: 0.49018003273322425\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6576964477933261\n",
      "Minor allele frequency: 0.49018003273322425\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 54. Done with individual 0/929\n",
      "Slice: 1/1. Site: 54. Done with individual 100/929\n",
      "Slice: 1/1. Site: 54. Done with individual 200/929\n",
      "Slice: 1/1. Site: 54. Done with individual 300/929\n",
      "Slice: 1/1. Site: 54. Done with individual 400/929\n",
      "Slice: 1/1. Site: 54. Done with individual 500/929\n",
      "Slice: 1/1. Site: 54. Done with individual 600/929\n",
      "Slice: 1/1. Site: 54. Done with individual 700/929\n",
      "Slice: 1/1. Site: 54. Done with individual 800/929\n",
      "Slice: 1/1. Site: 54. Done with individual 900/929\n",
      "Site index: 55, non ref allele frequency: 0.4942528735632184\n",
      "Site index: 55, ref allele frequency: 0.5057471264367817\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.18729817007534985\n",
      "Minor allele frequency: 0.4942528735632184\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 55. Done with individual 0/929\n",
      "Slice: 1/1. Site: 55. Done with individual 100/929\n",
      "Slice: 1/1. Site: 55. Done with individual 200/929\n",
      "Slice: 1/1. Site: 55. Done with individual 300/929\n",
      "Slice: 1/1. Site: 55. Done with individual 400/929\n",
      "Slice: 1/1. Site: 55. Done with individual 500/929\n",
      "Slice: 1/1. Site: 55. Done with individual 600/929\n",
      "Slice: 1/1. Site: 55. Done with individual 700/929\n",
      "Slice: 1/1. Site: 55. Done with individual 800/929\n",
      "Slice: 1/1. Site: 55. Done with individual 900/929\n",
      "Site index: 56, non ref allele frequency: 0.49783549783549785\n",
      "Site index: 56, ref allele frequency: 0.5021645021645021\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.24865446716899892\n",
      "Minor allele frequency: 0.49783549783549785\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 56. Done with individual 0/929\n",
      "Slice: 1/1. Site: 56. Done with individual 100/929\n",
      "Slice: 1/1. Site: 56. Done with individual 200/929\n",
      "Slice: 1/1. Site: 56. Done with individual 300/929\n",
      "Slice: 1/1. Site: 56. Done with individual 400/929\n",
      "Slice: 1/1. Site: 56. Done with individual 500/929\n",
      "Slice: 1/1. Site: 56. Done with individual 600/929\n",
      "Slice: 1/1. Site: 56. Done with individual 700/929\n",
      "Slice: 1/1. Site: 56. Done with individual 800/929\n",
      "Slice: 1/1. Site: 56. Done with individual 900/929\n",
      "Site index: 57, non ref allele frequency: 0.5056910569105691\n",
      "Site index: 57, ref allele frequency: 0.4943089430894309\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6620021528525296\n",
      "Minor allele frequency: 0.4943089430894309\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 57. Done with individual 0/929\n",
      "Slice: 1/1. Site: 57. Done with individual 100/929\n",
      "Slice: 1/1. Site: 57. Done with individual 200/929\n",
      "Slice: 1/1. Site: 57. Done with individual 300/929\n",
      "Slice: 1/1. Site: 57. Done with individual 400/929\n",
      "Slice: 1/1. Site: 57. Done with individual 500/929\n",
      "Slice: 1/1. Site: 57. Done with individual 600/929\n",
      "Slice: 1/1. Site: 57. Done with individual 700/929\n",
      "Slice: 1/1. Site: 57. Done with individual 800/929\n",
      "Slice: 1/1. Site: 57. Done with individual 900/929\n",
      "Site index: 58, non ref allele frequency: 0.4916387959866221\n",
      "Site index: 58, ref allele frequency: 0.5083612040133779\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3218514531754575\n",
      "Minor allele frequency: 0.4916387959866221\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 58. Done with individual 0/929\n",
      "Slice: 1/1. Site: 58. Done with individual 100/929\n",
      "Slice: 1/1. Site: 58. Done with individual 200/929\n",
      "Slice: 1/1. Site: 58. Done with individual 300/929\n",
      "Slice: 1/1. Site: 58. Done with individual 400/929\n",
      "Slice: 1/1. Site: 58. Done with individual 500/929\n",
      "Slice: 1/1. Site: 58. Done with individual 600/929\n",
      "Slice: 1/1. Site: 58. Done with individual 700/929\n",
      "Slice: 1/1. Site: 58. Done with individual 800/929\n",
      "Slice: 1/1. Site: 58. Done with individual 900/929\n",
      "Site index: 59, non ref allele frequency: 0.5057660626029654\n",
      "Site index: 59, ref allele frequency: 0.4942339373970346\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6533907427341227\n",
      "Minor allele frequency: 0.4942339373970346\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 59. Done with individual 0/929\n",
      "Slice: 1/1. Site: 59. Done with individual 100/929\n",
      "Slice: 1/1. Site: 59. Done with individual 200/929\n",
      "Slice: 1/1. Site: 59. Done with individual 300/929\n",
      "Slice: 1/1. Site: 59. Done with individual 400/929\n",
      "Slice: 1/1. Site: 59. Done with individual 500/929\n",
      "Slice: 1/1. Site: 59. Done with individual 600/929\n",
      "Slice: 1/1. Site: 59. Done with individual 700/929\n",
      "Slice: 1/1. Site: 59. Done with individual 800/929\n",
      "Slice: 1/1. Site: 59. Done with individual 900/929\n",
      "Site index: 60, non ref allele frequency: 0.5068285280728376\n",
      "Site index: 60, ref allele frequency: 0.4931714719271624\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7093649085037675\n",
      "Minor allele frequency: 0.4931714719271624\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 60. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 60. Done with individual 100/929\n",
      "Slice: 1/1. Site: 60. Done with individual 200/929\n",
      "Slice: 1/1. Site: 60. Done with individual 300/929\n",
      "Slice: 1/1. Site: 60. Done with individual 400/929\n",
      "Slice: 1/1. Site: 60. Done with individual 500/929\n",
      "Slice: 1/1. Site: 60. Done with individual 600/929\n",
      "Slice: 1/1. Site: 60. Done with individual 700/929\n",
      "Slice: 1/1. Site: 60. Done with individual 800/929\n",
      "Slice: 1/1. Site: 60. Done with individual 900/929\n",
      "Site index: 61, non ref allele frequency: 0.49117647058823527\n",
      "Site index: 61, ref allele frequency: 0.5088235294117647\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7319698600645855\n",
      "Minor allele frequency: 0.49117647058823527\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 61. Done with individual 0/929\n",
      "Slice: 1/1. Site: 61. Done with individual 100/929\n",
      "Slice: 1/1. Site: 61. Done with individual 200/929\n",
      "Slice: 1/1. Site: 61. Done with individual 300/929\n",
      "Slice: 1/1. Site: 61. Done with individual 400/929\n",
      "Slice: 1/1. Site: 61. Done with individual 500/929\n",
      "Slice: 1/1. Site: 61. Done with individual 600/929\n",
      "Slice: 1/1. Site: 61. Done with individual 700/929\n",
      "Slice: 1/1. Site: 61. Done with individual 800/929\n",
      "Slice: 1/1. Site: 61. Done with individual 900/929\n",
      "Site index: 62, non ref allele frequency: 0.5\n",
      "Site index: 62, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7308934337997847\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 62. Done with individual 0/929\n",
      "Slice: 1/1. Site: 62. Done with individual 100/929\n",
      "Slice: 1/1. Site: 62. Done with individual 200/929\n",
      "Slice: 1/1. Site: 62. Done with individual 300/929\n",
      "Slice: 1/1. Site: 62. Done with individual 400/929\n",
      "Slice: 1/1. Site: 62. Done with individual 500/929\n",
      "Slice: 1/1. Site: 62. Done with individual 600/929\n",
      "Slice: 1/1. Site: 62. Done with individual 700/929\n",
      "Slice: 1/1. Site: 62. Done with individual 800/929\n",
      "Slice: 1/1. Site: 62. Done with individual 900/929\n",
      "Site index: 63, non ref allele frequency: 0.5080775444264943\n",
      "Site index: 63, ref allele frequency: 0.49192245557350567\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6663078579117331\n",
      "Minor allele frequency: 0.49192245557350567\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 63. Done with individual 0/929\n",
      "Slice: 1/1. Site: 63. Done with individual 100/929\n",
      "Slice: 1/1. Site: 63. Done with individual 200/929\n",
      "Slice: 1/1. Site: 63. Done with individual 300/929\n",
      "Slice: 1/1. Site: 63. Done with individual 400/929\n",
      "Slice: 1/1. Site: 63. Done with individual 500/929\n",
      "Slice: 1/1. Site: 63. Done with individual 600/929\n",
      "Slice: 1/1. Site: 63. Done with individual 700/929\n",
      "Slice: 1/1. Site: 63. Done with individual 800/929\n",
      "Slice: 1/1. Site: 63. Done with individual 900/929\n",
      "Site index: 64, non ref allele frequency: 0.49323181049069376\n",
      "Site index: 64, ref allele frequency: 0.5067681895093062\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6361679224973089\n",
      "Minor allele frequency: 0.49323181049069376\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 64. Done with individual 0/929\n",
      "Slice: 1/1. Site: 64. Done with individual 100/929\n",
      "Slice: 1/1. Site: 64. Done with individual 200/929\n",
      "Slice: 1/1. Site: 64. Done with individual 300/929\n",
      "Slice: 1/1. Site: 64. Done with individual 400/929\n",
      "Slice: 1/1. Site: 64. Done with individual 500/929\n",
      "Slice: 1/1. Site: 64. Done with individual 600/929\n",
      "Slice: 1/1. Site: 64. Done with individual 700/929\n",
      "Slice: 1/1. Site: 64. Done with individual 800/929\n",
      "Slice: 1/1. Site: 64. Done with individual 900/929\n",
      "Site index: 65, non ref allele frequency: 0.49455676516329705\n",
      "Site index: 65, ref allele frequency: 0.505443234836703\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6921420882669537\n",
      "Minor allele frequency: 0.49455676516329705\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 65. Done with individual 0/929\n",
      "Slice: 1/1. Site: 65. Done with individual 100/929\n",
      "Slice: 1/1. Site: 65. Done with individual 200/929\n",
      "Slice: 1/1. Site: 65. Done with individual 300/929\n",
      "Slice: 1/1. Site: 65. Done with individual 400/929\n",
      "Slice: 1/1. Site: 65. Done with individual 500/929\n",
      "Slice: 1/1. Site: 65. Done with individual 600/929\n",
      "Slice: 1/1. Site: 65. Done with individual 700/929\n",
      "Slice: 1/1. Site: 65. Done with individual 800/929\n",
      "Slice: 1/1. Site: 65. Done with individual 900/929\n",
      "Site index: 66, non ref allele frequency: 0.5083932853717026\n",
      "Site index: 66, ref allele frequency: 0.49160671462829736\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4488697524219591\n",
      "Minor allele frequency: 0.49160671462829736\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 66. Done with individual 0/929\n",
      "Slice: 1/1. Site: 66. Done with individual 100/929\n",
      "Slice: 1/1. Site: 66. Done with individual 200/929\n",
      "Slice: 1/1. Site: 66. Done with individual 300/929\n",
      "Slice: 1/1. Site: 66. Done with individual 400/929\n",
      "Slice: 1/1. Site: 66. Done with individual 500/929\n",
      "Slice: 1/1. Site: 66. Done with individual 600/929\n",
      "Slice: 1/1. Site: 66. Done with individual 700/929\n",
      "Slice: 1/1. Site: 66. Done with individual 800/929\n",
      "Slice: 1/1. Site: 66. Done with individual 900/929\n",
      "Site index: 67, non ref allele frequency: 0.49796747967479676\n",
      "Site index: 67, ref allele frequency: 0.5020325203252033\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.26480086114101187\n",
      "Minor allele frequency: 0.49796747967479676\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 67. Done with individual 0/929\n",
      "Slice: 1/1. Site: 67. Done with individual 100/929\n",
      "Slice: 1/1. Site: 67. Done with individual 200/929\n",
      "Slice: 1/1. Site: 67. Done with individual 300/929\n",
      "Slice: 1/1. Site: 67. Done with individual 400/929\n",
      "Slice: 1/1. Site: 67. Done with individual 500/929\n",
      "Slice: 1/1. Site: 67. Done with individual 600/929\n",
      "Slice: 1/1. Site: 67. Done with individual 700/929\n",
      "Slice: 1/1. Site: 67. Done with individual 800/929\n",
      "Slice: 1/1. Site: 67. Done with individual 900/929\n",
      "Site index: 68, non ref allele frequency: 0.49843014128728413\n",
      "Site index: 68, ref allele frequency: 0.5015698587127159\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6856835306781486\n",
      "Minor allele frequency: 0.49843014128728413\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 68. Done with individual 0/929\n",
      "Slice: 1/1. Site: 68. Done with individual 100/929\n",
      "Slice: 1/1. Site: 68. Done with individual 200/929\n",
      "Slice: 1/1. Site: 68. Done with individual 300/929\n",
      "Slice: 1/1. Site: 68. Done with individual 400/929\n",
      "Slice: 1/1. Site: 68. Done with individual 500/929\n",
      "Slice: 1/1. Site: 68. Done with individual 600/929\n",
      "Slice: 1/1. Site: 68. Done with individual 700/929\n",
      "Slice: 1/1. Site: 68. Done with individual 800/929\n",
      "Slice: 1/1. Site: 68. Done with individual 900/929\n",
      "Site index: 69, non ref allele frequency: 0.5068807339449541\n",
      "Site index: 69, ref allele frequency: 0.49311926605504586\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7039827771797632\n",
      "Minor allele frequency: 0.49311926605504586\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 69. Done with individual 0/929\n",
      "Slice: 1/1. Site: 69. Done with individual 100/929\n",
      "Slice: 1/1. Site: 69. Done with individual 200/929\n",
      "Slice: 1/1. Site: 69. Done with individual 300/929\n",
      "Slice: 1/1. Site: 69. Done with individual 400/929\n",
      "Slice: 1/1. Site: 69. Done with individual 500/929\n",
      "Slice: 1/1. Site: 69. Done with individual 600/929\n",
      "Slice: 1/1. Site: 69. Done with individual 700/929\n",
      "Slice: 1/1. Site: 69. Done with individual 800/929\n",
      "Slice: 1/1. Site: 69. Done with individual 900/929\n",
      "Site index: 70, non ref allele frequency: 0.5038167938931297\n",
      "Site index: 70, ref allele frequency: 0.4961832061068702\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7050592034445641\n",
      "Minor allele frequency: 0.4961832061068702\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 70. Done with individual 0/929\n",
      "Slice: 1/1. Site: 70. Done with individual 100/929\n",
      "Slice: 1/1. Site: 70. Done with individual 200/929\n",
      "Slice: 1/1. Site: 70. Done with individual 300/929\n",
      "Slice: 1/1. Site: 70. Done with individual 400/929\n",
      "Slice: 1/1. Site: 70. Done with individual 500/929\n",
      "Slice: 1/1. Site: 70. Done with individual 600/929\n",
      "Slice: 1/1. Site: 70. Done with individual 700/929\n",
      "Slice: 1/1. Site: 70. Done with individual 800/929\n",
      "Slice: 1/1. Site: 70. Done with individual 900/929\n",
      "Site index: 71, non ref allele frequency: 0.506872852233677\n",
      "Site index: 71, ref allele frequency: 0.49312714776632305\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3132400430570506\n",
      "Minor allele frequency: 0.49312714776632305\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 71. Done with individual 0/929\n",
      "Slice: 1/1. Site: 71. Done with individual 100/929\n",
      "Slice: 1/1. Site: 71. Done with individual 200/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/1. Site: 71. Done with individual 300/929\n",
      "Slice: 1/1. Site: 71. Done with individual 400/929\n",
      "Slice: 1/1. Site: 71. Done with individual 500/929\n",
      "Slice: 1/1. Site: 71. Done with individual 600/929\n",
      "Slice: 1/1. Site: 71. Done with individual 700/929\n",
      "Slice: 1/1. Site: 71. Done with individual 800/929\n",
      "Slice: 1/1. Site: 71. Done with individual 900/929\n",
      "Site index: 72, non ref allele frequency: 0.491869918699187\n",
      "Site index: 72, ref allele frequency: 0.508130081300813\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5296017222820237\n",
      "Minor allele frequency: 0.491869918699187\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 72. Done with individual 0/929\n",
      "Slice: 1/1. Site: 72. Done with individual 100/929\n",
      "Slice: 1/1. Site: 72. Done with individual 200/929\n",
      "Slice: 1/1. Site: 72. Done with individual 300/929\n",
      "Slice: 1/1. Site: 72. Done with individual 400/929\n",
      "Slice: 1/1. Site: 72. Done with individual 500/929\n",
      "Slice: 1/1. Site: 72. Done with individual 600/929\n",
      "Slice: 1/1. Site: 72. Done with individual 700/929\n",
      "Slice: 1/1. Site: 72. Done with individual 800/929\n",
      "Slice: 1/1. Site: 72. Done with individual 900/929\n",
      "Site index: 73, non ref allele frequency: 0.5032679738562091\n",
      "Site index: 73, ref allele frequency: 0.49673202614379086\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4940796555435953\n",
      "Minor allele frequency: 0.49673202614379086\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 73. Done with individual 0/929\n",
      "Slice: 1/1. Site: 73. Done with individual 100/929\n",
      "Slice: 1/1. Site: 73. Done with individual 200/929\n",
      "Slice: 1/1. Site: 73. Done with individual 300/929\n",
      "Slice: 1/1. Site: 73. Done with individual 400/929\n",
      "Slice: 1/1. Site: 73. Done with individual 500/929\n",
      "Slice: 1/1. Site: 73. Done with individual 600/929\n",
      "Slice: 1/1. Site: 73. Done with individual 700/929\n",
      "Slice: 1/1. Site: 73. Done with individual 800/929\n",
      "Slice: 1/1. Site: 73. Done with individual 900/929\n",
      "Site index: 74, non ref allele frequency: 0.4954682779456193\n",
      "Site index: 74, ref allele frequency: 0.5045317220543807\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.71259418729817\n",
      "Minor allele frequency: 0.4954682779456193\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 74. Done with individual 0/929\n",
      "Slice: 1/1. Site: 74. Done with individual 100/929\n",
      "Slice: 1/1. Site: 74. Done with individual 200/929\n",
      "Slice: 1/1. Site: 74. Done with individual 300/929\n",
      "Slice: 1/1. Site: 74. Done with individual 400/929\n",
      "Slice: 1/1. Site: 74. Done with individual 500/929\n",
      "Slice: 1/1. Site: 74. Done with individual 600/929\n",
      "Slice: 1/1. Site: 74. Done with individual 700/929\n",
      "Slice: 1/1. Site: 74. Done with individual 800/929\n",
      "Slice: 1/1. Site: 74. Done with individual 900/929\n",
      "Site index: 75, non ref allele frequency: 0.5019342359767892\n",
      "Site index: 75, ref allele frequency: 0.49806576402321084\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5565123789020452\n",
      "Minor allele frequency: 0.49806576402321084\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 75. Done with individual 0/929\n",
      "Slice: 1/1. Site: 75. Done with individual 100/929\n",
      "Slice: 1/1. Site: 75. Done with individual 200/929\n",
      "Slice: 1/1. Site: 75. Done with individual 300/929\n",
      "Slice: 1/1. Site: 75. Done with individual 400/929\n",
      "Slice: 1/1. Site: 75. Done with individual 500/929\n",
      "Slice: 1/1. Site: 75. Done with individual 600/929\n",
      "Slice: 1/1. Site: 75. Done with individual 700/929\n",
      "Slice: 1/1. Site: 75. Done with individual 800/929\n",
      "Slice: 1/1. Site: 75. Done with individual 900/929\n",
      "Site index: 76, non ref allele frequency: 0.4972477064220184\n",
      "Site index: 76, ref allele frequency: 0.5027522935779817\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5866523143164694\n",
      "Minor allele frequency: 0.4972477064220184\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 76. Done with individual 0/929\n",
      "Slice: 1/1. Site: 76. Done with individual 100/929\n",
      "Slice: 1/1. Site: 76. Done with individual 200/929\n",
      "Slice: 1/1. Site: 76. Done with individual 300/929\n",
      "Slice: 1/1. Site: 76. Done with individual 400/929\n",
      "Slice: 1/1. Site: 76. Done with individual 500/929\n",
      "Slice: 1/1. Site: 76. Done with individual 600/929\n",
      "Slice: 1/1. Site: 76. Done with individual 700/929\n",
      "Slice: 1/1. Site: 76. Done with individual 800/929\n",
      "Slice: 1/1. Site: 76. Done with individual 900/929\n",
      "Site index: 77, non ref allele frequency: 0.5047619047619047\n",
      "Site index: 77, ref allele frequency: 0.49523809523809526\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5651237890204521\n",
      "Minor allele frequency: 0.49523809523809526\n",
      "Passed guardrails\n",
      "Slice: 1/1. Site: 77. Done with individual 0/929\n",
      "Slice: 1/1. Site: 77. Done with individual 100/929\n",
      "Slice: 1/1. Site: 77. Done with individual 200/929\n",
      "Slice: 1/1. Site: 77. Done with individual 300/929\n",
      "Slice: 1/1. Site: 77. Done with individual 400/929\n",
      "Slice: 1/1. Site: 77. Done with individual 500/929\n",
      "Slice: 1/1. Site: 77. Done with individual 600/929\n",
      "Slice: 1/1. Site: 77. Done with individual 700/929\n",
      "Slice: 1/1. Site: 77. Done with individual 800/929\n",
      "Slice: 1/1. Site: 77. Done with individual 900/929\n",
      "chr_name is chr9\n",
      "output_dir is C:\\Data\\HUJI\\hgdp\\mock/chr9/\n",
      "Total num of sites: 178\n",
      "Desired size of slice: 50\n",
      "Number of slices: 4\n",
      "Size of last slice: 28\n",
      "chr9: slice 0/3\n",
      "\n",
      "#############\n",
      "\n",
      "Slice index: 0. Slice size: 50 \n",
      "Output files: C:\\Data\\HUJI\\hgdp\\mock/chr9/count_dist_slice_0.tsv C:\\Data\\HUJI\\hgdp\\mock/chr9/sites_ids_slice_0.tsv\n",
      "Site index: 0, non ref allele frequency: 0.5\n",
      "Site index: 0, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4951560818083961\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 0. Done with individual 0/929\n",
      "Slice: 0/3. Site: 0. Done with individual 100/929\n",
      "Slice: 0/3. Site: 0. Done with individual 200/929\n",
      "Slice: 0/3. Site: 0. Done with individual 300/929\n",
      "Slice: 0/3. Site: 0. Done with individual 400/929\n",
      "Slice: 0/3. Site: 0. Done with individual 500/929\n",
      "Slice: 0/3. Site: 0. Done with individual 600/929\n",
      "Slice: 0/3. Site: 0. Done with individual 700/929\n",
      "Slice: 0/3. Site: 0. Done with individual 800/929\n",
      "Slice: 0/3. Site: 0. Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.5\n",
      "Site index: 1, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4908503767491927\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 1. Done with individual 0/929\n",
      "Slice: 0/3. Site: 1. Done with individual 100/929\n",
      "Slice: 0/3. Site: 1. Done with individual 200/929\n",
      "Slice: 0/3. Site: 1. Done with individual 300/929\n",
      "Slice: 0/3. Site: 1. Done with individual 400/929\n",
      "Slice: 0/3. Site: 1. Done with individual 500/929\n",
      "Slice: 0/3. Site: 1. Done with individual 600/929\n",
      "Slice: 0/3. Site: 1. Done with individual 700/929\n",
      "Slice: 0/3. Site: 1. Done with individual 800/929\n",
      "Slice: 0/3. Site: 1. Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.49746192893401014\n",
      "Site index: 2, ref allele frequency: 0.5025380710659898\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4241119483315393\n",
      "Minor allele frequency: 0.49746192893401014\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 2. Done with individual 0/929\n",
      "Slice: 0/3. Site: 2. Done with individual 100/929\n",
      "Slice: 0/3. Site: 2. Done with individual 200/929\n",
      "Slice: 0/3. Site: 2. Done with individual 300/929\n",
      "Slice: 0/3. Site: 2. Done with individual 400/929\n",
      "Slice: 0/3. Site: 2. Done with individual 500/929\n",
      "Slice: 0/3. Site: 2. Done with individual 600/929\n",
      "Slice: 0/3. Site: 2. Done with individual 700/929\n",
      "Slice: 0/3. Site: 2. Done with individual 800/929\n",
      "Slice: 0/3. Site: 2. Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.5\n",
      "Site index: 3, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.30893433799784714\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 3. Done with individual 0/929\n",
      "Slice: 0/3. Site: 3. Done with individual 100/929\n",
      "Slice: 0/3. Site: 3. Done with individual 200/929\n",
      "Slice: 0/3. Site: 3. Done with individual 300/929\n",
      "Slice: 0/3. Site: 3. Done with individual 400/929\n",
      "Slice: 0/3. Site: 3. Done with individual 500/929\n",
      "Slice: 0/3. Site: 3. Done with individual 600/929\n",
      "Slice: 0/3. Site: 3. Done with individual 700/929\n",
      "Slice: 0/3. Site: 3. Done with individual 800/929\n",
      "Slice: 0/3. Site: 3. Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.4969040247678019\n",
      "Site index: 4, ref allele frequency: 0.5030959752321982\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34768568353067814\n",
      "Minor allele frequency: 0.4969040247678019\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 4. Done with individual 0/929\n",
      "Slice: 0/3. Site: 4. Done with individual 100/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/3. Site: 4. Done with individual 200/929\n",
      "Slice: 0/3. Site: 4. Done with individual 300/929\n",
      "Slice: 0/3. Site: 4. Done with individual 400/929\n",
      "Slice: 0/3. Site: 4. Done with individual 500/929\n",
      "Slice: 0/3. Site: 4. Done with individual 600/929\n",
      "Slice: 0/3. Site: 4. Done with individual 700/929\n",
      "Slice: 0/3. Site: 4. Done with individual 800/929\n",
      "Slice: 0/3. Site: 4. Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.49534883720930234\n",
      "Site index: 5, ref allele frequency: 0.5046511627906977\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.23143164693218515\n",
      "Minor allele frequency: 0.49534883720930234\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 5. Done with individual 0/929\n",
      "Slice: 0/3. Site: 5. Done with individual 100/929\n",
      "Slice: 0/3. Site: 5. Done with individual 200/929\n",
      "Slice: 0/3. Site: 5. Done with individual 300/929\n",
      "Slice: 0/3. Site: 5. Done with individual 400/929\n",
      "Slice: 0/3. Site: 5. Done with individual 500/929\n",
      "Slice: 0/3. Site: 5. Done with individual 600/929\n",
      "Slice: 0/3. Site: 5. Done with individual 700/929\n",
      "Slice: 0/3. Site: 5. Done with individual 800/929\n",
      "Slice: 0/3. Site: 5. Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.4935897435897436\n",
      "Site index: 6, ref allele frequency: 0.5064102564102564\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.16792249730893433\n",
      "Minor allele frequency: 0.4935897435897436\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 6. Done with individual 0/929\n",
      "Slice: 0/3. Site: 6. Done with individual 100/929\n",
      "Slice: 0/3. Site: 6. Done with individual 200/929\n",
      "Slice: 0/3. Site: 6. Done with individual 300/929\n",
      "Slice: 0/3. Site: 6. Done with individual 400/929\n",
      "Slice: 0/3. Site: 6. Done with individual 500/929\n",
      "Slice: 0/3. Site: 6. Done with individual 600/929\n",
      "Slice: 0/3. Site: 6. Done with individual 700/929\n",
      "Slice: 0/3. Site: 6. Done with individual 800/929\n",
      "Slice: 0/3. Site: 6. Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.4966442953020134\n",
      "Site index: 7, ref allele frequency: 0.5033557046979866\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.48116254036598494\n",
      "Minor allele frequency: 0.4966442953020134\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 7. Done with individual 0/929\n",
      "Slice: 0/3. Site: 7. Done with individual 100/929\n",
      "Slice: 0/3. Site: 7. Done with individual 200/929\n",
      "Slice: 0/3. Site: 7. Done with individual 300/929\n",
      "Slice: 0/3. Site: 7. Done with individual 400/929\n",
      "Slice: 0/3. Site: 7. Done with individual 500/929\n",
      "Slice: 0/3. Site: 7. Done with individual 600/929\n",
      "Slice: 0/3. Site: 7. Done with individual 700/929\n",
      "Slice: 0/3. Site: 7. Done with individual 800/929\n",
      "Slice: 0/3. Site: 7. Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.49230769230769234\n",
      "Site index: 8, ref allele frequency: 0.5076923076923077\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34983853606027987\n",
      "Minor allele frequency: 0.49230769230769234\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 8. Done with individual 0/929\n",
      "Slice: 0/3. Site: 8. Done with individual 100/929\n",
      "Slice: 0/3. Site: 8. Done with individual 200/929\n",
      "Slice: 0/3. Site: 8. Done with individual 300/929\n",
      "Slice: 0/3. Site: 8. Done with individual 400/929\n",
      "Slice: 0/3. Site: 8. Done with individual 500/929\n",
      "Slice: 0/3. Site: 8. Done with individual 600/929\n",
      "Slice: 0/3. Site: 8. Done with individual 700/929\n",
      "Slice: 0/3. Site: 8. Done with individual 800/929\n",
      "Slice: 0/3. Site: 8. Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.494475138121547\n",
      "Site index: 9, ref allele frequency: 0.505524861878453\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.19483315392895587\n",
      "Minor allele frequency: 0.494475138121547\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 9. Done with individual 0/929\n",
      "Slice: 0/3. Site: 9. Done with individual 100/929\n",
      "Slice: 0/3. Site: 9. Done with individual 200/929\n",
      "Slice: 0/3. Site: 9. Done with individual 300/929\n",
      "Slice: 0/3. Site: 9. Done with individual 400/929\n",
      "Slice: 0/3. Site: 9. Done with individual 500/929\n",
      "Slice: 0/3. Site: 9. Done with individual 600/929\n",
      "Slice: 0/3. Site: 9. Done with individual 700/929\n",
      "Slice: 0/3. Site: 9. Done with individual 800/929\n",
      "Slice: 0/3. Site: 9. Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.4948630136986301\n",
      "Site index: 10, ref allele frequency: 0.5051369863013698\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.628632938643703\n",
      "Minor allele frequency: 0.4948630136986301\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 10. Done with individual 0/929\n",
      "Slice: 0/3. Site: 10. Done with individual 100/929\n",
      "Slice: 0/3. Site: 10. Done with individual 200/929\n",
      "Slice: 0/3. Site: 10. Done with individual 300/929\n",
      "Slice: 0/3. Site: 10. Done with individual 400/929\n",
      "Slice: 0/3. Site: 10. Done with individual 500/929\n",
      "Slice: 0/3. Site: 10. Done with individual 600/929\n",
      "Slice: 0/3. Site: 10. Done with individual 700/929\n",
      "Slice: 0/3. Site: 10. Done with individual 800/929\n",
      "Slice: 0/3. Site: 10. Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.4942434210526316\n",
      "Site index: 11, ref allele frequency: 0.5057565789473685\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6544671689989235\n",
      "Minor allele frequency: 0.4942434210526316\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 11. Done with individual 0/929\n",
      "Slice: 0/3. Site: 11. Done with individual 100/929\n",
      "Slice: 0/3. Site: 11. Done with individual 200/929\n",
      "Slice: 0/3. Site: 11. Done with individual 300/929\n",
      "Slice: 0/3. Site: 11. Done with individual 400/929\n",
      "Slice: 0/3. Site: 11. Done with individual 500/929\n",
      "Slice: 0/3. Site: 11. Done with individual 600/929\n",
      "Slice: 0/3. Site: 11. Done with individual 700/929\n",
      "Slice: 0/3. Site: 11. Done with individual 800/929\n",
      "Slice: 0/3. Site: 11. Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.4919786096256685\n",
      "Site index: 12, ref allele frequency: 0.5080213903743316\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.20129171151776104\n",
      "Minor allele frequency: 0.4919786096256685\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 12. Done with individual 0/929\n",
      "Slice: 0/3. Site: 12. Done with individual 100/929\n",
      "Slice: 0/3. Site: 12. Done with individual 200/929\n",
      "Slice: 0/3. Site: 12. Done with individual 300/929\n",
      "Slice: 0/3. Site: 12. Done with individual 400/929\n",
      "Slice: 0/3. Site: 12. Done with individual 500/929\n",
      "Slice: 0/3. Site: 12. Done with individual 600/929\n",
      "Slice: 0/3. Site: 12. Done with individual 700/929\n",
      "Slice: 0/3. Site: 12. Done with individual 800/929\n",
      "Slice: 0/3. Site: 12. Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.4974025974025974\n",
      "Site index: 13, ref allele frequency: 0.5025974025974026\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8288482238966631\n",
      "Minor allele frequency: 0.4974025974025974\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 13. Done with individual 0/929\n",
      "Slice: 0/3. Site: 13. Done with individual 100/929\n",
      "Slice: 0/3. Site: 13. Done with individual 200/929\n",
      "Slice: 0/3. Site: 13. Done with individual 300/929\n",
      "Slice: 0/3. Site: 13. Done with individual 400/929\n",
      "Slice: 0/3. Site: 13. Done with individual 500/929\n",
      "Slice: 0/3. Site: 13. Done with individual 600/929\n",
      "Slice: 0/3. Site: 13. Done with individual 700/929\n",
      "Slice: 0/3. Site: 13. Done with individual 800/929\n",
      "Slice: 0/3. Site: 13. Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.5\n",
      "Site index: 14, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.48869752421959095\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 14. Done with individual 0/929\n",
      "Slice: 0/3. Site: 14. Done with individual 100/929\n",
      "Slice: 0/3. Site: 14. Done with individual 200/929\n",
      "Slice: 0/3. Site: 14. Done with individual 300/929\n",
      "Slice: 0/3. Site: 14. Done with individual 400/929\n",
      "Slice: 0/3. Site: 14. Done with individual 500/929\n",
      "Slice: 0/3. Site: 14. Done with individual 600/929\n",
      "Slice: 0/3. Site: 14. Done with individual 700/929\n",
      "Slice: 0/3. Site: 14. Done with individual 800/929\n",
      "Slice: 0/3. Site: 14. Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.5\n",
      "Site index: 15, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6275565123789021\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 15. Done with individual 0/929\n",
      "Slice: 0/3. Site: 15. Done with individual 100/929\n",
      "Slice: 0/3. Site: 15. Done with individual 200/929\n",
      "Slice: 0/3. Site: 15. Done with individual 300/929\n",
      "Slice: 0/3. Site: 15. Done with individual 400/929\n",
      "Slice: 0/3. Site: 15. Done with individual 500/929\n",
      "Slice: 0/3. Site: 15. Done with individual 600/929\n",
      "Slice: 0/3. Site: 15. Done with individual 700/929\n",
      "Slice: 0/3. Site: 15. Done with individual 800/929\n",
      "Slice: 0/3. Site: 15. Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.5\n",
      "Site index: 16, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5608180839612487\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 16. Done with individual 0/929\n",
      "Slice: 0/3. Site: 16. Done with individual 100/929\n",
      "Slice: 0/3. Site: 16. Done with individual 200/929\n",
      "Slice: 0/3. Site: 16. Done with individual 300/929\n",
      "Slice: 0/3. Site: 16. Done with individual 400/929\n",
      "Slice: 0/3. Site: 16. Done with individual 500/929\n",
      "Slice: 0/3. Site: 16. Done with individual 600/929\n",
      "Slice: 0/3. Site: 16. Done with individual 700/929\n",
      "Slice: 0/3. Site: 16. Done with individual 800/929\n",
      "Slice: 0/3. Site: 16. Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.49280575539568344\n",
      "Site index: 17, ref allele frequency: 0.5071942446043165\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.1496232508073197\n",
      "Minor allele frequency: 0.49280575539568344\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 17. Done with individual 0/929\n",
      "Slice: 0/3. Site: 17. Done with individual 100/929\n",
      "Slice: 0/3. Site: 17. Done with individual 200/929\n",
      "Slice: 0/3. Site: 17. Done with individual 300/929\n",
      "Slice: 0/3. Site: 17. Done with individual 400/929\n",
      "Slice: 0/3. Site: 17. Done with individual 500/929\n",
      "Slice: 0/3. Site: 17. Done with individual 600/929\n",
      "Slice: 0/3. Site: 17. Done with individual 700/929\n",
      "Slice: 0/3. Site: 17. Done with individual 800/929\n",
      "Slice: 0/3. Site: 17. Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.5\n",
      "Site index: 18, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5059203444564048\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 18. Done with individual 0/929\n",
      "Slice: 0/3. Site: 18. Done with individual 100/929\n",
      "Slice: 0/3. Site: 18. Done with individual 200/929\n",
      "Slice: 0/3. Site: 18. Done with individual 300/929\n",
      "Slice: 0/3. Site: 18. Done with individual 400/929\n",
      "Slice: 0/3. Site: 18. Done with individual 500/929\n",
      "Slice: 0/3. Site: 18. Done with individual 600/929\n",
      "Slice: 0/3. Site: 18. Done with individual 700/929\n",
      "Slice: 0/3. Site: 18. Done with individual 800/929\n",
      "Slice: 0/3. Site: 18. Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.5\n",
      "Site index: 19, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.37782561894510225\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 19. Done with individual 0/929\n",
      "Slice: 0/3. Site: 19. Done with individual 100/929\n",
      "Slice: 0/3. Site: 19. Done with individual 200/929\n",
      "Slice: 0/3. Site: 19. Done with individual 300/929\n",
      "Slice: 0/3. Site: 19. Done with individual 400/929\n",
      "Slice: 0/3. Site: 19. Done with individual 500/929\n",
      "Slice: 0/3. Site: 19. Done with individual 600/929\n",
      "Slice: 0/3. Site: 19. Done with individual 700/929\n",
      "Slice: 0/3. Site: 19. Done with individual 800/929\n",
      "Slice: 0/3. Site: 19. Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.5\n",
      "Site index: 20, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.47255113024757806\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 20. Done with individual 0/929\n",
      "Slice: 0/3. Site: 20. Done with individual 100/929\n",
      "Slice: 0/3. Site: 20. Done with individual 200/929\n",
      "Slice: 0/3. Site: 20. Done with individual 300/929\n",
      "Slice: 0/3. Site: 20. Done with individual 400/929\n",
      "Slice: 0/3. Site: 20. Done with individual 500/929\n",
      "Slice: 0/3. Site: 20. Done with individual 600/929\n",
      "Slice: 0/3. Site: 20. Done with individual 700/929\n",
      "Slice: 0/3. Site: 20. Done with individual 800/929\n",
      "Slice: 0/3. Site: 20. Done with individual 900/929\n",
      "Site index: 21, non ref allele frequency: 0.5\n",
      "Site index: 21, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7265877287405813\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 21. Done with individual 0/929\n",
      "Slice: 0/3. Site: 21. Done with individual 100/929\n",
      "Slice: 0/3. Site: 21. Done with individual 200/929\n",
      "Slice: 0/3. Site: 21. Done with individual 300/929\n",
      "Slice: 0/3. Site: 21. Done with individual 400/929\n",
      "Slice: 0/3. Site: 21. Done with individual 500/929\n",
      "Slice: 0/3. Site: 21. Done with individual 600/929\n",
      "Slice: 0/3. Site: 21. Done with individual 700/929\n",
      "Slice: 0/3. Site: 21. Done with individual 800/929\n",
      "Slice: 0/3. Site: 21. Done with individual 900/929\n",
      "Site index: 22, non ref allele frequency: 0.5\n",
      "Site index: 22, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.25834230355220666\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 22. Done with individual 0/929\n",
      "Slice: 0/3. Site: 22. Done with individual 100/929\n",
      "Slice: 0/3. Site: 22. Done with individual 200/929\n",
      "Slice: 0/3. Site: 22. Done with individual 300/929\n",
      "Slice: 0/3. Site: 22. Done with individual 400/929\n",
      "Slice: 0/3. Site: 22. Done with individual 500/929\n",
      "Slice: 0/3. Site: 22. Done with individual 600/929\n",
      "Slice: 0/3. Site: 22. Done with individual 700/929\n",
      "Slice: 0/3. Site: 22. Done with individual 800/929\n",
      "Slice: 0/3. Site: 22. Done with individual 900/929\n",
      "Site index: 23, non ref allele frequency: 0.49099099099099097\n",
      "Site index: 23, ref allele frequency: 0.509009009009009\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.11948331539289558\n",
      "Minor allele frequency: 0.49099099099099097\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 23. Done with individual 0/929\n",
      "Slice: 0/3. Site: 23. Done with individual 100/929\n",
      "Slice: 0/3. Site: 23. Done with individual 200/929\n",
      "Slice: 0/3. Site: 23. Done with individual 300/929\n",
      "Slice: 0/3. Site: 23. Done with individual 400/929\n",
      "Slice: 0/3. Site: 23. Done with individual 500/929\n",
      "Slice: 0/3. Site: 23. Done with individual 600/929\n",
      "Slice: 0/3. Site: 23. Done with individual 700/929\n",
      "Slice: 0/3. Site: 23. Done with individual 800/929\n",
      "Slice: 0/3. Site: 23. Done with individual 900/929\n",
      "Site index: 24, non ref allele frequency: 0.4976190476190476\n",
      "Site index: 24, ref allele frequency: 0.5023809523809524\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22604951560818085\n",
      "Minor allele frequency: 0.4976190476190476\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 24. Done with individual 0/929\n",
      "Slice: 0/3. Site: 24. Done with individual 100/929\n",
      "Slice: 0/3. Site: 24. Done with individual 200/929\n",
      "Slice: 0/3. Site: 24. Done with individual 300/929\n",
      "Slice: 0/3. Site: 24. Done with individual 400/929\n",
      "Slice: 0/3. Site: 24. Done with individual 500/929\n",
      "Slice: 0/3. Site: 24. Done with individual 600/929\n",
      "Slice: 0/3. Site: 24. Done with individual 700/929\n",
      "Slice: 0/3. Site: 24. Done with individual 800/929\n",
      "Slice: 0/3. Site: 24. Done with individual 900/929\n",
      "Site index: 25, non ref allele frequency: 0.5\n",
      "Site index: 25, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8460710441334769\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 25. Done with individual 0/929\n",
      "Slice: 0/3. Site: 25. Done with individual 100/929\n",
      "Slice: 0/3. Site: 25. Done with individual 200/929\n",
      "Slice: 0/3. Site: 25. Done with individual 300/929\n",
      "Slice: 0/3. Site: 25. Done with individual 400/929\n",
      "Slice: 0/3. Site: 25. Done with individual 500/929\n",
      "Slice: 0/3. Site: 25. Done with individual 600/929\n",
      "Slice: 0/3. Site: 25. Done with individual 700/929\n",
      "Slice: 0/3. Site: 25. Done with individual 800/929\n",
      "Slice: 0/3. Site: 25. Done with individual 900/929\n",
      "Site index: 26, non ref allele frequency: 0.5\n",
      "Site index: 26, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8385360602798708\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 26. Done with individual 0/929\n",
      "Slice: 0/3. Site: 26. Done with individual 100/929\n",
      "Slice: 0/3. Site: 26. Done with individual 200/929\n",
      "Slice: 0/3. Site: 26. Done with individual 300/929\n",
      "Slice: 0/3. Site: 26. Done with individual 400/929\n",
      "Slice: 0/3. Site: 26. Done with individual 500/929\n",
      "Slice: 0/3. Site: 26. Done with individual 600/929\n",
      "Slice: 0/3. Site: 26. Done with individual 700/929\n",
      "Slice: 0/3. Site: 26. Done with individual 800/929\n",
      "Slice: 0/3. Site: 26. Done with individual 900/929\n",
      "Site index: 27, non ref allele frequency: 0.49473684210526314\n",
      "Site index: 27, ref allele frequency: 0.5052631578947369\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.40904198062432723\n",
      "Minor allele frequency: 0.49473684210526314\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 27. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/3. Site: 27. Done with individual 100/929\n",
      "Slice: 0/3. Site: 27. Done with individual 200/929\n",
      "Slice: 0/3. Site: 27. Done with individual 300/929\n",
      "Slice: 0/3. Site: 27. Done with individual 400/929\n",
      "Slice: 0/3. Site: 27. Done with individual 500/929\n",
      "Slice: 0/3. Site: 27. Done with individual 600/929\n",
      "Slice: 0/3. Site: 27. Done with individual 700/929\n",
      "Slice: 0/3. Site: 27. Done with individual 800/929\n",
      "Slice: 0/3. Site: 27. Done with individual 900/929\n",
      "Site index: 28, non ref allele frequency: 0.5\n",
      "Site index: 28, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5317545748116254\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 28. Done with individual 0/929\n",
      "Slice: 0/3. Site: 28. Done with individual 100/929\n",
      "Slice: 0/3. Site: 28. Done with individual 200/929\n",
      "Slice: 0/3. Site: 28. Done with individual 300/929\n",
      "Slice: 0/3. Site: 28. Done with individual 400/929\n",
      "Slice: 0/3. Site: 28. Done with individual 500/929\n",
      "Slice: 0/3. Site: 28. Done with individual 600/929\n",
      "Slice: 0/3. Site: 28. Done with individual 700/929\n",
      "Slice: 0/3. Site: 28. Done with individual 800/929\n",
      "Slice: 0/3. Site: 28. Done with individual 900/929\n",
      "Site index: 29, non ref allele frequency: 0.5\n",
      "Site index: 29, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.24865446716899892\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 29. Done with individual 0/929\n",
      "Slice: 0/3. Site: 29. Done with individual 100/929\n",
      "Slice: 0/3. Site: 29. Done with individual 200/929\n",
      "Slice: 0/3. Site: 29. Done with individual 300/929\n",
      "Slice: 0/3. Site: 29. Done with individual 400/929\n",
      "Slice: 0/3. Site: 29. Done with individual 500/929\n",
      "Slice: 0/3. Site: 29. Done with individual 600/929\n",
      "Slice: 0/3. Site: 29. Done with individual 700/929\n",
      "Slice: 0/3. Site: 29. Done with individual 800/929\n",
      "Slice: 0/3. Site: 29. Done with individual 900/929\n",
      "Site index: 30, non ref allele frequency: 0.5\n",
      "Site index: 30, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.11410118406889128\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 30. Done with individual 0/929\n",
      "Slice: 0/3. Site: 30. Done with individual 100/929\n",
      "Slice: 0/3. Site: 30. Done with individual 200/929\n",
      "Slice: 0/3. Site: 30. Done with individual 300/929\n",
      "Slice: 0/3. Site: 30. Done with individual 400/929\n",
      "Slice: 0/3. Site: 30. Done with individual 500/929\n",
      "Slice: 0/3. Site: 30. Done with individual 600/929\n",
      "Slice: 0/3. Site: 30. Done with individual 700/929\n",
      "Slice: 0/3. Site: 30. Done with individual 800/929\n",
      "Slice: 0/3. Site: 30. Done with individual 900/929\n",
      "Site index: 31, non ref allele frequency: 0.49330357142857145\n",
      "Site index: 31, ref allele frequency: 0.5066964285714286\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4822389666307858\n",
      "Minor allele frequency: 0.49330357142857145\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 31. Done with individual 0/929\n",
      "Slice: 0/3. Site: 31. Done with individual 100/929\n",
      "Slice: 0/3. Site: 31. Done with individual 200/929\n",
      "Slice: 0/3. Site: 31. Done with individual 300/929\n",
      "Slice: 0/3. Site: 31. Done with individual 400/929\n",
      "Slice: 0/3. Site: 31. Done with individual 500/929\n",
      "Slice: 0/3. Site: 31. Done with individual 600/929\n",
      "Slice: 0/3. Site: 31. Done with individual 700/929\n",
      "Slice: 0/3. Site: 31. Done with individual 800/929\n",
      "Slice: 0/3. Site: 31. Done with individual 900/929\n",
      "Site index: 32, non ref allele frequency: 0.4931506849315068\n",
      "Site index: 32, ref allele frequency: 0.5068493150684932\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.55005382131324\n",
      "Minor allele frequency: 0.4931506849315068\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 32. Done with individual 0/929\n",
      "Slice: 0/3. Site: 32. Done with individual 100/929\n",
      "Slice: 0/3. Site: 32. Done with individual 200/929\n",
      "Slice: 0/3. Site: 32. Done with individual 300/929\n",
      "Slice: 0/3. Site: 32. Done with individual 400/929\n",
      "Slice: 0/3. Site: 32. Done with individual 500/929\n",
      "Slice: 0/3. Site: 32. Done with individual 600/929\n",
      "Slice: 0/3. Site: 32. Done with individual 700/929\n",
      "Slice: 0/3. Site: 32. Done with individual 800/929\n",
      "Slice: 0/3. Site: 32. Done with individual 900/929\n",
      "Site index: 33, non ref allele frequency: 0.49580712788259956\n",
      "Site index: 33, ref allele frequency: 0.5041928721174004\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5134553283100107\n",
      "Minor allele frequency: 0.49580712788259956\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 33. Done with individual 0/929\n",
      "Slice: 0/3. Site: 33. Done with individual 100/929\n",
      "Slice: 0/3. Site: 33. Done with individual 200/929\n",
      "Slice: 0/3. Site: 33. Done with individual 300/929\n",
      "Slice: 0/3. Site: 33. Done with individual 400/929\n",
      "Slice: 0/3. Site: 33. Done with individual 500/929\n",
      "Slice: 0/3. Site: 33. Done with individual 600/929\n",
      "Slice: 0/3. Site: 33. Done with individual 700/929\n",
      "Slice: 0/3. Site: 33. Done with individual 800/929\n",
      "Slice: 0/3. Site: 33. Done with individual 900/929\n",
      "Site index: 34, non ref allele frequency: 0.506578947368421\n",
      "Site index: 34, ref allele frequency: 0.4934210526315789\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.40904198062432723\n",
      "Minor allele frequency: 0.4934210526315789\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 34. Done with individual 0/929\n",
      "Slice: 0/3. Site: 34. Done with individual 100/929\n",
      "Slice: 0/3. Site: 34. Done with individual 200/929\n",
      "Slice: 0/3. Site: 34. Done with individual 300/929\n",
      "Slice: 0/3. Site: 34. Done with individual 400/929\n",
      "Slice: 0/3. Site: 34. Done with individual 500/929\n",
      "Slice: 0/3. Site: 34. Done with individual 600/929\n",
      "Slice: 0/3. Site: 34. Done with individual 700/929\n",
      "Slice: 0/3. Site: 34. Done with individual 800/929\n",
      "Slice: 0/3. Site: 34. Done with individual 900/929\n",
      "Site index: 35, non ref allele frequency: 0.49012158054711247\n",
      "Site index: 35, ref allele frequency: 0.5098784194528876\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7082884822389667\n",
      "Minor allele frequency: 0.49012158054711247\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 35. Done with individual 0/929\n",
      "Slice: 0/3. Site: 35. Done with individual 100/929\n",
      "Slice: 0/3. Site: 35. Done with individual 200/929\n",
      "Slice: 0/3. Site: 35. Done with individual 300/929\n",
      "Slice: 0/3. Site: 35. Done with individual 400/929\n",
      "Slice: 0/3. Site: 35. Done with individual 500/929\n",
      "Slice: 0/3. Site: 35. Done with individual 600/929\n",
      "Slice: 0/3. Site: 35. Done with individual 700/929\n",
      "Slice: 0/3. Site: 35. Done with individual 800/929\n",
      "Slice: 0/3. Site: 35. Done with individual 900/929\n",
      "Site index: 36, non ref allele frequency: 0.5015948963317385\n",
      "Site index: 36, ref allele frequency: 0.4984051036682616\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6749192680301399\n",
      "Minor allele frequency: 0.4984051036682616\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 36. Done with individual 0/929\n",
      "Slice: 0/3. Site: 36. Done with individual 100/929\n",
      "Slice: 0/3. Site: 36. Done with individual 200/929\n",
      "Slice: 0/3. Site: 36. Done with individual 300/929\n",
      "Slice: 0/3. Site: 36. Done with individual 400/929\n",
      "Slice: 0/3. Site: 36. Done with individual 500/929\n",
      "Slice: 0/3. Site: 36. Done with individual 600/929\n",
      "Slice: 0/3. Site: 36. Done with individual 700/929\n",
      "Slice: 0/3. Site: 36. Done with individual 800/929\n",
      "Slice: 0/3. Site: 36. Done with individual 900/929\n",
      "Site index: 37, non ref allele frequency: 0.5\n",
      "Site index: 37, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7319698600645855\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 37. Done with individual 0/929\n",
      "Slice: 0/3. Site: 37. Done with individual 100/929\n",
      "Slice: 0/3. Site: 37. Done with individual 200/929\n",
      "Slice: 0/3. Site: 37. Done with individual 300/929\n",
      "Slice: 0/3. Site: 37. Done with individual 400/929\n",
      "Slice: 0/3. Site: 37. Done with individual 500/929\n",
      "Slice: 0/3. Site: 37. Done with individual 600/929\n",
      "Slice: 0/3. Site: 37. Done with individual 700/929\n",
      "Slice: 0/3. Site: 37. Done with individual 800/929\n",
      "Slice: 0/3. Site: 37. Done with individual 900/929\n",
      "Site index: 38, non ref allele frequency: 0.5\n",
      "Site index: 38, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3627556512378902\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 38. Done with individual 0/929\n",
      "Slice: 0/3. Site: 38. Done with individual 100/929\n",
      "Slice: 0/3. Site: 38. Done with individual 200/929\n",
      "Slice: 0/3. Site: 38. Done with individual 300/929\n",
      "Slice: 0/3. Site: 38. Done with individual 400/929\n",
      "Slice: 0/3. Site: 38. Done with individual 500/929\n",
      "Slice: 0/3. Site: 38. Done with individual 600/929\n",
      "Slice: 0/3. Site: 38. Done with individual 700/929\n",
      "Slice: 0/3. Site: 38. Done with individual 800/929\n",
      "Slice: 0/3. Site: 38. Done with individual 900/929\n",
      "Site index: 39, non ref allele frequency: 0.4951923076923077\n",
      "Site index: 39, ref allele frequency: 0.5048076923076923\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2238966630785791\n",
      "Minor allele frequency: 0.4951923076923077\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 39. Done with individual 0/929\n",
      "Slice: 0/3. Site: 39. Done with individual 100/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 0/3. Site: 39. Done with individual 200/929\n",
      "Slice: 0/3. Site: 39. Done with individual 300/929\n",
      "Slice: 0/3. Site: 39. Done with individual 400/929\n",
      "Slice: 0/3. Site: 39. Done with individual 500/929\n",
      "Slice: 0/3. Site: 39. Done with individual 600/929\n",
      "Slice: 0/3. Site: 39. Done with individual 700/929\n",
      "Slice: 0/3. Site: 39. Done with individual 800/929\n",
      "Slice: 0/3. Site: 39. Done with individual 900/929\n",
      "Site index: 40, non ref allele frequency: 0.49707602339181284\n",
      "Site index: 40, ref allele frequency: 0.5029239766081871\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.18406889128094725\n",
      "Minor allele frequency: 0.49707602339181284\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 40. Done with individual 0/929\n",
      "Slice: 0/3. Site: 40. Done with individual 100/929\n",
      "Slice: 0/3. Site: 40. Done with individual 200/929\n",
      "Slice: 0/3. Site: 40. Done with individual 300/929\n",
      "Slice: 0/3. Site: 40. Done with individual 400/929\n",
      "Slice: 0/3. Site: 40. Done with individual 500/929\n",
      "Slice: 0/3. Site: 40. Done with individual 600/929\n",
      "Slice: 0/3. Site: 40. Done with individual 700/929\n",
      "Slice: 0/3. Site: 40. Done with individual 800/929\n",
      "Slice: 0/3. Site: 40. Done with individual 900/929\n",
      "Site index: 41, non ref allele frequency: 0.4909090909090909\n",
      "Site index: 41, ref allele frequency: 0.509090909090909\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7104413347685683\n",
      "Minor allele frequency: 0.4909090909090909\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 41. Done with individual 0/929\n",
      "Slice: 0/3. Site: 41. Done with individual 100/929\n",
      "Slice: 0/3. Site: 41. Done with individual 200/929\n",
      "Slice: 0/3. Site: 41. Done with individual 300/929\n",
      "Slice: 0/3. Site: 41. Done with individual 400/929\n",
      "Slice: 0/3. Site: 41. Done with individual 500/929\n",
      "Slice: 0/3. Site: 41. Done with individual 600/929\n",
      "Slice: 0/3. Site: 41. Done with individual 700/929\n",
      "Slice: 0/3. Site: 41. Done with individual 800/929\n",
      "Slice: 0/3. Site: 41. Done with individual 900/929\n",
      "Site index: 42, non ref allele frequency: 0.4985549132947977\n",
      "Site index: 42, ref allele frequency: 0.5014450867052023\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7448869752421959\n",
      "Minor allele frequency: 0.4985549132947977\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 42. Done with individual 0/929\n",
      "Slice: 0/3. Site: 42. Done with individual 100/929\n",
      "Slice: 0/3. Site: 42. Done with individual 200/929\n",
      "Slice: 0/3. Site: 42. Done with individual 300/929\n",
      "Slice: 0/3. Site: 42. Done with individual 400/929\n",
      "Slice: 0/3. Site: 42. Done with individual 500/929\n",
      "Slice: 0/3. Site: 42. Done with individual 600/929\n",
      "Slice: 0/3. Site: 42. Done with individual 700/929\n",
      "Slice: 0/3. Site: 42. Done with individual 800/929\n",
      "Slice: 0/3. Site: 42. Done with individual 900/929\n",
      "Site index: 43, non ref allele frequency: 0.5094339622641509\n",
      "Site index: 43, ref allele frequency: 0.49056603773584906\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5705059203444564\n",
      "Minor allele frequency: 0.49056603773584906\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 43. Done with individual 0/929\n",
      "Slice: 0/3. Site: 43. Done with individual 100/929\n",
      "Slice: 0/3. Site: 43. Done with individual 200/929\n",
      "Slice: 0/3. Site: 43. Done with individual 300/929\n",
      "Slice: 0/3. Site: 43. Done with individual 400/929\n",
      "Slice: 0/3. Site: 43. Done with individual 500/929\n",
      "Slice: 0/3. Site: 43. Done with individual 600/929\n",
      "Slice: 0/3. Site: 43. Done with individual 700/929\n",
      "Slice: 0/3. Site: 43. Done with individual 800/929\n",
      "Slice: 0/3. Site: 43. Done with individual 900/929\n",
      "Site index: 44, non ref allele frequency: 0.49887133182844245\n",
      "Site index: 44, ref allele frequency: 0.5011286681715575\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.47685683530678147\n",
      "Minor allele frequency: 0.49887133182844245\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 44. Done with individual 0/929\n",
      "Slice: 0/3. Site: 44. Done with individual 100/929\n",
      "Slice: 0/3. Site: 44. Done with individual 200/929\n",
      "Slice: 0/3. Site: 44. Done with individual 300/929\n",
      "Slice: 0/3. Site: 44. Done with individual 400/929\n",
      "Slice: 0/3. Site: 44. Done with individual 500/929\n",
      "Slice: 0/3. Site: 44. Done with individual 600/929\n",
      "Slice: 0/3. Site: 44. Done with individual 700/929\n",
      "Slice: 0/3. Site: 44. Done with individual 800/929\n",
      "Slice: 0/3. Site: 44. Done with individual 900/929\n",
      "Site index: 45, non ref allele frequency: 0.4919678714859438\n",
      "Site index: 45, ref allele frequency: 0.5080321285140562\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2680301399354144\n",
      "Minor allele frequency: 0.4919678714859438\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 45. Done with individual 0/929\n",
      "Slice: 0/3. Site: 45. Done with individual 100/929\n",
      "Slice: 0/3. Site: 45. Done with individual 200/929\n",
      "Slice: 0/3. Site: 45. Done with individual 300/929\n",
      "Slice: 0/3. Site: 45. Done with individual 400/929\n",
      "Slice: 0/3. Site: 45. Done with individual 500/929\n",
      "Slice: 0/3. Site: 45. Done with individual 600/929\n",
      "Slice: 0/3. Site: 45. Done with individual 700/929\n",
      "Slice: 0/3. Site: 45. Done with individual 800/929\n",
      "Slice: 0/3. Site: 45. Done with individual 900/929\n",
      "Site index: 46, non ref allele frequency: 0.4929742388758782\n",
      "Site index: 46, ref allele frequency: 0.5070257611241218\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4596340150699677\n",
      "Minor allele frequency: 0.4929742388758782\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 46. Done with individual 0/929\n",
      "Slice: 0/3. Site: 46. Done with individual 100/929\n",
      "Slice: 0/3. Site: 46. Done with individual 200/929\n",
      "Slice: 0/3. Site: 46. Done with individual 300/929\n",
      "Slice: 0/3. Site: 46. Done with individual 400/929\n",
      "Slice: 0/3. Site: 46. Done with individual 500/929\n",
      "Slice: 0/3. Site: 46. Done with individual 600/929\n",
      "Slice: 0/3. Site: 46. Done with individual 700/929\n",
      "Slice: 0/3. Site: 46. Done with individual 800/929\n",
      "Slice: 0/3. Site: 46. Done with individual 900/929\n",
      "Site index: 47, non ref allele frequency: 0.4991103202846975\n",
      "Site index: 47, ref allele frequency: 0.5008896797153025\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.604951560818084\n",
      "Minor allele frequency: 0.4991103202846975\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 47. Done with individual 0/929\n",
      "Slice: 0/3. Site: 47. Done with individual 100/929\n",
      "Slice: 0/3. Site: 47. Done with individual 200/929\n",
      "Slice: 0/3. Site: 47. Done with individual 300/929\n",
      "Slice: 0/3. Site: 47. Done with individual 400/929\n",
      "Slice: 0/3. Site: 47. Done with individual 500/929\n",
      "Slice: 0/3. Site: 47. Done with individual 600/929\n",
      "Slice: 0/3. Site: 47. Done with individual 700/929\n",
      "Slice: 0/3. Site: 47. Done with individual 800/929\n",
      "Slice: 0/3. Site: 47. Done with individual 900/929\n",
      "Site index: 48, non ref allele frequency: 0.5\n",
      "Site index: 48, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.620021528525296\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 48. Done with individual 0/929\n",
      "Slice: 0/3. Site: 48. Done with individual 100/929\n",
      "Slice: 0/3. Site: 48. Done with individual 200/929\n",
      "Slice: 0/3. Site: 48. Done with individual 300/929\n",
      "Slice: 0/3. Site: 48. Done with individual 400/929\n",
      "Slice: 0/3. Site: 48. Done with individual 500/929\n",
      "Slice: 0/3. Site: 48. Done with individual 600/929\n",
      "Slice: 0/3. Site: 48. Done with individual 700/929\n",
      "Slice: 0/3. Site: 48. Done with individual 800/929\n",
      "Slice: 0/3. Site: 48. Done with individual 900/929\n",
      "Site index: 49, non ref allele frequency: 0.5\n",
      "Site index: 49, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9364908503767492\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 0/3. Site: 49. Done with individual 0/929\n",
      "Slice: 0/3. Site: 49. Done with individual 100/929\n",
      "Slice: 0/3. Site: 49. Done with individual 200/929\n",
      "Slice: 0/3. Site: 49. Done with individual 300/929\n",
      "Slice: 0/3. Site: 49. Done with individual 400/929\n",
      "Slice: 0/3. Site: 49. Done with individual 500/929\n",
      "Slice: 0/3. Site: 49. Done with individual 600/929\n",
      "Slice: 0/3. Site: 49. Done with individual 700/929\n",
      "Slice: 0/3. Site: 49. Done with individual 800/929\n",
      "Slice: 0/3. Site: 49. Done with individual 900/929\n",
      "chr9: slice 1/3\n",
      "\n",
      "#############\n",
      "\n",
      "Slice index: 1. Slice size: 50 \n",
      "Output files: C:\\Data\\HUJI\\hgdp\\mock/chr9/count_dist_slice_1.tsv C:\\Data\\HUJI\\hgdp\\mock/chr9/sites_ids_slice_1.tsv\n",
      "Site index: 0, non ref allele frequency: 0.5\n",
      "Site index: 0, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.9117330462863293\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 0. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/3. Site: 0. Done with individual 100/929\n",
      "Slice: 1/3. Site: 0. Done with individual 200/929\n",
      "Slice: 1/3. Site: 0. Done with individual 300/929\n",
      "Slice: 1/3. Site: 0. Done with individual 400/929\n",
      "Slice: 1/3. Site: 0. Done with individual 500/929\n",
      "Slice: 1/3. Site: 0. Done with individual 600/929\n",
      "Slice: 1/3. Site: 0. Done with individual 700/929\n",
      "Slice: 1/3. Site: 0. Done with individual 800/929\n",
      "Slice: 1/3. Site: 0. Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.5\n",
      "Site index: 1, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8708288482238966\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 1. Done with individual 0/929\n",
      "Slice: 1/3. Site: 1. Done with individual 100/929\n",
      "Slice: 1/3. Site: 1. Done with individual 200/929\n",
      "Slice: 1/3. Site: 1. Done with individual 300/929\n",
      "Slice: 1/3. Site: 1. Done with individual 400/929\n",
      "Slice: 1/3. Site: 1. Done with individual 500/929\n",
      "Slice: 1/3. Site: 1. Done with individual 600/929\n",
      "Slice: 1/3. Site: 1. Done with individual 700/929\n",
      "Slice: 1/3. Site: 1. Done with individual 800/929\n",
      "Slice: 1/3. Site: 1. Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.5\n",
      "Site index: 2, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8783638320775027\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 2. Done with individual 0/929\n",
      "Slice: 1/3. Site: 2. Done with individual 100/929\n",
      "Slice: 1/3. Site: 2. Done with individual 200/929\n",
      "Slice: 1/3. Site: 2. Done with individual 300/929\n",
      "Slice: 1/3. Site: 2. Done with individual 400/929\n",
      "Slice: 1/3. Site: 2. Done with individual 500/929\n",
      "Slice: 1/3. Site: 2. Done with individual 600/929\n",
      "Slice: 1/3. Site: 2. Done with individual 700/929\n",
      "Slice: 1/3. Site: 2. Done with individual 800/929\n",
      "Slice: 1/3. Site: 2. Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.49823321554770317\n",
      "Site index: 3, ref allele frequency: 0.5017667844522968\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6092572658772875\n",
      "Minor allele frequency: 0.49823321554770317\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 3. Done with individual 0/929\n",
      "Slice: 1/3. Site: 3. Done with individual 100/929\n",
      "Slice: 1/3. Site: 3. Done with individual 200/929\n",
      "Slice: 1/3. Site: 3. Done with individual 300/929\n",
      "Slice: 1/3. Site: 3. Done with individual 400/929\n",
      "Slice: 1/3. Site: 3. Done with individual 500/929\n",
      "Slice: 1/3. Site: 3. Done with individual 600/929\n",
      "Slice: 1/3. Site: 3. Done with individual 700/929\n",
      "Slice: 1/3. Site: 3. Done with individual 800/929\n",
      "Slice: 1/3. Site: 3. Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.5\n",
      "Site index: 4, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.13885898815931108\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 4. Done with individual 0/929\n",
      "Slice: 1/3. Site: 4. Done with individual 100/929\n",
      "Slice: 1/3. Site: 4. Done with individual 200/929\n",
      "Slice: 1/3. Site: 4. Done with individual 300/929\n",
      "Slice: 1/3. Site: 4. Done with individual 400/929\n",
      "Slice: 1/3. Site: 4. Done with individual 500/929\n",
      "Slice: 1/3. Site: 4. Done with individual 600/929\n",
      "Slice: 1/3. Site: 4. Done with individual 700/929\n",
      "Slice: 1/3. Site: 4. Done with individual 800/929\n",
      "Slice: 1/3. Site: 4. Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.4964622641509434\n",
      "Site index: 5, ref allele frequency: 0.5035377358490566\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4564047362755651\n",
      "Minor allele frequency: 0.4964622641509434\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 5. Done with individual 0/929\n",
      "Slice: 1/3. Site: 5. Done with individual 100/929\n",
      "Slice: 1/3. Site: 5. Done with individual 200/929\n",
      "Slice: 1/3. Site: 5. Done with individual 300/929\n",
      "Slice: 1/3. Site: 5. Done with individual 400/929\n",
      "Slice: 1/3. Site: 5. Done with individual 500/929\n",
      "Slice: 1/3. Site: 5. Done with individual 600/929\n",
      "Slice: 1/3. Site: 5. Done with individual 700/929\n",
      "Slice: 1/3. Site: 5. Done with individual 800/929\n",
      "Slice: 1/3. Site: 5. Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.5\n",
      "Site index: 6, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6921420882669537\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 6. Done with individual 0/929\n",
      "Slice: 1/3. Site: 6. Done with individual 100/929\n",
      "Slice: 1/3. Site: 6. Done with individual 200/929\n",
      "Slice: 1/3. Site: 6. Done with individual 300/929\n",
      "Slice: 1/3. Site: 6. Done with individual 400/929\n",
      "Slice: 1/3. Site: 6. Done with individual 500/929\n",
      "Slice: 1/3. Site: 6. Done with individual 600/929\n",
      "Slice: 1/3. Site: 6. Done with individual 700/929\n",
      "Slice: 1/3. Site: 6. Done with individual 800/929\n",
      "Slice: 1/3. Site: 6. Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.5\n",
      "Site index: 7, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5640473627556513\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 7. Done with individual 0/929\n",
      "Slice: 1/3. Site: 7. Done with individual 100/929\n",
      "Slice: 1/3. Site: 7. Done with individual 200/929\n",
      "Slice: 1/3. Site: 7. Done with individual 300/929\n",
      "Slice: 1/3. Site: 7. Done with individual 400/929\n",
      "Slice: 1/3. Site: 7. Done with individual 500/929\n",
      "Slice: 1/3. Site: 7. Done with individual 600/929\n",
      "Slice: 1/3. Site: 7. Done with individual 700/929\n",
      "Slice: 1/3. Site: 7. Done with individual 800/929\n",
      "Slice: 1/3. Site: 7. Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.502127659574468\n",
      "Site index: 8, ref allele frequency: 0.4978723404255319\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5059203444564048\n",
      "Minor allele frequency: 0.4978723404255319\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 8. Done with individual 0/929\n",
      "Slice: 1/3. Site: 8. Done with individual 100/929\n",
      "Slice: 1/3. Site: 8. Done with individual 200/929\n",
      "Slice: 1/3. Site: 8. Done with individual 300/929\n",
      "Slice: 1/3. Site: 8. Done with individual 400/929\n",
      "Slice: 1/3. Site: 8. Done with individual 500/929\n",
      "Slice: 1/3. Site: 8. Done with individual 600/929\n",
      "Slice: 1/3. Site: 8. Done with individual 700/929\n",
      "Slice: 1/3. Site: 8. Done with individual 800/929\n",
      "Slice: 1/3. Site: 8. Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.49029126213592233\n",
      "Site index: 9, ref allele frequency: 0.5097087378640777\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.1108719052744887\n",
      "Minor allele frequency: 0.49029126213592233\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 9. Done with individual 0/929\n",
      "Slice: 1/3. Site: 9. Done with individual 100/929\n",
      "Slice: 1/3. Site: 9. Done with individual 200/929\n",
      "Slice: 1/3. Site: 9. Done with individual 300/929\n",
      "Slice: 1/3. Site: 9. Done with individual 400/929\n",
      "Slice: 1/3. Site: 9. Done with individual 500/929\n",
      "Slice: 1/3. Site: 9. Done with individual 600/929\n",
      "Slice: 1/3. Site: 9. Done with individual 700/929\n",
      "Slice: 1/3. Site: 9. Done with individual 800/929\n",
      "Slice: 1/3. Site: 9. Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.49917627677100496\n",
      "Site index: 10, ref allele frequency: 0.500823723228995\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6533907427341227\n",
      "Minor allele frequency: 0.49917627677100496\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 10. Done with individual 0/929\n",
      "Slice: 1/3. Site: 10. Done with individual 100/929\n",
      "Slice: 1/3. Site: 10. Done with individual 200/929\n",
      "Slice: 1/3. Site: 10. Done with individual 300/929\n",
      "Slice: 1/3. Site: 10. Done with individual 400/929\n",
      "Slice: 1/3. Site: 10. Done with individual 500/929\n",
      "Slice: 1/3. Site: 10. Done with individual 600/929\n",
      "Slice: 1/3. Site: 10. Done with individual 700/929\n",
      "Slice: 1/3. Site: 10. Done with individual 800/929\n",
      "Slice: 1/3. Site: 10. Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.502851711026616\n",
      "Site index: 11, ref allele frequency: 0.49714828897338403\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.566200215285253\n",
      "Minor allele frequency: 0.49714828897338403\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 11. Done with individual 0/929\n",
      "Slice: 1/3. Site: 11. Done with individual 100/929\n",
      "Slice: 1/3. Site: 11. Done with individual 200/929\n",
      "Slice: 1/3. Site: 11. Done with individual 300/929\n",
      "Slice: 1/3. Site: 11. Done with individual 400/929\n",
      "Slice: 1/3. Site: 11. Done with individual 500/929\n",
      "Slice: 1/3. Site: 11. Done with individual 600/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/3. Site: 11. Done with individual 700/929\n",
      "Slice: 1/3. Site: 11. Done with individual 800/929\n",
      "Slice: 1/3. Site: 11. Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.4957627118644068\n",
      "Site index: 12, ref allele frequency: 0.5042372881355932\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.38105489773950485\n",
      "Minor allele frequency: 0.4957627118644068\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 12. Done with individual 0/929\n",
      "Slice: 1/3. Site: 12. Done with individual 100/929\n",
      "Slice: 1/3. Site: 12. Done with individual 200/929\n",
      "Slice: 1/3. Site: 12. Done with individual 300/929\n",
      "Slice: 1/3. Site: 12. Done with individual 400/929\n",
      "Slice: 1/3. Site: 12. Done with individual 500/929\n",
      "Slice: 1/3. Site: 12. Done with individual 600/929\n",
      "Slice: 1/3. Site: 12. Done with individual 700/929\n",
      "Slice: 1/3. Site: 12. Done with individual 800/929\n",
      "Slice: 1/3. Site: 12. Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.49019607843137253\n",
      "Site index: 13, ref allele frequency: 0.5098039215686274\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.10979547900968784\n",
      "Minor allele frequency: 0.49019607843137253\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 13. Done with individual 0/929\n",
      "Slice: 1/3. Site: 13. Done with individual 100/929\n",
      "Slice: 1/3. Site: 13. Done with individual 200/929\n",
      "Slice: 1/3. Site: 13. Done with individual 300/929\n",
      "Slice: 1/3. Site: 13. Done with individual 400/929\n",
      "Slice: 1/3. Site: 13. Done with individual 500/929\n",
      "Slice: 1/3. Site: 13. Done with individual 600/929\n",
      "Slice: 1/3. Site: 13. Done with individual 700/929\n",
      "Slice: 1/3. Site: 13. Done with individual 800/929\n",
      "Slice: 1/3. Site: 13. Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.4974025974025974\n",
      "Site index: 14, ref allele frequency: 0.5025974025974026\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.41442411194833156\n",
      "Minor allele frequency: 0.4974025974025974\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 14. Done with individual 0/929\n",
      "Slice: 1/3. Site: 14. Done with individual 100/929\n",
      "Slice: 1/3. Site: 14. Done with individual 200/929\n",
      "Slice: 1/3. Site: 14. Done with individual 300/929\n",
      "Slice: 1/3. Site: 14. Done with individual 400/929\n",
      "Slice: 1/3. Site: 14. Done with individual 500/929\n",
      "Slice: 1/3. Site: 14. Done with individual 600/929\n",
      "Slice: 1/3. Site: 14. Done with individual 700/929\n",
      "Slice: 1/3. Site: 14. Done with individual 800/929\n",
      "Slice: 1/3. Site: 14. Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.49567099567099565\n",
      "Site index: 15, ref allele frequency: 0.5043290043290043\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.49730893433799783\n",
      "Minor allele frequency: 0.49567099567099565\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 15. Done with individual 0/929\n",
      "Slice: 1/3. Site: 15. Done with individual 100/929\n",
      "Slice: 1/3. Site: 15. Done with individual 200/929\n",
      "Slice: 1/3. Site: 15. Done with individual 300/929\n",
      "Slice: 1/3. Site: 15. Done with individual 400/929\n",
      "Slice: 1/3. Site: 15. Done with individual 500/929\n",
      "Slice: 1/3. Site: 15. Done with individual 600/929\n",
      "Slice: 1/3. Site: 15. Done with individual 700/929\n",
      "Slice: 1/3. Site: 15. Done with individual 800/929\n",
      "Slice: 1/3. Site: 15. Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.5\n",
      "Site index: 16, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8073196986006459\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 16. Done with individual 0/929\n",
      "Slice: 1/3. Site: 16. Done with individual 100/929\n",
      "Slice: 1/3. Site: 16. Done with individual 200/929\n",
      "Slice: 1/3. Site: 16. Done with individual 300/929\n",
      "Slice: 1/3. Site: 16. Done with individual 400/929\n",
      "Slice: 1/3. Site: 16. Done with individual 500/929\n",
      "Slice: 1/3. Site: 16. Done with individual 600/929\n",
      "Slice: 1/3. Site: 16. Done with individual 700/929\n",
      "Slice: 1/3. Site: 16. Done with individual 800/929\n",
      "Slice: 1/3. Site: 16. Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.5\n",
      "Site index: 17, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.8127018299246501\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 17. Done with individual 0/929\n",
      "Slice: 1/3. Site: 17. Done with individual 100/929\n",
      "Slice: 1/3. Site: 17. Done with individual 200/929\n",
      "Slice: 1/3. Site: 17. Done with individual 300/929\n",
      "Slice: 1/3. Site: 17. Done with individual 400/929\n",
      "Slice: 1/3. Site: 17. Done with individual 500/929\n",
      "Slice: 1/3. Site: 17. Done with individual 600/929\n",
      "Slice: 1/3. Site: 17. Done with individual 700/929\n",
      "Slice: 1/3. Site: 17. Done with individual 800/929\n",
      "Slice: 1/3. Site: 17. Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.5\n",
      "Site index: 18, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5339074273412271\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 18. Done with individual 0/929\n",
      "Slice: 1/3. Site: 18. Done with individual 100/929\n",
      "Slice: 1/3. Site: 18. Done with individual 200/929\n",
      "Slice: 1/3. Site: 18. Done with individual 300/929\n",
      "Slice: 1/3. Site: 18. Done with individual 400/929\n",
      "Slice: 1/3. Site: 18. Done with individual 500/929\n",
      "Slice: 1/3. Site: 18. Done with individual 600/929\n",
      "Slice: 1/3. Site: 18. Done with individual 700/929\n",
      "Slice: 1/3. Site: 18. Done with individual 800/929\n",
      "Slice: 1/3. Site: 18. Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.49377593360995853\n",
      "Site index: 19, ref allele frequency: 0.5062240663900415\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.25941872981700753\n",
      "Minor allele frequency: 0.49377593360995853\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 19. Done with individual 0/929\n",
      "Slice: 1/3. Site: 19. Done with individual 100/929\n",
      "Slice: 1/3. Site: 19. Done with individual 200/929\n",
      "Slice: 1/3. Site: 19. Done with individual 300/929\n",
      "Slice: 1/3. Site: 19. Done with individual 400/929\n",
      "Slice: 1/3. Site: 19. Done with individual 500/929\n",
      "Slice: 1/3. Site: 19. Done with individual 600/929\n",
      "Slice: 1/3. Site: 19. Done with individual 700/929\n",
      "Slice: 1/3. Site: 19. Done with individual 800/929\n",
      "Slice: 1/3. Site: 19. Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.5\n",
      "Site index: 20, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7072120559741658\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 20. Done with individual 0/929\n",
      "Slice: 1/3. Site: 20. Done with individual 100/929\n",
      "Slice: 1/3. Site: 20. Done with individual 200/929\n",
      "Slice: 1/3. Site: 20. Done with individual 300/929\n",
      "Slice: 1/3. Site: 20. Done with individual 400/929\n",
      "Slice: 1/3. Site: 20. Done with individual 500/929\n",
      "Slice: 1/3. Site: 20. Done with individual 600/929\n",
      "Slice: 1/3. Site: 20. Done with individual 700/929\n",
      "Slice: 1/3. Site: 20. Done with individual 800/929\n",
      "Slice: 1/3. Site: 20. Done with individual 900/929\n",
      "Site index: 21, non ref allele frequency: 0.5\n",
      "Site index: 21, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7190527448869752\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 21. Done with individual 0/929\n",
      "Slice: 1/3. Site: 21. Done with individual 100/929\n",
      "Slice: 1/3. Site: 21. Done with individual 200/929\n",
      "Slice: 1/3. Site: 21. Done with individual 300/929\n",
      "Slice: 1/3. Site: 21. Done with individual 400/929\n",
      "Slice: 1/3. Site: 21. Done with individual 500/929\n",
      "Slice: 1/3. Site: 21. Done with individual 600/929\n",
      "Slice: 1/3. Site: 21. Done with individual 700/929\n",
      "Slice: 1/3. Site: 21. Done with individual 800/929\n",
      "Slice: 1/3. Site: 21. Done with individual 900/929\n",
      "Site index: 22, non ref allele frequency: 0.5\n",
      "Site index: 22, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4251883745963401\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 22. Done with individual 0/929\n",
      "Slice: 1/3. Site: 22. Done with individual 100/929\n",
      "Slice: 1/3. Site: 22. Done with individual 200/929\n",
      "Slice: 1/3. Site: 22. Done with individual 300/929\n",
      "Slice: 1/3. Site: 22. Done with individual 400/929\n",
      "Slice: 1/3. Site: 22. Done with individual 500/929\n",
      "Slice: 1/3. Site: 22. Done with individual 600/929\n",
      "Slice: 1/3. Site: 22. Done with individual 700/929\n",
      "Slice: 1/3. Site: 22. Done with individual 800/929\n",
      "Slice: 1/3. Site: 22. Done with individual 900/929\n",
      "Site index: 23, non ref allele frequency: 0.5\n",
      "Site index: 23, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3261571582346609\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 23. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/3. Site: 23. Done with individual 100/929\n",
      "Slice: 1/3. Site: 23. Done with individual 200/929\n",
      "Slice: 1/3. Site: 23. Done with individual 300/929\n",
      "Slice: 1/3. Site: 23. Done with individual 400/929\n",
      "Slice: 1/3. Site: 23. Done with individual 500/929\n",
      "Slice: 1/3. Site: 23. Done with individual 600/929\n",
      "Slice: 1/3. Site: 23. Done with individual 700/929\n",
      "Slice: 1/3. Site: 23. Done with individual 800/929\n",
      "Slice: 1/3. Site: 23. Done with individual 900/929\n",
      "Site index: 24, non ref allele frequency: 0.49458204334365324\n",
      "Site index: 24, ref allele frequency: 0.5054179566563467\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6953713670613563\n",
      "Minor allele frequency: 0.49458204334365324\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 24. Done with individual 0/929\n",
      "Slice: 1/3. Site: 24. Done with individual 100/929\n",
      "Slice: 1/3. Site: 24. Done with individual 200/929\n",
      "Slice: 1/3. Site: 24. Done with individual 300/929\n",
      "Slice: 1/3. Site: 24. Done with individual 400/929\n",
      "Slice: 1/3. Site: 24. Done with individual 500/929\n",
      "Slice: 1/3. Site: 24. Done with individual 600/929\n",
      "Slice: 1/3. Site: 24. Done with individual 700/929\n",
      "Slice: 1/3. Site: 24. Done with individual 800/929\n",
      "Slice: 1/3. Site: 24. Done with individual 900/929\n",
      "Site index: 25, non ref allele frequency: 0.49611801242236025\n",
      "Site index: 25, ref allele frequency: 0.5038819875776398\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6932185145317545\n",
      "Minor allele frequency: 0.49611801242236025\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 25. Done with individual 0/929\n",
      "Slice: 1/3. Site: 25. Done with individual 100/929\n",
      "Slice: 1/3. Site: 25. Done with individual 200/929\n",
      "Slice: 1/3. Site: 25. Done with individual 300/929\n",
      "Slice: 1/3. Site: 25. Done with individual 400/929\n",
      "Slice: 1/3. Site: 25. Done with individual 500/929\n",
      "Slice: 1/3. Site: 25. Done with individual 600/929\n",
      "Slice: 1/3. Site: 25. Done with individual 700/929\n",
      "Slice: 1/3. Site: 25. Done with individual 800/929\n",
      "Slice: 1/3. Site: 25. Done with individual 900/929\n",
      "Site index: 26, non ref allele frequency: 0.5024232633279483\n",
      "Site index: 26, ref allele frequency: 0.4975767366720517\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6663078579117331\n",
      "Minor allele frequency: 0.4975767366720517\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 26. Done with individual 0/929\n",
      "Slice: 1/3. Site: 26. Done with individual 100/929\n",
      "Slice: 1/3. Site: 26. Done with individual 200/929\n",
      "Slice: 1/3. Site: 26. Done with individual 300/929\n",
      "Slice: 1/3. Site: 26. Done with individual 400/929\n",
      "Slice: 1/3. Site: 26. Done with individual 500/929\n",
      "Slice: 1/3. Site: 26. Done with individual 600/929\n",
      "Slice: 1/3. Site: 26. Done with individual 700/929\n",
      "Slice: 1/3. Site: 26. Done with individual 800/929\n",
      "Slice: 1/3. Site: 26. Done with individual 900/929\n",
      "Site index: 27, non ref allele frequency: 0.4985507246376812\n",
      "Site index: 27, ref allele frequency: 0.5014492753623189\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3713670613562971\n",
      "Minor allele frequency: 0.4985507246376812\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 27. Done with individual 0/929\n",
      "Slice: 1/3. Site: 27. Done with individual 100/929\n",
      "Slice: 1/3. Site: 27. Done with individual 200/929\n",
      "Slice: 1/3. Site: 27. Done with individual 300/929\n",
      "Slice: 1/3. Site: 27. Done with individual 400/929\n",
      "Slice: 1/3. Site: 27. Done with individual 500/929\n",
      "Slice: 1/3. Site: 27. Done with individual 600/929\n",
      "Slice: 1/3. Site: 27. Done with individual 700/929\n",
      "Slice: 1/3. Site: 27. Done with individual 800/929\n",
      "Slice: 1/3. Site: 27. Done with individual 900/929\n",
      "Site index: 28, non ref allele frequency: 0.5073746312684366\n",
      "Site index: 28, ref allele frequency: 0.49262536873156343\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7298170075349839\n",
      "Minor allele frequency: 0.49262536873156343\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 28. Done with individual 0/929\n",
      "Slice: 1/3. Site: 28. Done with individual 100/929\n",
      "Slice: 1/3. Site: 28. Done with individual 200/929\n",
      "Slice: 1/3. Site: 28. Done with individual 300/929\n",
      "Slice: 1/3. Site: 28. Done with individual 400/929\n",
      "Slice: 1/3. Site: 28. Done with individual 500/929\n",
      "Slice: 1/3. Site: 28. Done with individual 600/929\n",
      "Slice: 1/3. Site: 28. Done with individual 700/929\n",
      "Slice: 1/3. Site: 28. Done with individual 800/929\n",
      "Slice: 1/3. Site: 28. Done with individual 900/929\n",
      "Site index: 29, non ref allele frequency: 0.494750656167979\n",
      "Site index: 29, ref allele frequency: 0.505249343832021\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4101184068891281\n",
      "Minor allele frequency: 0.494750656167979\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 29. Done with individual 0/929\n",
      "Slice: 1/3. Site: 29. Done with individual 100/929\n",
      "Slice: 1/3. Site: 29. Done with individual 200/929\n",
      "Slice: 1/3. Site: 29. Done with individual 300/929\n",
      "Slice: 1/3. Site: 29. Done with individual 400/929\n",
      "Slice: 1/3. Site: 29. Done with individual 500/929\n",
      "Slice: 1/3. Site: 29. Done with individual 600/929\n",
      "Slice: 1/3. Site: 29. Done with individual 700/929\n",
      "Slice: 1/3. Site: 29. Done with individual 800/929\n",
      "Slice: 1/3. Site: 29. Done with individual 900/929\n",
      "Site index: 30, non ref allele frequency: 0.5067415730337078\n",
      "Site index: 30, ref allele frequency: 0.49325842696629213\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4790096878363832\n",
      "Minor allele frequency: 0.49325842696629213\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 30. Done with individual 0/929\n",
      "Slice: 1/3. Site: 30. Done with individual 100/929\n",
      "Slice: 1/3. Site: 30. Done with individual 200/929\n",
      "Slice: 1/3. Site: 30. Done with individual 300/929\n",
      "Slice: 1/3. Site: 30. Done with individual 400/929\n",
      "Slice: 1/3. Site: 30. Done with individual 500/929\n",
      "Slice: 1/3. Site: 30. Done with individual 600/929\n",
      "Slice: 1/3. Site: 30. Done with individual 700/929\n",
      "Slice: 1/3. Site: 30. Done with individual 800/929\n",
      "Slice: 1/3. Site: 30. Done with individual 900/929\n",
      "Site index: 31, non ref allele frequency: 0.5043668122270742\n",
      "Site index: 31, ref allele frequency: 0.49563318777292575\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2465016146393972\n",
      "Minor allele frequency: 0.49563318777292575\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 31. Done with individual 0/929\n",
      "Slice: 1/3. Site: 31. Done with individual 100/929\n",
      "Slice: 1/3. Site: 31. Done with individual 200/929\n",
      "Slice: 1/3. Site: 31. Done with individual 300/929\n",
      "Slice: 1/3. Site: 31. Done with individual 400/929\n",
      "Slice: 1/3. Site: 31. Done with individual 500/929\n",
      "Slice: 1/3. Site: 31. Done with individual 600/929\n",
      "Slice: 1/3. Site: 31. Done with individual 700/929\n",
      "Slice: 1/3. Site: 31. Done with individual 800/929\n",
      "Slice: 1/3. Site: 31. Done with individual 900/929\n",
      "Site index: 32, non ref allele frequency: 0.4967948717948718\n",
      "Site index: 32, ref allele frequency: 0.5032051282051282\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.33584499461786865\n",
      "Minor allele frequency: 0.4967948717948718\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 32. Done with individual 0/929\n",
      "Slice: 1/3. Site: 32. Done with individual 100/929\n",
      "Slice: 1/3. Site: 32. Done with individual 200/929\n",
      "Slice: 1/3. Site: 32. Done with individual 300/929\n",
      "Slice: 1/3. Site: 32. Done with individual 400/929\n",
      "Slice: 1/3. Site: 32. Done with individual 500/929\n",
      "Slice: 1/3. Site: 32. Done with individual 600/929\n",
      "Slice: 1/3. Site: 32. Done with individual 700/929\n",
      "Slice: 1/3. Site: 32. Done with individual 800/929\n",
      "Slice: 1/3. Site: 32. Done with individual 900/929\n",
      "Site index: 33, non ref allele frequency: 0.49634146341463414\n",
      "Site index: 33, ref allele frequency: 0.5036585365853659\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.4413347685683531\n",
      "Minor allele frequency: 0.49634146341463414\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 33. Done with individual 0/929\n",
      "Slice: 1/3. Site: 33. Done with individual 100/929\n",
      "Slice: 1/3. Site: 33. Done with individual 200/929\n",
      "Slice: 1/3. Site: 33. Done with individual 300/929\n",
      "Slice: 1/3. Site: 33. Done with individual 400/929\n",
      "Slice: 1/3. Site: 33. Done with individual 500/929\n",
      "Slice: 1/3. Site: 33. Done with individual 600/929\n",
      "Slice: 1/3. Site: 33. Done with individual 700/929\n",
      "Slice: 1/3. Site: 33. Done with individual 800/929\n",
      "Slice: 1/3. Site: 33. Done with individual 900/929\n",
      "Site index: 34, non ref allele frequency: 0.49107142857142855\n",
      "Site index: 34, ref allele frequency: 0.5089285714285714\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.12055974165769645\n",
      "Minor allele frequency: 0.49107142857142855\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 34. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/3. Site: 34. Done with individual 100/929\n",
      "Slice: 1/3. Site: 34. Done with individual 200/929\n",
      "Slice: 1/3. Site: 34. Done with individual 300/929\n",
      "Slice: 1/3. Site: 34. Done with individual 400/929\n",
      "Slice: 1/3. Site: 34. Done with individual 500/929\n",
      "Slice: 1/3. Site: 34. Done with individual 600/929\n",
      "Slice: 1/3. Site: 34. Done with individual 700/929\n",
      "Slice: 1/3. Site: 34. Done with individual 800/929\n",
      "Slice: 1/3. Site: 34. Done with individual 900/929\n",
      "Site index: 35, non ref allele frequency: 0.5\n",
      "Site index: 35, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.10118406889128095\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 35. Done with individual 0/929\n",
      "Slice: 1/3. Site: 35. Done with individual 100/929\n",
      "Slice: 1/3. Site: 35. Done with individual 200/929\n",
      "Slice: 1/3. Site: 35. Done with individual 300/929\n",
      "Slice: 1/3. Site: 35. Done with individual 400/929\n",
      "Slice: 1/3. Site: 35. Done with individual 500/929\n",
      "Slice: 1/3. Site: 35. Done with individual 600/929\n",
      "Slice: 1/3. Site: 35. Done with individual 700/929\n",
      "Slice: 1/3. Site: 35. Done with individual 800/929\n",
      "Slice: 1/3. Site: 35. Done with individual 900/929\n",
      "Site index: 36, non ref allele frequency: 0.4954337899543379\n",
      "Site index: 36, ref allele frequency: 0.5045662100456622\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2357373519913886\n",
      "Minor allele frequency: 0.4954337899543379\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 36. Done with individual 0/929\n",
      "Slice: 1/3. Site: 36. Done with individual 100/929\n",
      "Slice: 1/3. Site: 36. Done with individual 200/929\n",
      "Slice: 1/3. Site: 36. Done with individual 300/929\n",
      "Slice: 1/3. Site: 36. Done with individual 400/929\n",
      "Slice: 1/3. Site: 36. Done with individual 500/929\n",
      "Slice: 1/3. Site: 36. Done with individual 600/929\n",
      "Slice: 1/3. Site: 36. Done with individual 700/929\n",
      "Slice: 1/3. Site: 36. Done with individual 800/929\n",
      "Slice: 1/3. Site: 36. Done with individual 900/929\n",
      "Site index: 37, non ref allele frequency: 0.5\n",
      "Site index: 37, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3293864370290635\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 37. Done with individual 0/929\n",
      "Slice: 1/3. Site: 37. Done with individual 100/929\n",
      "Slice: 1/3. Site: 37. Done with individual 200/929\n",
      "Slice: 1/3. Site: 37. Done with individual 300/929\n",
      "Slice: 1/3. Site: 37. Done with individual 400/929\n",
      "Slice: 1/3. Site: 37. Done with individual 500/929\n",
      "Slice: 1/3. Site: 37. Done with individual 600/929\n",
      "Slice: 1/3. Site: 37. Done with individual 700/929\n",
      "Slice: 1/3. Site: 37. Done with individual 800/929\n",
      "Slice: 1/3. Site: 37. Done with individual 900/929\n",
      "Site index: 38, non ref allele frequency: 0.49523809523809526\n",
      "Site index: 38, ref allele frequency: 0.5047619047619047\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5651237890204521\n",
      "Minor allele frequency: 0.49523809523809526\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 38. Done with individual 0/929\n",
      "Slice: 1/3. Site: 38. Done with individual 100/929\n",
      "Slice: 1/3. Site: 38. Done with individual 200/929\n",
      "Slice: 1/3. Site: 38. Done with individual 300/929\n",
      "Slice: 1/3. Site: 38. Done with individual 400/929\n",
      "Slice: 1/3. Site: 38. Done with individual 500/929\n",
      "Slice: 1/3. Site: 38. Done with individual 600/929\n",
      "Slice: 1/3. Site: 38. Done with individual 700/929\n",
      "Slice: 1/3. Site: 38. Done with individual 800/929\n",
      "Slice: 1/3. Site: 38. Done with individual 900/929\n",
      "Site index: 39, non ref allele frequency: 0.4961832061068702\n",
      "Site index: 39, ref allele frequency: 0.5038167938931297\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.14101184068891282\n",
      "Minor allele frequency: 0.4961832061068702\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 39. Done with individual 0/929\n",
      "Slice: 1/3. Site: 39. Done with individual 100/929\n",
      "Slice: 1/3. Site: 39. Done with individual 200/929\n",
      "Slice: 1/3. Site: 39. Done with individual 300/929\n",
      "Slice: 1/3. Site: 39. Done with individual 400/929\n",
      "Slice: 1/3. Site: 39. Done with individual 500/929\n",
      "Slice: 1/3. Site: 39. Done with individual 600/929\n",
      "Slice: 1/3. Site: 39. Done with individual 700/929\n",
      "Slice: 1/3. Site: 39. Done with individual 800/929\n",
      "Slice: 1/3. Site: 39. Done with individual 900/929\n",
      "Site index: 40, non ref allele frequency: 0.49056603773584906\n",
      "Site index: 40, ref allele frequency: 0.5094339622641509\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22820236813778255\n",
      "Minor allele frequency: 0.49056603773584906\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 40. Done with individual 0/929\n",
      "Slice: 1/3. Site: 40. Done with individual 100/929\n",
      "Slice: 1/3. Site: 40. Done with individual 200/929\n",
      "Slice: 1/3. Site: 40. Done with individual 300/929\n",
      "Slice: 1/3. Site: 40. Done with individual 400/929\n",
      "Slice: 1/3. Site: 40. Done with individual 500/929\n",
      "Slice: 1/3. Site: 40. Done with individual 600/929\n",
      "Slice: 1/3. Site: 40. Done with individual 700/929\n",
      "Slice: 1/3. Site: 40. Done with individual 800/929\n",
      "Slice: 1/3. Site: 40. Done with individual 900/929\n",
      "Site index: 41, non ref allele frequency: 0.49193548387096775\n",
      "Site index: 41, ref allele frequency: 0.5080645161290323\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.26695371367061355\n",
      "Minor allele frequency: 0.49193548387096775\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 41. Done with individual 0/929\n",
      "Slice: 1/3. Site: 41. Done with individual 100/929\n",
      "Slice: 1/3. Site: 41. Done with individual 200/929\n",
      "Slice: 1/3. Site: 41. Done with individual 300/929\n",
      "Slice: 1/3. Site: 41. Done with individual 400/929\n",
      "Slice: 1/3. Site: 41. Done with individual 500/929\n",
      "Slice: 1/3. Site: 41. Done with individual 600/929\n",
      "Slice: 1/3. Site: 41. Done with individual 700/929\n",
      "Slice: 1/3. Site: 41. Done with individual 800/929\n",
      "Slice: 1/3. Site: 41. Done with individual 900/929\n",
      "Site index: 42, non ref allele frequency: 0.4934210526315789\n",
      "Site index: 42, ref allele frequency: 0.506578947368421\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.32723358449946177\n",
      "Minor allele frequency: 0.4934210526315789\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 42. Done with individual 0/929\n",
      "Slice: 1/3. Site: 42. Done with individual 100/929\n",
      "Slice: 1/3. Site: 42. Done with individual 200/929\n",
      "Slice: 1/3. Site: 42. Done with individual 300/929\n",
      "Slice: 1/3. Site: 42. Done with individual 400/929\n",
      "Slice: 1/3. Site: 42. Done with individual 500/929\n",
      "Slice: 1/3. Site: 42. Done with individual 600/929\n",
      "Slice: 1/3. Site: 42. Done with individual 700/929\n",
      "Slice: 1/3. Site: 42. Done with individual 800/929\n",
      "Slice: 1/3. Site: 42. Done with individual 900/929\n",
      "Site index: 43, non ref allele frequency: 0.5041551246537396\n",
      "Site index: 43, ref allele frequency: 0.49584487534626037\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.38858988159311086\n",
      "Minor allele frequency: 0.49584487534626037\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 43. Done with individual 0/929\n",
      "Slice: 1/3. Site: 43. Done with individual 100/929\n",
      "Slice: 1/3. Site: 43. Done with individual 200/929\n",
      "Slice: 1/3. Site: 43. Done with individual 300/929\n",
      "Slice: 1/3. Site: 43. Done with individual 400/929\n",
      "Slice: 1/3. Site: 43. Done with individual 500/929\n",
      "Slice: 1/3. Site: 43. Done with individual 600/929\n",
      "Slice: 1/3. Site: 43. Done with individual 700/929\n",
      "Slice: 1/3. Site: 43. Done with individual 800/929\n",
      "Slice: 1/3. Site: 43. Done with individual 900/929\n",
      "Site index: 44, non ref allele frequency: 0.5023584905660378\n",
      "Site index: 44, ref allele frequency: 0.49764150943396224\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22820236813778255\n",
      "Minor allele frequency: 0.49764150943396224\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 44. Done with individual 0/929\n",
      "Slice: 1/3. Site: 44. Done with individual 100/929\n",
      "Slice: 1/3. Site: 44. Done with individual 200/929\n",
      "Slice: 1/3. Site: 44. Done with individual 300/929\n",
      "Slice: 1/3. Site: 44. Done with individual 400/929\n",
      "Slice: 1/3. Site: 44. Done with individual 500/929\n",
      "Slice: 1/3. Site: 44. Done with individual 600/929\n",
      "Slice: 1/3. Site: 44. Done with individual 700/929\n",
      "Slice: 1/3. Site: 44. Done with individual 800/929\n",
      "Slice: 1/3. Site: 44. Done with individual 900/929\n",
      "Site index: 45, non ref allele frequency: 0.49850299401197606\n",
      "Site index: 45, ref allele frequency: 0.5014970059880239\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3595263724434876\n",
      "Minor allele frequency: 0.49850299401197606\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 45. Done with individual 0/929\n",
      "Slice: 1/3. Site: 45. Done with individual 100/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 1/3. Site: 45. Done with individual 200/929\n",
      "Slice: 1/3. Site: 45. Done with individual 300/929\n",
      "Slice: 1/3. Site: 45. Done with individual 400/929\n",
      "Slice: 1/3. Site: 45. Done with individual 500/929\n",
      "Slice: 1/3. Site: 45. Done with individual 600/929\n",
      "Slice: 1/3. Site: 45. Done with individual 700/929\n",
      "Slice: 1/3. Site: 45. Done with individual 800/929\n",
      "Slice: 1/3. Site: 45. Done with individual 900/929\n",
      "Site index: 46, non ref allele frequency: 0.5049019607843137\n",
      "Site index: 46, ref allele frequency: 0.4950980392156863\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.21959095801937567\n",
      "Minor allele frequency: 0.4950980392156863\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 46. Done with individual 0/929\n",
      "Slice: 1/3. Site: 46. Done with individual 100/929\n",
      "Slice: 1/3. Site: 46. Done with individual 200/929\n",
      "Slice: 1/3. Site: 46. Done with individual 300/929\n",
      "Slice: 1/3. Site: 46. Done with individual 400/929\n",
      "Slice: 1/3. Site: 46. Done with individual 500/929\n",
      "Slice: 1/3. Site: 46. Done with individual 600/929\n",
      "Slice: 1/3. Site: 46. Done with individual 700/929\n",
      "Slice: 1/3. Site: 46. Done with individual 800/929\n",
      "Slice: 1/3. Site: 46. Done with individual 900/929\n",
      "Site index: 47, non ref allele frequency: 0.4935064935064935\n",
      "Site index: 47, ref allele frequency: 0.5064935064935064\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.16576964477933262\n",
      "Minor allele frequency: 0.4935064935064935\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 47. Done with individual 0/929\n",
      "Slice: 1/3. Site: 47. Done with individual 100/929\n",
      "Slice: 1/3. Site: 47. Done with individual 200/929\n",
      "Slice: 1/3. Site: 47. Done with individual 300/929\n",
      "Slice: 1/3. Site: 47. Done with individual 400/929\n",
      "Slice: 1/3. Site: 47. Done with individual 500/929\n",
      "Slice: 1/3. Site: 47. Done with individual 600/929\n",
      "Slice: 1/3. Site: 47. Done with individual 700/929\n",
      "Slice: 1/3. Site: 47. Done with individual 800/929\n",
      "Slice: 1/3. Site: 47. Done with individual 900/929\n",
      "Site index: 48, non ref allele frequency: 0.5024390243902439\n",
      "Site index: 48, ref allele frequency: 0.4975609756097561\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.22066738428417654\n",
      "Minor allele frequency: 0.4975609756097561\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 48. Done with individual 0/929\n",
      "Slice: 1/3. Site: 48. Done with individual 100/929\n",
      "Slice: 1/3. Site: 48. Done with individual 200/929\n",
      "Slice: 1/3. Site: 48. Done with individual 300/929\n",
      "Slice: 1/3. Site: 48. Done with individual 400/929\n",
      "Slice: 1/3. Site: 48. Done with individual 500/929\n",
      "Slice: 1/3. Site: 48. Done with individual 600/929\n",
      "Slice: 1/3. Site: 48. Done with individual 700/929\n",
      "Slice: 1/3. Site: 48. Done with individual 800/929\n",
      "Slice: 1/3. Site: 48. Done with individual 900/929\n",
      "Site index: 49, non ref allele frequency: 0.49049429657794674\n",
      "Site index: 49, ref allele frequency: 0.5095057034220533\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2831001076426265\n",
      "Minor allele frequency: 0.49049429657794674\n",
      "Passed guardrails\n",
      "Slice: 1/3. Site: 49. Done with individual 0/929\n",
      "Slice: 1/3. Site: 49. Done with individual 100/929\n",
      "Slice: 1/3. Site: 49. Done with individual 200/929\n",
      "Slice: 1/3. Site: 49. Done with individual 300/929\n",
      "Slice: 1/3. Site: 49. Done with individual 400/929\n",
      "Slice: 1/3. Site: 49. Done with individual 500/929\n",
      "Slice: 1/3. Site: 49. Done with individual 600/929\n",
      "Slice: 1/3. Site: 49. Done with individual 700/929\n",
      "Slice: 1/3. Site: 49. Done with individual 800/929\n",
      "Slice: 1/3. Site: 49. Done with individual 900/929\n",
      "chr9: slice 2/3\n",
      "\n",
      "#############\n",
      "\n",
      "Slice index: 2. Slice size: 50 \n",
      "Output files: C:\\Data\\HUJI\\hgdp\\mock/chr9/count_dist_slice_2.tsv C:\\Data\\HUJI\\hgdp\\mock/chr9/sites_ids_slice_2.tsv\n",
      "Site index: 0, non ref allele frequency: 0.496309963099631\n",
      "Site index: 0, ref allele frequency: 0.503690036900369\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2917115177610334\n",
      "Minor allele frequency: 0.496309963099631\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 0. Done with individual 0/929\n",
      "Slice: 2/3. Site: 0. Done with individual 100/929\n",
      "Slice: 2/3. Site: 0. Done with individual 200/929\n",
      "Slice: 2/3. Site: 0. Done with individual 300/929\n",
      "Slice: 2/3. Site: 0. Done with individual 400/929\n",
      "Slice: 2/3. Site: 0. Done with individual 500/929\n",
      "Slice: 2/3. Site: 0. Done with individual 600/929\n",
      "Slice: 2/3. Site: 0. Done with individual 700/929\n",
      "Slice: 2/3. Site: 0. Done with individual 800/929\n",
      "Slice: 2/3. Site: 0. Done with individual 900/929\n",
      "Site index: 1, non ref allele frequency: 0.49570200573065903\n",
      "Site index: 1, ref allele frequency: 0.504297994269341\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3756727664155005\n",
      "Minor allele frequency: 0.49570200573065903\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 1. Done with individual 0/929\n",
      "Slice: 2/3. Site: 1. Done with individual 100/929\n",
      "Slice: 2/3. Site: 1. Done with individual 200/929\n",
      "Slice: 2/3. Site: 1. Done with individual 300/929\n",
      "Slice: 2/3. Site: 1. Done with individual 400/929\n",
      "Slice: 2/3. Site: 1. Done with individual 500/929\n",
      "Slice: 2/3. Site: 1. Done with individual 600/929\n",
      "Slice: 2/3. Site: 1. Done with individual 700/929\n",
      "Slice: 2/3. Site: 1. Done with individual 800/929\n",
      "Slice: 2/3. Site: 1. Done with individual 900/929\n",
      "Site index: 2, non ref allele frequency: 0.4969230769230769\n",
      "Site index: 2, ref allele frequency: 0.5030769230769231\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.34983853606027987\n",
      "Minor allele frequency: 0.4969230769230769\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 2. Done with individual 0/929\n",
      "Slice: 2/3. Site: 2. Done with individual 100/929\n",
      "Slice: 2/3. Site: 2. Done with individual 200/929\n",
      "Slice: 2/3. Site: 2. Done with individual 300/929\n",
      "Slice: 2/3. Site: 2. Done with individual 400/929\n",
      "Slice: 2/3. Site: 2. Done with individual 500/929\n",
      "Slice: 2/3. Site: 2. Done with individual 600/929\n",
      "Slice: 2/3. Site: 2. Done with individual 700/929\n",
      "Slice: 2/3. Site: 2. Done with individual 800/929\n",
      "Slice: 2/3. Site: 2. Done with individual 900/929\n",
      "Site index: 3, non ref allele frequency: 0.49572649572649574\n",
      "Site index: 3, ref allele frequency: 0.5042735042735043\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.2518837459634015\n",
      "Minor allele frequency: 0.49572649572649574\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 3. Done with individual 0/929\n",
      "Slice: 2/3. Site: 3. Done with individual 100/929\n",
      "Slice: 2/3. Site: 3. Done with individual 200/929\n",
      "Slice: 2/3. Site: 3. Done with individual 300/929\n",
      "Slice: 2/3. Site: 3. Done with individual 400/929\n",
      "Slice: 2/3. Site: 3. Done with individual 500/929\n",
      "Slice: 2/3. Site: 3. Done with individual 600/929\n",
      "Slice: 2/3. Site: 3. Done with individual 700/929\n",
      "Slice: 2/3. Site: 3. Done with individual 800/929\n",
      "Slice: 2/3. Site: 3. Done with individual 900/929\n",
      "Site index: 4, non ref allele frequency: 0.5094850948509485\n",
      "Site index: 4, ref allele frequency: 0.4905149051490515\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.39720129171151775\n",
      "Minor allele frequency: 0.4905149051490515\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 4. Done with individual 0/929\n",
      "Slice: 2/3. Site: 4. Done with individual 100/929\n",
      "Slice: 2/3. Site: 4. Done with individual 200/929\n",
      "Slice: 2/3. Site: 4. Done with individual 300/929\n",
      "Slice: 2/3. Site: 4. Done with individual 400/929\n",
      "Slice: 2/3. Site: 4. Done with individual 500/929\n",
      "Slice: 2/3. Site: 4. Done with individual 600/929\n",
      "Slice: 2/3. Site: 4. Done with individual 700/929\n",
      "Slice: 2/3. Site: 4. Done with individual 800/929\n",
      "Slice: 2/3. Site: 4. Done with individual 900/929\n",
      "Site index: 5, non ref allele frequency: 0.49375866851595007\n",
      "Site index: 5, ref allele frequency: 0.5062413314840499\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7761033369214209\n",
      "Minor allele frequency: 0.49375866851595007\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 5. Done with individual 0/929\n",
      "Slice: 2/3. Site: 5. Done with individual 100/929\n",
      "Slice: 2/3. Site: 5. Done with individual 200/929\n",
      "Slice: 2/3. Site: 5. Done with individual 300/929\n",
      "Slice: 2/3. Site: 5. Done with individual 400/929\n",
      "Slice: 2/3. Site: 5. Done with individual 500/929\n",
      "Slice: 2/3. Site: 5. Done with individual 600/929\n",
      "Slice: 2/3. Site: 5. Done with individual 700/929\n",
      "Slice: 2/3. Site: 5. Done with individual 800/929\n",
      "Slice: 2/3. Site: 5. Done with individual 900/929\n",
      "Site index: 6, non ref allele frequency: 0.5\n",
      "Site index: 6, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.36598493003229277\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 6. Done with individual 0/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 2/3. Site: 6. Done with individual 100/929\n",
      "Slice: 2/3. Site: 6. Done with individual 200/929\n",
      "Slice: 2/3. Site: 6. Done with individual 300/929\n",
      "Slice: 2/3. Site: 6. Done with individual 400/929\n",
      "Slice: 2/3. Site: 6. Done with individual 500/929\n",
      "Slice: 2/3. Site: 6. Done with individual 600/929\n",
      "Slice: 2/3. Site: 6. Done with individual 700/929\n",
      "Slice: 2/3. Site: 6. Done with individual 800/929\n",
      "Slice: 2/3. Site: 6. Done with individual 900/929\n",
      "Site index: 7, non ref allele frequency: 0.5087579617834395\n",
      "Site index: 7, ref allele frequency: 0.4912420382165605\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6759956942949408\n",
      "Minor allele frequency: 0.4912420382165605\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 7. Done with individual 0/929\n",
      "Slice: 2/3. Site: 7. Done with individual 100/929\n",
      "Slice: 2/3. Site: 7. Done with individual 200/929\n",
      "Slice: 2/3. Site: 7. Done with individual 300/929\n",
      "Slice: 2/3. Site: 7. Done with individual 400/929\n",
      "Slice: 2/3. Site: 7. Done with individual 500/929\n",
      "Slice: 2/3. Site: 7. Done with individual 600/929\n",
      "Slice: 2/3. Site: 7. Done with individual 700/929\n",
      "Slice: 2/3. Site: 7. Done with individual 800/929\n",
      "Slice: 2/3. Site: 7. Done with individual 900/929\n",
      "Site index: 8, non ref allele frequency: 0.49920886075949367\n",
      "Site index: 8, ref allele frequency: 0.5007911392405063\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6803013993541442\n",
      "Minor allele frequency: 0.49920886075949367\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 8. Done with individual 0/929\n",
      "Slice: 2/3. Site: 8. Done with individual 100/929\n",
      "Slice: 2/3. Site: 8. Done with individual 200/929\n",
      "Slice: 2/3. Site: 8. Done with individual 300/929\n",
      "Slice: 2/3. Site: 8. Done with individual 400/929\n",
      "Slice: 2/3. Site: 8. Done with individual 500/929\n",
      "Slice: 2/3. Site: 8. Done with individual 600/929\n",
      "Slice: 2/3. Site: 8. Done with individual 700/929\n",
      "Slice: 2/3. Site: 8. Done with individual 800/929\n",
      "Slice: 2/3. Site: 8. Done with individual 900/929\n",
      "Site index: 9, non ref allele frequency: 0.5083333333333333\n",
      "Site index: 9, ref allele frequency: 0.49166666666666664\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.77502691065662\n",
      "Minor allele frequency: 0.49166666666666664\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 9. Done with individual 0/929\n",
      "Slice: 2/3. Site: 9. Done with individual 100/929\n",
      "Slice: 2/3. Site: 9. Done with individual 200/929\n",
      "Slice: 2/3. Site: 9. Done with individual 300/929\n",
      "Slice: 2/3. Site: 9. Done with individual 400/929\n",
      "Slice: 2/3. Site: 9. Done with individual 500/929\n",
      "Slice: 2/3. Site: 9. Done with individual 600/929\n",
      "Slice: 2/3. Site: 9. Done with individual 700/929\n",
      "Slice: 2/3. Site: 9. Done with individual 800/929\n",
      "Slice: 2/3. Site: 9. Done with individual 900/929\n",
      "Site index: 10, non ref allele frequency: 0.5085106382978724\n",
      "Site index: 10, ref allele frequency: 0.49148936170212765\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7588805166846071\n",
      "Minor allele frequency: 0.49148936170212765\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 10. Done with individual 0/929\n",
      "Slice: 2/3. Site: 10. Done with individual 100/929\n",
      "Slice: 2/3. Site: 10. Done with individual 200/929\n",
      "Slice: 2/3. Site: 10. Done with individual 300/929\n",
      "Slice: 2/3. Site: 10. Done with individual 400/929\n",
      "Slice: 2/3. Site: 10. Done with individual 500/929\n",
      "Slice: 2/3. Site: 10. Done with individual 600/929\n",
      "Slice: 2/3. Site: 10. Done with individual 700/929\n",
      "Slice: 2/3. Site: 10. Done with individual 800/929\n",
      "Slice: 2/3. Site: 10. Done with individual 900/929\n",
      "Site index: 11, non ref allele frequency: 0.5063191153238547\n",
      "Site index: 11, ref allele frequency: 0.49368088467614535\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6813778256189451\n",
      "Minor allele frequency: 0.49368088467614535\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 11. Done with individual 0/929\n",
      "Slice: 2/3. Site: 11. Done with individual 100/929\n",
      "Slice: 2/3. Site: 11. Done with individual 200/929\n",
      "Slice: 2/3. Site: 11. Done with individual 300/929\n",
      "Slice: 2/3. Site: 11. Done with individual 400/929\n",
      "Slice: 2/3. Site: 11. Done with individual 500/929\n",
      "Slice: 2/3. Site: 11. Done with individual 600/929\n",
      "Slice: 2/3. Site: 11. Done with individual 700/929\n",
      "Slice: 2/3. Site: 11. Done with individual 800/929\n",
      "Slice: 2/3. Site: 11. Done with individual 900/929\n",
      "Site index: 12, non ref allele frequency: 0.5097629009762901\n",
      "Site index: 12, ref allele frequency: 0.4902370990237099\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7717976318622174\n",
      "Minor allele frequency: 0.4902370990237099\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 12. Done with individual 0/929\n",
      "Slice: 2/3. Site: 12. Done with individual 100/929\n",
      "Slice: 2/3. Site: 12. Done with individual 200/929\n",
      "Slice: 2/3. Site: 12. Done with individual 300/929\n",
      "Slice: 2/3. Site: 12. Done with individual 400/929\n",
      "Slice: 2/3. Site: 12. Done with individual 500/929\n",
      "Slice: 2/3. Site: 12. Done with individual 600/929\n",
      "Slice: 2/3. Site: 12. Done with individual 700/929\n",
      "Slice: 2/3. Site: 12. Done with individual 800/929\n",
      "Slice: 2/3. Site: 12. Done with individual 900/929\n",
      "Site index: 13, non ref allele frequency: 0.5021582733812949\n",
      "Site index: 13, ref allele frequency: 0.497841726618705\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7481162540365985\n",
      "Minor allele frequency: 0.497841726618705\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 13. Done with individual 0/929\n",
      "Slice: 2/3. Site: 13. Done with individual 100/929\n",
      "Slice: 2/3. Site: 13. Done with individual 200/929\n",
      "Slice: 2/3. Site: 13. Done with individual 300/929\n",
      "Slice: 2/3. Site: 13. Done with individual 400/929\n",
      "Slice: 2/3. Site: 13. Done with individual 500/929\n",
      "Slice: 2/3. Site: 13. Done with individual 600/929\n",
      "Slice: 2/3. Site: 13. Done with individual 700/929\n",
      "Slice: 2/3. Site: 13. Done with individual 800/929\n",
      "Slice: 2/3. Site: 13. Done with individual 900/929\n",
      "Site index: 14, non ref allele frequency: 0.5053929121725732\n",
      "Site index: 14, ref allele frequency: 0.4946070878274268\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6986006458557589\n",
      "Minor allele frequency: 0.4946070878274268\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 14. Done with individual 0/929\n",
      "Slice: 2/3. Site: 14. Done with individual 100/929\n",
      "Slice: 2/3. Site: 14. Done with individual 200/929\n",
      "Slice: 2/3. Site: 14. Done with individual 300/929\n",
      "Slice: 2/3. Site: 14. Done with individual 400/929\n",
      "Slice: 2/3. Site: 14. Done with individual 500/929\n",
      "Slice: 2/3. Site: 14. Done with individual 600/929\n",
      "Slice: 2/3. Site: 14. Done with individual 700/929\n",
      "Slice: 2/3. Site: 14. Done with individual 800/929\n",
      "Slice: 2/3. Site: 14. Done with individual 900/929\n",
      "Site index: 15, non ref allele frequency: 0.5078125\n",
      "Site index: 15, ref allele frequency: 0.4921875\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.6889128094725512\n",
      "Minor allele frequency: 0.4921875\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 15. Done with individual 0/929\n",
      "Slice: 2/3. Site: 15. Done with individual 100/929\n",
      "Slice: 2/3. Site: 15. Done with individual 200/929\n",
      "Slice: 2/3. Site: 15. Done with individual 300/929\n",
      "Slice: 2/3. Site: 15. Done with individual 400/929\n",
      "Slice: 2/3. Site: 15. Done with individual 500/929\n",
      "Slice: 2/3. Site: 15. Done with individual 600/929\n",
      "Slice: 2/3. Site: 15. Done with individual 700/929\n",
      "Slice: 2/3. Site: 15. Done with individual 800/929\n",
      "Slice: 2/3. Site: 15. Done with individual 900/929\n",
      "Site index: 16, non ref allele frequency: 0.5\n",
      "Site index: 16, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7707212055974165\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 16. Done with individual 0/929\n",
      "Slice: 2/3. Site: 16. Done with individual 100/929\n",
      "Slice: 2/3. Site: 16. Done with individual 200/929\n",
      "Slice: 2/3. Site: 16. Done with individual 300/929\n",
      "Slice: 2/3. Site: 16. Done with individual 400/929\n",
      "Slice: 2/3. Site: 16. Done with individual 500/929\n",
      "Slice: 2/3. Site: 16. Done with individual 600/929\n",
      "Slice: 2/3. Site: 16. Done with individual 700/929\n",
      "Slice: 2/3. Site: 16. Done with individual 800/929\n",
      "Slice: 2/3. Site: 16. Done with individual 900/929\n",
      "Site index: 17, non ref allele frequency: 0.49122807017543857\n",
      "Site index: 17, ref allele frequency: 0.5087719298245614\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.3681377825618945\n",
      "Minor allele frequency: 0.49122807017543857\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 17. Done with individual 0/929\n",
      "Slice: 2/3. Site: 17. Done with individual 100/929\n",
      "Slice: 2/3. Site: 17. Done with individual 200/929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Slice: 2/3. Site: 17. Done with individual 300/929\n",
      "Slice: 2/3. Site: 17. Done with individual 400/929\n",
      "Slice: 2/3. Site: 17. Done with individual 500/929\n",
      "Slice: 2/3. Site: 17. Done with individual 600/929\n",
      "Slice: 2/3. Site: 17. Done with individual 700/929\n",
      "Slice: 2/3. Site: 17. Done with individual 800/929\n",
      "Slice: 2/3. Site: 17. Done with individual 900/929\n",
      "Site index: 18, non ref allele frequency: 0.5068762278978389\n",
      "Site index: 18, ref allele frequency: 0.4931237721021611\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.5479009687836384\n",
      "Minor allele frequency: 0.4931237721021611\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 18. Done with individual 0/929\n",
      "Slice: 2/3. Site: 18. Done with individual 100/929\n",
      "Slice: 2/3. Site: 18. Done with individual 200/929\n",
      "Slice: 2/3. Site: 18. Done with individual 300/929\n",
      "Slice: 2/3. Site: 18. Done with individual 400/929\n",
      "Slice: 2/3. Site: 18. Done with individual 500/929\n",
      "Slice: 2/3. Site: 18. Done with individual 600/929\n",
      "Slice: 2/3. Site: 18. Done with individual 700/929\n",
      "Slice: 2/3. Site: 18. Done with individual 800/929\n",
      "Slice: 2/3. Site: 18. Done with individual 900/929\n",
      "Site index: 19, non ref allele frequency: 0.5007246376811594\n",
      "Site index: 19, ref allele frequency: 0.49927536231884057\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7427341227125942\n",
      "Minor allele frequency: 0.49927536231884057\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 19. Done with individual 0/929\n",
      "Slice: 2/3. Site: 19. Done with individual 100/929\n",
      "Slice: 2/3. Site: 19. Done with individual 200/929\n",
      "Slice: 2/3. Site: 19. Done with individual 300/929\n",
      "Slice: 2/3. Site: 19. Done with individual 400/929\n",
      "Slice: 2/3. Site: 19. Done with individual 500/929\n",
      "Slice: 2/3. Site: 19. Done with individual 600/929\n",
      "Slice: 2/3. Site: 19. Done with individual 700/929\n",
      "Slice: 2/3. Site: 19. Done with individual 800/929\n",
      "Slice: 2/3. Site: 19. Done with individual 900/929\n",
      "Site index: 20, non ref allele frequency: 0.5\n",
      "Site index: 20, ref allele frequency: 0.5\n",
      "Check guardrails\n",
      "Precentage of valid sites: 0.7405812701829925\n",
      "Minor allele frequency: 0.5\n",
      "Passed guardrails\n",
      "Slice: 2/3. Site: 20. Done with individual 0/929\n",
      "Slice: 2/3. Site: 20. Done with individual 100/929\n",
      "Slice: 2/3. Site: 20. Done with individual 200/929\n",
      "Slice: 2/3. Site: 20. Done with individual 300/929\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-10-786467d26af2>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mbase_output_path\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'C:\\Data\\HUJI\\hgdp\\mock'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mbuild_per_slice_distances_matrix\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpath_to_012_files\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbase_output_path\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdesired_num_of_sites_per_output\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m100\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_minor_freq_expected\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.49\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_minor_freq_expected\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.5\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_valid_sites_precentage\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m \u001b[0mbuild_per_slice_distances_matrix\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpath_to_012_files\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbase_output_path\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdesired_num_of_sites_per_output\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m50\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_minor_freq_expected\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.49\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_minor_freq_expected\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.5\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_valid_sites_precentage\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0.1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-8-fce180065d8b>\u001b[0m in \u001b[0;36mbuild_per_slice_distances_matrix\u001b[1;34m(path_to_012_files, base_output_path, desired_num_of_sites_per_output, min_minor_freq_expected, max_minor_freq_expected, min_valid_sites_precentage)\u001b[0m\n\u001b[0;32m     61\u001b[0m         \u001b[0m_write_output_sites_file\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0moutput_sites_file\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_df\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     62\u001b[0m         \u001b[0m_log\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf'Output files: {output_count_dist_file} {output_sites_file}'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 63\u001b[1;33m         \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_slice_calc_pairwise_distances_with_guardrails\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnum_of_slices\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_df\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_valid_sites_precentage\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_minor_freq_expected\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_minor_freq_expected\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     64\u001b[0m         \u001b[0m_write_pairwise_distances\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0moutput_count_dist_file\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-4-18aaeeacec48>\u001b[0m in \u001b[0;36m_slice_calc_pairwise_distances_with_guardrails\u001b[1;34m(slice_index, num_of_slices, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected)\u001b[0m\n\u001b[0;32m     25\u001b[0m     \u001b[0mslice_pairwise_dist\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_build_pairwise_db\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_df\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m0.0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     26\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0msite_index\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 27\u001b[1;33m         \u001b[0m_site_calc_pairwise_distances_with_guardrails\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnum_of_slices\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msite_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_df\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_valid_sites_precentage\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_minor_freq_expected\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_minor_freq_expected\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     28\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     29\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-4-18aaeeacec48>\u001b[0m in \u001b[0;36m_site_calc_pairwise_distances_with_guardrails\u001b[1;34m(slice_index, num_of_slices, site_index, slice_df, min_valid_sites_precentage, min_minor_freq_expected, max_minor_freq_expected, slice_pairwise_counts, slice_pairwise_dist)\u001b[0m\n\u001b[0;32m     67\u001b[0m     \u001b[1;31m# Note that if we split this by chromozome, the big ones may have around 1000 in a given freq range,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     68\u001b[0m     \u001b[1;31m# which can take ~8 minutes. 10k will be around 83 minutes - also cool.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 69\u001b[1;33m     \u001b[0m_site_calc_pairwise_distances\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnum_of_slices\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msite_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mgenotypes\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnum_individuals\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mref_freq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_ref_freq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     70\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     71\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-4-18aaeeacec48>\u001b[0m in \u001b[0;36m_site_calc_pairwise_distances\u001b[1;34m(slice_index, num_of_slices, site_index, genotypes, num_individuals, ref_freq, non_ref_freq, slice_pairwise_counts, slice_pairwise_dist)\u001b[0m\n\u001b[0;32m     44\u001b[0m                 \u001b[1;31m# this is a valid entry, we add 1 to the count\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     45\u001b[0m                 \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi2\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mi1\u001b[0m\u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 46\u001b[1;33m                 \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi2\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mi1\u001b[0m\u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m+=\u001b[0m \u001b[0m_calc_dist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mi1_val\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mi2_val\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mref_freq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_ref_freq\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     47\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     48\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0m_site_calc_pairwise_distances_with_guardrails\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslice_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnum_of_slices\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msite_index\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_df\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_valid_sites_precentage\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmin_minor_freq_expected\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_minor_freq_expected\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_counts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mslice_pairwise_dist\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-4-18aaeeacec48>\u001b[0m in \u001b[0;36m_calc_dist\u001b[1;34m(i1_val, i2_val, ref_freq, non_ref_freq)\u001b[0m\n\u001b[0;32m    109\u001b[0m     \u001b[1;31m# formula to use:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    110\u001b[0m     \u001b[1;31m#   (1-f_non_ref)(v1_val*v2_val) + (1-f_ref)((2-v1_val)*(2-v2_val))/4\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 111\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mnon_ref_freq\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m*\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mi1_val\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0mi2_val\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mref_freq\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m*\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mi1_val\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m*\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mi2_val\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m/\u001b[0m\u001b[1;36m4\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    112\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    113\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "path_to_012_files = 'C:\\Data\\HUJI\\hgdp\\mock\\out'\n",
    "base_output_path = 'C:\\Data\\HUJI\\hgdp\\mock'\n",
    "build_per_slice_distances_matrix(path_to_012_files, base_output_path, desired_num_of_sites_per_output=100, min_minor_freq_expected=0.49, max_minor_freq_expected=0.5, min_valid_sites_precentage=0.1)\n",
    "build_per_slice_distances_matrix(path_to_012_files, base_output_path, desired_num_of_sites_per_output=50, min_minor_freq_expected=0.49, max_minor_freq_expected=0.5, min_valid_sites_precentage=0.1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-06T08:39:08.899839Z",
     "start_time": "2021-01-06T08:39:08.886624Z"
    }
   },
   "outputs": [],
   "source": [
    "#merge dist files\n",
    "def merge_slices_dist_counts_files(files):\n",
    "    # get number of individuals from the first file:\n",
    "    with open(files[0],'r') as f:\n",
    "        l = f.readline()\n",
    "        n = len(l.split(' '))+1\n",
    "    # init\n",
    "    pairwise_counts = _build_pairwise_db(n, 0)\n",
    "    pairwise_dist = _build_pairwise_db(n, 0.0)\n",
    "\n",
    "    # go over the files and sum the values\n",
    "    for f in files:\n",
    "        with open(f,'r') as f:\n",
    "            # note that while i can represent the index of the individual, j doesnot - it is merly the entrance in the matrix.\n",
    "            i=-1\n",
    "            for line in f.readlines():\n",
    "                i+=1\n",
    "                # drop the \\n\n",
    "                cs_ds = line[:-1].split(' ')\n",
    "                j=-1\n",
    "                # go over the counts(cs) and distances(ds)\n",
    "                for c_d in cs_ds:\n",
    "                    j+=1\n",
    "                    c,d = c_d.split(',')\n",
    "                    pairwise_counts[i][j]+=int(c)\n",
    "                    pairwise_dist[i][j]+=float(d)\n",
    "    return pairwise_counts, pairwise_dist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-01-06T08:40:32.571880Z",
     "start_time": "2021-01-06T08:40:31.190702Z"
    }
   },
   "outputs": [],
   "source": [
    "p1 = r\"C:\\Data\\HUJI\\hgdp\\mock\\chr9\\count_dist_slice_7.tsv\"\n",
    "p2 = r\"C:\\Data\\HUJI\\hgdp\\mock\\chr9\\count_dist_slice_8.tsv\"\n",
    "files = [p1, p2]\n",
    "pairwise_counts, pairwise_dist = merge_slices_dist_counts_files(files)\n",
    "output_merged_count_dist_file = r\"C:\\Data\\HUJI\\hgdp\\mock\\chr9\\count_dist_merged.tsv\"\n",
    "_write_pairwise_distances(output_merged_count_dist_file, pairwise_counts, pairwise_dist)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adhoc playground"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compare vcf file sizes (cluster vs source ftp server)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-12-24T18:32:44.953204Z",
     "start_time": "2020-12-24T18:32:44.923134Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size1</th>\n",
       "      <th>key</th>\n",
       "      <th>size2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [size1, key, size2]\n",
       "Index: []"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sizes_cluster = \"C:\\Data\\HUJI\\hgdp\\sizes_cluster.tsv\"\n",
    "sizes_source = \"C:\\Data\\HUJI\\hgdp\\sizes_source.tsv\"\n",
    "import pandas as pd\n",
    "source = pd.read_csv(sizes_source, sep='\\t', header=None, names =  ['key','size2'])\n",
    "source['size2'] = source['size2'].str.replace(' KB', '')\n",
    "source.head()\n",
    "cluster = pd.read_csv(sizes_cluster, sep=' ', header=None, names =  ['size1','key'])\n",
    "\n",
    "cluster.head()\n",
    "joined = cluster.join(source.set_index('key'), on='key')\n",
    "joined = joined[joined['key'].str.endswith('.gz')]\n",
    "joined[joined['size1']==joined['size2']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "261.867px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "oldHeight": 623.516666,
   "position": {
    "height": "40px",
    "left": "1242.33px",
    "right": "20px",
    "top": "120px",
    "width": "350px"
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "varInspector_section_display": "none",
   "window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
